# WGS

## Files in repo
all_mapping_stats.csv ->  All mapping statistics generated by flagstat, including Paired/Num.reads, Coverage, and Total reads.

SPU_LOC_conv.txt      ->  Downloaded from String DB, conversion between the old SPU and the new LOC tags of urchin genes.

annotation.txt        ->  Chr, start and stop position for every urchin gene. Downloaded from NCBI (as gff, v5.0), only rows labelled "gene" were kept.

bayenv_conversion.py -> 

## Steps

### Step 1
From raw sequence data to SNPs -> [Sequence processing](https://github.com/Cpetak/WGS/blob/main/sequence_processing.md)

### Step 2
From temporal pH and temperature data to scaled mean and variability -> [Environmental data processing](https://github.com/Cpetak/WGS/blob/main/env_data_processing.R)

### Step 3
Converting the output of ANGSD with the SNPs into the format Bayenv expects -> [Converting for Bayenv](https://github.com/Cpetak/WGS/blob/main/bayenv_conversion.py)

### Step 4
Estimating covarience matrix and Bayenv factors -> [Running Bayenv](https://github.com/Cpetak/WGS/blob/main/Running_bayenv.md)

### Step 5
Generating dataframe with Bayenv factor, number of neighbors, expression level and SNP number -> [Feature engineering](https://github.com/Cpetak/WGS/blob/main/get_gene_from_snp.ipynb)

### Step 6
Exploring and modeling variables -> [Playing with variables](https://github.com/Cpetak/WGS/blob/main/playing_with_variables-new.ipynb)
