{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop(df[df['type'] == \"region\"].index, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gene', 'mRNA', 'exon', 'CDS', 'snRNA', 'pseudogene', 'cDNA_match',\n",
       "       'tRNA', 'lnc_RNA', 'transcript', 'snoRNA', 'rRNA',\n",
       "       'primary_transcript', 'miRNA', 'sequence_feature', 'tandem_repeat',\n",
       "       'promoter'], dtype=object)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci = pd.read_csv('top_goodhe_01', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR1\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR1\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR1\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR1\n",
      "ERROR2\n",
      "ERROR2\n",
      "ERROR1\n"
     ]
    }
   ],
   "source": [
    "new_df=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    temp=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    if len(temp) == 1:\n",
    "        print(\"only one\")\n",
    "        print(temp['type'])\n",
    "    if len(temp) > 1:\n",
    "        #print(\"more than 1\")\n",
    "        #checking for gene overlap\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    ids.append(ID)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        if uni > 1:\n",
    "            #print(\"genes overlap\")\n",
    "            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"genes_overlap\", \"gene\": \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "        \n",
    "        if uni == 1:\n",
    "            if temp['type'].str.contains('gene').any():\n",
    "                if temp['type'].str.contains('pseudogene').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                else:\n",
    "                    if temp['type'].str.contains('exon').any(): \n",
    "                        if temp['type'].str.contains('CDS').any():\n",
    "                            #print(\"in exon\")\n",
    "                            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"exon\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                        else:\n",
    "                            direction=temp['s2'].iloc[0]\n",
    "                            e_start=temp[temp[\"type\"]==\"exon\"].iloc[0].start\n",
    "                            e_stop=temp[temp[\"type\"]==\"exon\"].iloc[0].stop\n",
    "                            g_start=temp[temp[\"type\"]==\"gene\"].iloc[0].start\n",
    "                            g_stop=temp[temp[\"type\"]==\"gene\"].iloc[0].stop\n",
    "                            if direction == \"+\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('5prime1')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"3prime1\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                else:\n",
    "                                    print(\"ERROR1\") #alternative UTR?\n",
    "                                    #print(temp)\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            elif direction == \"-\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('3prime2')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"5prime2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True)\n",
    "                                else:\n",
    "                                    print(\"ERROR2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            else:\n",
    "                                print(\"ERROR3\")\n",
    "                                #print(temp)\n",
    "\n",
    "                    else:\n",
    "                        #print(\"in intron\")\n",
    "                        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"intron\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "\n",
    "            else:\n",
    "                print(\"not gene\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>809765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>22684111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>36849648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>13121687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>20882520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>27269202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>25996126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>alternative UTR</td>\n",
       "      <td>12214610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>3170334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>LOC582877</td>\n",
       "      <td>intron</td>\n",
       "      <td>16380245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr       gene              hol       pos\n",
       "0     NW_022145599.1         NA        not_annot    809765\n",
       "1     NW_022145613.1         NA        not_annot  22684111\n",
       "2     NW_022145600.1         NA        not_annot  36849648\n",
       "3     NW_022145612.1         NA        not_annot  13121687\n",
       "4     NW_022145595.1         NA        not_annot  20882520\n",
       "...              ...        ...              ...       ...\n",
       "6842  NW_022145597.1         NA        not_annot  27269202\n",
       "6843  NW_022145597.1         NA        not_annot  25996126\n",
       "6844  NW_022145596.1         NA  alternative UTR  12214610\n",
       "6845  NW_022145595.1         NA        not_annot   3170334\n",
       "6846  NW_022145595.1  LOC582877           intron  16380245\n",
       "\n",
       "[6847 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.pos = new_df.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.to_csv(\"annotated_top_goodhe_01.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_df[new_df[\"hol\"]==\"pseudogene\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['not_annot', 'exon', \"3'UTR\", 'intron', 'genes_overlap',\n",
       "       'alternative UTR', \"5'UTR\", 'pseudogene'], dtype=object)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.hol.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, outside of intron"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "are intronic ones located in:\n",
    "    1 ATAC\n",
    "    2 lvar\n",
    "    3 chip \n",
    "    4 lnc\n",
    "    5 promoter\n",
    "    6 TFBS\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])\n",
    "promoters = promoters[promoters['type']==\"gene\"]\n",
    "\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"+\"), \"start\"] -= 1000\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"-\"), \"stop\"] += 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = pd.read_csv(\"atac_mapped_translatedto5.0.csv\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "chip = pd.read_csv('AlxChip_mapped_translatedto5.0.txt', sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lvar = pd.read_csv('lvar_spur_48_50_mapped_translatedto50.csv', header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lnc = pd.read_csv(\"lncRNA_mapped_translatedto5.0.csv\", header=1, names=[\"chr\",\"start\",\"stop\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"not_annot\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in chip\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lvar\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in chip\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in chip\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in chip\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lvar\n",
      "in chip\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "enh_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in promoters\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"promoter\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['promoter', 'lnc', 'lvar', 'chip'], dtype=object)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en = enh_check[(enh_check[\"hol\"]==\"lvar\") | (enh_check[\"hol\"]==\"chip\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en=pot_en.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pot_en)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reminder: chosen atac supp (or lvar or chip) too strict?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, inside of intron"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"intron\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in atac\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in atac\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "int_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['lnc', 'chip', 'atac', 'lvar'], dtype=object)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>8855042.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>18662627.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>47720769.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>35316800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16140373.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16695182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr   hol         pos\n",
       "83   NW_022145604.1  lvar   8855042.0\n",
       "96   NW_022145600.1  lvar  18662627.0\n",
       "97   NW_022145612.1  lvar  47720769.0\n",
       "100  NW_022145610.1  lvar  35316800.0\n",
       "114  NW_022145595.1  lvar  16140373.0\n",
       "118  NW_022145605.1  lvar  16695182.0\n",
       "142  NW_022145612.1  lvar  14929793.0"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check[int_check[\"hol\"]==\"lvar\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in = int_check[(int_check[\"hol\"]==\"lvar\") | (int_check[\"hol\"]==\"chip\") | (int_check[\"hol\"]==\"atac\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en=pot_en.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>32414825.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>15623469.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>23728887.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>11083748.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>24095444.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>1071667.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>21062821.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>7615785.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>5814450.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>11429281.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>18717183.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>NW_022145607.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>5727443.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NW_022145607.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>31512598.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>NW_022145611.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>11486890.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>NW_022145611.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>32018192.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>NW_022145611.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>35224169.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>37776829.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>chip</td>\n",
       "      <td>61822570.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>lvar</td>\n",
       "      <td>30714752.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr   hol         pos\n",
       "77   NW_022145594.1  lvar  32414825.0\n",
       "247  NW_022145597.1  chip  15623469.0\n",
       "199  NW_022145597.1  chip  23728887.0\n",
       "76   NW_022145598.1  lvar  11083748.0\n",
       "339  NW_022145598.1  lvar  24095444.0\n",
       "246  NW_022145600.1  lvar   1071667.0\n",
       "43   NW_022145600.1  chip  21062821.0\n",
       "224  NW_022145602.1  chip   7615785.0\n",
       "140  NW_022145603.1  chip   5814450.0\n",
       "104  NW_022145604.1  chip  11429281.0\n",
       "119  NW_022145605.1  lvar  18717183.0\n",
       "175  NW_022145607.1  chip   5727443.0\n",
       "48   NW_022145607.1  lvar  31512598.0\n",
       "98   NW_022145611.1  lvar  11486890.0\n",
       "198  NW_022145611.1  chip  32018192.0\n",
       "327  NW_022145611.1  lvar  35224169.0\n",
       "7    NW_022145612.1  lvar  37776829.0\n",
       "248  NW_022145612.1  chip  61822570.0\n",
       "166  NW_022145613.1  lvar  30714752.0"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pot_en"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
