{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop(df[df['type'] == \"region\"].index, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci = pd.read_csv('top_goodhe', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR1\n"
     ]
    }
   ],
   "source": [
    "new_df=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    temp=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    if len(temp) == 1:\n",
    "        print(\"only one\")\n",
    "        print(temp['type'])\n",
    "    if len(temp) > 1:\n",
    "        #print(\"more than 1\")\n",
    "        #checking for gene overlap\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    ids.append(ID)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        if uni > 1:\n",
    "            #print(\"genes overlap\")\n",
    "            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"genes_overlap\", \"gene\": \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "        \n",
    "        if uni == 1:\n",
    "            if temp['type'].str.contains('gene').any():\n",
    "                if temp['type'].str.contains('exon').any(): \n",
    "                    if temp['type'].str.contains('CDS').any():\n",
    "                        #print(\"in exon\")\n",
    "                        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"exon\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                    else:\n",
    "                        direction=temp['s2'].iloc[0]\n",
    "                        e_start=temp[temp[\"type\"]==\"exon\"].iloc[0].start\n",
    "                        e_stop=temp[temp[\"type\"]==\"exon\"].iloc[0].stop\n",
    "                        g_start=temp[temp[\"type\"]==\"gene\"].iloc[0].start\n",
    "                        g_stop=temp[temp[\"type\"]==\"gene\"].iloc[0].stop\n",
    "                        if direction == \"+\":\n",
    "                            if e_start == g_start:\n",
    "                                #print('5prime1')\n",
    "                                new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                            elif e_stop == g_stop:\n",
    "                                #print(\"3prime1\")\n",
    "                                new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                            else:\n",
    "                                print(\"ERROR1\") #alternative UTR?\n",
    "                                #print(temp)\n",
    "                                new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                        elif direction == \"-\":\n",
    "                            if e_start == g_start:\n",
    "                                #print('3prime2')\n",
    "                                new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                            elif e_stop == g_stop:\n",
    "                                #print(\"5prime2\")\n",
    "                                new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True)\n",
    "                            else:\n",
    "                                print(\"ERROR2\")\n",
    "                                new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                        else:\n",
    "                            print(\"ERROR3\")\n",
    "                            #print(temp)\n",
    "\n",
    "                else:\n",
    "                    #print(\"in intron\")\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"intron\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "\n",
    "            else:\n",
    "                print(\"not gene\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.pos = new_df.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>LOC752046</td>\n",
       "      <td>exon</td>\n",
       "      <td>22321899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NW_022145609.1</td>\n",
       "      <td>LOC589355</td>\n",
       "      <td>exon</td>\n",
       "      <td>7625794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC100891707</td>\n",
       "      <td>exon</td>\n",
       "      <td>14824625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105447203</td>\n",
       "      <td>exon</td>\n",
       "      <td>11729117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579596</td>\n",
       "      <td>exon</td>\n",
       "      <td>30320957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>LOC594103</td>\n",
       "      <td>exon</td>\n",
       "      <td>22238462</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr          gene   hol       pos\n",
       "5   NW_022145603.1     LOC752046  exon  22321899\n",
       "47  NW_022145609.1     LOC589355  exon   7625794\n",
       "48  NW_022145598.1  LOC100891707  exon  14824625\n",
       "62  NW_022145612.1  LOC105447203  exon  11729117\n",
       "64  NW_022145596.1     LOC579596  exon  30320957\n",
       "66  NW_022145604.1     LOC594103  exon  22238462"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df[new_df[\"hol\"]==\"exon\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#proteins of exons:  homeodomain-interacting protein kinase 2, transcription activator BRG1 (Imschenetzky 2003, might not have role, have to check)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#locomotion-related protein Hikaru genki, uncharaterised, carbonic anhydrase-related protein (Sp-Cah10L) - melissa's ultimate list, differentially expressed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#tyrosine-protein phosphatase non-receptor type 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reminder: we want bayenv!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105446642</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>21838405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC752451</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>27036813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC581215</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>51908692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>LOC581450</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>488171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC100890874</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>30238726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105446336</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>17902099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>LOC589754</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>25795628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC591109</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>30045837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC581215</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>51908700</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr          gene    hol       pos\n",
       "6   NW_022145612.1  LOC105446642  3'UTR  21838405\n",
       "19  NW_022145610.1     LOC752451  3'UTR  27036813\n",
       "20  NW_022145612.1     LOC581215  3'UTR  51908692\n",
       "38  NW_022145615.1     LOC581450  3'UTR    488171\n",
       "40  NW_022145610.1  LOC100890874  3'UTR  30238726\n",
       "41  NW_022145612.1  LOC105446336  3'UTR  17902099\n",
       "42  NW_022145615.1     LOC589754  3'UTR  25795628\n",
       "73  NW_022145612.1     LOC591109  3'UTR  30045837\n",
       "91  NW_022145612.1     LOC581215  3'UTR  51908700"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df[new_df[\"hol\"]==\"3'UTR\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#proteins of utrs: uncharacterised, crumbs cell polarity complex component like - in melissas list, \"The cleavage to the 16 cell \"(Moorhouse 2015)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gramd4 death-inducing protein, in supp file of Genome-wide signals of positive selection in strongylocentrotid sea urchins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sp-Brdg1, heparan sulfate 2-O-sulfotransferase 1, atrial natriuretic peptide receptor 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# alpha-N-acetylgalactosaminide alpha-2,6-sialyltransferase 2-like,  dedicator of cytokinesis protein 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#same as 3rd!!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>5949644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>NW_022145609.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>11963040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>22428600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>15580113</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr gene            hol       pos\n",
       "43  NW_022145603.1   NA  genes_overlap   5949644\n",
       "55  NW_022145609.1   NA  genes_overlap  11963040\n",
       "61  NW_022145597.1   NA  genes_overlap  22428600\n",
       "72  NW_022145614.1   NA  genes_overlap  15580113"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df[new_df[\"hol\"]==\"genes_overlap\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#receptor for egg jelly 2 protein (rej2) ! interesting, lysosomal alpha-glucosidase pseudogene, "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#exon of serine/threonine-protein kinase PAK 3 / intron of prostatic spermine-binding protein-like"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#kinesin-like protein KRP180"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these last 4 I didn't check in melissa's list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>alternative UTR</td>\n",
       "      <td>8409205</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr gene              hol      pos\n",
       "53  NW_022145614.1   NA  alternative UTR  8409205"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df[new_df[\"hol\"]==\"alternative UTR\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#RING finger protein 151"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['not_annot', 'exon', \"3'UTR\", 'intron', 'genes_overlap',\n",
       "       'alternative UTR'], dtype=object)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.hol.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, outside of intron"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "are intronic ones located in:\n",
    "    1 ATAC\n",
    "    2 lvar\n",
    "    3 chip \n",
    "    4 lnc\n",
    "    5 promoter\n",
    "    6 TFBS\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])\n",
    "promoters = promoters[promoters['type']==\"gene\"]\n",
    "\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"+\"), \"start\"] -= 1000\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"-\"), \"stop\"] += 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = pd.read_csv(\"atac_mapped_translatedto5.0.csv\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "chip = pd.read_csv('AlxChip_mapped_translatedto5.0.txt', sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lvar = pd.read_csv('lvar_spur_48_50_mapped_translatedto50.csv', header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lnc = pd.read_csv(\"lncRNA_mapped_translatedto5.0.csv\", header=1, names=[\"chr\",\"start\",\"stop\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"not_annot\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "7\n",
      "10\n",
      "11\n",
      "12\n",
      "14\n",
      "15\n",
      "in promoters\n",
      "16\n",
      "17\n",
      "18\n",
      "21\n",
      "in promoters\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "28\n",
      "29\n",
      "31\n",
      "33\n",
      "in lnc\n",
      "37\n",
      "39\n",
      "44\n",
      "49\n",
      "in promoters\n",
      "54\n",
      "56\n",
      "57\n",
      "58\n",
      "69\n",
      "70\n",
      "71\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "82\n",
      "85\n",
      "86\n",
      "87\n",
      "88\n",
      "90\n"
     ]
    }
   ],
   "source": [
    "enh_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in promoters\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"promoter\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>19923027.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>51906905.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>lnc</td>\n",
       "      <td>30428006.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>promoter</td>\n",
       "      <td>16886774.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              chr       hol         pos\n",
       "0  NW_022145598.1  promoter  19923027.0\n",
       "1  NW_022145612.1  promoter  51906905.0\n",
       "2  NW_022145603.1       lnc  30428006.0\n",
       "3  NW_022145600.1  promoter  16886774.0"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#promoters: LOC115923223, LOC581330, LOC589501, nothing very interesting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, inside of intron"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"intron\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in lnc\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "int_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>lnc</td>\n",
       "      <td>15709285.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              chr  hol         pos\n",
       "0  NW_022145603.1  lnc  15709285.0\n",
       "1  NW_022145601.1  lnc  22606114.0"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#there is no point to check TFBS if no putative enhancer is recovered"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
