{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#making this notebook to take out regions from possible enhancers\n",
    "#this is NOT for checking Fst outliers in genes or checking if eg Fst is in UTR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these are either overlaping or not important due to low sample size\n",
    "df.drop(df[df['type'] == \"region\"].index, inplace = True)\n",
    "df.drop(df[df['type'] == \"mRNA\"].index, inplace = True)\n",
    "df.drop(df[df['type'] == \"exon\"].index, inplace = True)\n",
    "df.drop(df[df['type'] == \"CDS\"].index, inplace = True)\n",
    "df.drop(df[df['type'] == \"sequence_feature\"].index, inplace = True)\n",
    "df.drop(df[df['type'] == \"tandem_repeat\"].index, inplace = True)\n",
    "df.drop(df[df['type'] == \"promoter\"].index, inplace = True)\n",
    "#so we are left with eg gene, lncRNA, rRNA snRNA, etc... we want to take out a region that overlaps with any of these as then they are not cis-regulatory regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gene', 'snRNA', 'pseudogene', 'cDNA_match', 'tRNA', 'lnc_RNA',\n",
       "       'transcript', 'snoRNA', 'rRNA', 'primary_transcript', 'miRNA'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#to account for promoters, since we are searching for non promoter/intron enhancers\n",
    "#if + strand -> substract 1000 from start\n",
    "#else add 1000 to stop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[(df[\"type\"]==\"gene\") & (df[\"s2\"]==\"+\"), \"start\"] -= 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[(df[\"type\"]==\"gene\") & (df[\"s2\"]==\"-\"), \"stop\"] += 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=df.drop(columns=['source','type','s1', 's2', 's3','info'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df[\"type\"] = \"annot\"\n",
    "df=df.sort_values(by=[\"chr\",'start'])\n",
    "all_chrs=df.chr.unique()\n",
    "#df[\"length\"] = df[\"stop\"] - df[\"start\"]\n",
    "\n",
    "#CHECK IF THERE ARE NEGATIVE NUMBERS\n",
    "#min(df[\"length\"]) <= 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#check if there is overlap between regions of the same set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_segments(src):\n",
    "    if len(src) <= 1:\n",
    "        return src\n",
    "    \n",
    "    a = src[0]\n",
    "    dest = []\n",
    "    for b in src[1:]:\n",
    "        # print(a,b, check_overlap(a,b))\n",
    "        if check_overlap(a,b):\n",
    "            a = (min(a[0],b[0]),max(a[1],b[1]))\n",
    "        else:\n",
    "            dest.append(a)\n",
    "            a = b\n",
    "    dest.append(a)\n",
    "    return dest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#returns TRUE if there is overlap\n",
    "def check_overlap(a, b):\n",
    "    return max(0, min(a[1], b[1]) - max(a[0], b[0])) > 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_segs(df, chromosome):\n",
    "    return [(start,stop) for _,(_,start,stop) in df[df['chr']==chromosome].iterrows()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "271"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(all_chrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = pd.DataFrame()\n",
    "i=1\n",
    "for ch in all_chrs:\n",
    "    #print(i)\n",
    "    src = get_segs(df,ch)\n",
    "    tmp = clean_segments(src)\n",
    "    for t in tmp:\n",
    "        results = results.append({'chr' : ch, 'start' : t[0], 'stop' : t[1]}, ignore_index = True) \n",
    "    i=i+1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "271"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results.chr.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "results[\"start\"]= results[\"start\"].astype(int)\n",
    "results[\"stop\"]= results[\"stop\"].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "results.to_csv(\"nonoverlapping_nointronpromoter_gffannotations.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
