{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter\n",
    "from matplotlib import pyplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop(df[df['type'] == \"region\"].index, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gene', 'mRNA', 'exon', 'CDS', 'snRNA', 'pseudogene', 'cDNA_match',\n",
       "       'tRNA', 'lnc_RNA', 'transcript', 'snoRNA', 'rRNA',\n",
       "       'primary_transcript', 'miRNA', 'sequence_feature', 'tandem_repeat',\n",
       "       'promoter'], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci = pd.read_csv('top_goodhe_01', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1_809765</td>\n",
       "      <td>0.2058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1_22684111</td>\n",
       "      <td>0.1950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1_36849648</td>\n",
       "      <td>0.1883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1_13121687</td>\n",
       "      <td>0.1882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1_20882520</td>\n",
       "      <td>0.1873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1_27269202</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1_25996126</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1_12214610</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1_3170334</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1_16380245</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            0       1\n",
       "0       NW_022145599.1_809765  0.2058\n",
       "1     NW_022145613.1_22684111  0.1950\n",
       "2     NW_022145600.1_36849648  0.1883\n",
       "3     NW_022145612.1_13121687  0.1882\n",
       "4     NW_022145595.1_20882520  0.1873\n",
       "...                       ...     ...\n",
       "6842  NW_022145597.1_27269202  0.1000\n",
       "6843  NW_022145597.1_25996126  0.1000\n",
       "6844  NW_022145596.1_12214610  0.1000\n",
       "6845   NW_022145595.1_3170334  0.1000\n",
       "6846  NW_022145595.1_16380245  0.1000\n",
       "\n",
       "[6847 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loci"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#---------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR1\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-2c392f1a2651>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"_\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mch\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mpos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"_\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m     \u001b[0mtemp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'chr'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0mch\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m&\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'start'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m<=\u001b[0m\u001b[0mpos\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m&\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'stop'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m>=\u001b[0m\u001b[0mpos\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtemp\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m         \u001b[0;31m#print(\"not in annotation\")\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/ops/common.py\u001b[0m in \u001b[0;36mnew_method\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     63\u001b[0m         \u001b[0mother\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mitem_from_zerodim\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     64\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 65\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     66\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     67\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mnew_method\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/arraylike.py\u001b[0m in \u001b[0;36m__eq__\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     27\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0munpack_zerodim_and_defer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"__eq__\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__eq__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 29\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cmp_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moperator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0meq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     30\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0munpack_zerodim_and_defer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"__ne__\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_cmp_method\u001b[0;34m(self, other, op)\u001b[0m\n\u001b[1;32m   4937\u001b[0m         \u001b[0mrvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mextract_numpy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4938\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4939\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcomparison_op\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4940\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4941\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_construct_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres_values\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mres_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomparison_op\u001b[0;34m(left, right, op)\u001b[0m\n\u001b[1;32m    241\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    242\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mis_object_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 243\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcomp_method_OBJECT_ARRAY\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    244\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    245\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomp_method_OBJECT_ARRAY\u001b[0;34m(op, x, y)\u001b[0m\n\u001b[1;32m     53\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvec_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscalar_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "new_df=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    temp=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    if len(temp) == 1:\n",
    "        print(\"only one\")\n",
    "        print(temp['type'])\n",
    "    if len(temp) > 1:\n",
    "        #print(\"more than 1\")\n",
    "        #checking for gene overlap\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    ids.append(ID)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        if uni > 1:\n",
    "            #print(\"genes overlap\")\n",
    "            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"genes_overlap\", \"gene\": \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "        \n",
    "        if uni == 1:\n",
    "            if temp['type'].str.contains('gene').any():\n",
    "                if temp['type'].str.contains('pseudogene').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                else:\n",
    "                    if temp['type'].str.contains('exon').any(): \n",
    "                        if temp['type'].str.contains('CDS').any():\n",
    "                            #print(\"in exon\")\n",
    "                            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"exon\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                        else:\n",
    "                            direction=temp['s2'].iloc[0]\n",
    "                            e_start=temp[temp[\"type\"]==\"exon\"].iloc[0].start\n",
    "                            e_stop=temp[temp[\"type\"]==\"exon\"].iloc[0].stop\n",
    "                            g_start=temp[temp[\"type\"]==\"gene\"].iloc[0].start\n",
    "                            g_stop=temp[temp[\"type\"]==\"gene\"].iloc[0].stop\n",
    "                            if direction == \"+\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('5prime1')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"3prime1\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                else:\n",
    "                                    print(\"ERROR1\") #alternative UTR?\n",
    "                                    #print(temp)\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            elif direction == \"-\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('3prime2')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"5prime2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True)\n",
    "                                else:\n",
    "                                    print(\"ERROR2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            else:\n",
    "                                print(\"ERROR3\")\n",
    "                                #print(temp)\n",
    "\n",
    "                    else:\n",
    "                        #print(\"in intron\")\n",
    "                        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"intron\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "\n",
    "            else:\n",
    "                print(\"not gene\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.pos = new_df.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.to_csv(\"annotated_top_goodhe_01.csv\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#------------------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.read_csv('annotated_top_goodhe_01.csv', sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>809765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>22684111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>36849648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>13121687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>20882520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>27269202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>25996126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alternative UTR</td>\n",
       "      <td>12214610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>3170334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>LOC582877</td>\n",
       "      <td>intron</td>\n",
       "      <td>16380245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr       gene              hol       pos\n",
       "0     NW_022145599.1        NaN        not_annot    809765\n",
       "1     NW_022145613.1        NaN        not_annot  22684111\n",
       "2     NW_022145600.1        NaN        not_annot  36849648\n",
       "3     NW_022145612.1        NaN        not_annot  13121687\n",
       "4     NW_022145595.1        NaN        not_annot  20882520\n",
       "...              ...        ...              ...       ...\n",
       "6842  NW_022145597.1        NaN        not_annot  27269202\n",
       "6843  NW_022145597.1        NaN        not_annot  25996126\n",
       "6844  NW_022145596.1        NaN  alternative UTR  12214610\n",
       "6845  NW_022145595.1        NaN        not_annot   3170334\n",
       "6846  NW_022145595.1  LOC582877           intron  16380245\n",
       "\n",
       "[6847 rows x 4 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_df[new_df[\"hol\"]==\"pseudogene\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['not_annot', 'exon', \"3'UTR\", 'intron', 'genes_overlap',\n",
       "       'alternative UTR', \"5'UTR\", 'pseudogene'], dtype=object)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.hol.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, outside of intron"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "are intronic ones located in:\n",
    "    1 ATAC\n",
    "    2 lvar\n",
    "    3 chip \n",
    "    4 lnc\n",
    "    5 promoter\n",
    "    6 TFBS\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])\n",
    "promoters = promoters[promoters['type']==\"gene\"]\n",
    "\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"+\"), \"start\"] -= 1000\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"-\"), \"stop\"] += 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = pd.read_csv(\"atac_mapped_translatedto5.0.csv\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "chip = pd.read_csv('AlxChip_mapped_translatedto5.0.txt', sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lvar = pd.read_csv('lvar_spur_48_50_mapped_translatedto50.csv', header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lnc = pd.read_csv(\"lncRNA_mapped_translatedto5.0.csv\", header=1, names=[\"chr\",\"start\",\"stop\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"not_annot\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in chip\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lvar\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in chip\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in chip\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in chip\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lvar\n",
      "in chip\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "enh_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in promoters\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"promoter\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['promoter', 'lnc', 'lvar', 'chip'], dtype=object)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en = enh_check[(enh_check[\"hol\"]==\"lvar\") | (enh_check[\"hol\"]==\"chip\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en=pot_en.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pot_en)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reminder: chosen atac supp (or lvar or chip) too strict?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, inside of intron"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"intron\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in atac\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in atac\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "int_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    gene=row.gene\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\",'gene':gene },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\",'gene':gene },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\",'gene':gene },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\",'gene':gene },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['lnc', 'chip', 'atac', 'lvar'], dtype=object)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>LOC582776</td>\n",
       "      <td>lvar</td>\n",
       "      <td>8855042.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>LOC577823</td>\n",
       "      <td>lvar</td>\n",
       "      <td>18662627.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580332</td>\n",
       "      <td>lvar</td>\n",
       "      <td>47720769.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC100894087</td>\n",
       "      <td>lvar</td>\n",
       "      <td>35316800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>LOC115921692</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16140373.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC588549</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16695182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol         pos\n",
       "83   NW_022145604.1     LOC582776  lvar   8855042.0\n",
       "96   NW_022145600.1     LOC577823  lvar  18662627.0\n",
       "97   NW_022145612.1     LOC580332  lvar  47720769.0\n",
       "100  NW_022145610.1  LOC100894087  lvar  35316800.0\n",
       "114  NW_022145595.1  LOC115921692  lvar  16140373.0\n",
       "118  NW_022145605.1     LOC588549  lvar  16695182.0\n",
       "142  NW_022145612.1     LOC580713  lvar  14929793.0"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check[int_check[\"hol\"]==\"lvar\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in = int_check[(int_check[\"hol\"]==\"lvar\") | (int_check[\"hol\"]==\"chip\") | (int_check[\"hol\"]==\"atac\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in=pot_in.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>LOC752651</td>\n",
       "      <td>lnc</td>\n",
       "      <td>15709285.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC100894079</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC579367</td>\n",
       "      <td>lnc</td>\n",
       "      <td>41191507.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC105444100</td>\n",
       "      <td>lnc</td>\n",
       "      <td>20905720.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC105444622</td>\n",
       "      <td>lnc</td>\n",
       "      <td>1523941.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC105441528</td>\n",
       "      <td>lnc</td>\n",
       "      <td>9915090.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC105444132</td>\n",
       "      <td>lnc</td>\n",
       "      <td>17378166.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>NW_022145576.1</td>\n",
       "      <td>LOC100891941</td>\n",
       "      <td>lnc</td>\n",
       "      <td>2132339.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC100893229</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22327453.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol         pos\n",
       "0    NW_022145603.1     LOC752651   lnc  15709285.0\n",
       "1    NW_022145601.1  LOC100894079   lnc  22606114.0\n",
       "2    NW_022145612.1     LOC579367   lnc  41191507.0\n",
       "3    NW_022145614.1  LOC105444100   lnc  20905720.0\n",
       "4    NW_022145610.1  LOC105444622   lnc   1523941.0\n",
       "..              ...           ...   ...         ...\n",
       "139  NW_022145594.1  LOC105441528   lnc   9915090.0\n",
       "140  NW_022145598.1  LOC105444132   lnc  17378166.0\n",
       "141  NW_022145576.1  LOC100891941   lnc   2132339.0\n",
       "142  NW_022145612.1     LOC580713  lvar  14929793.0\n",
       "143  NW_022145605.1  LOC100893229   lnc  22327453.0\n",
       "\n",
       "[144 rows x 4 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## let's check for my old list (what i wanted to get probes for)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd = pd.read_csv('my_old_list.csv', header=None, names=[\"chr\",\"start\",\"stop\",\"hol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd.drop([0], inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd=erd.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd[\"start\"]=erd.start.astype(float)\n",
    "erd[\"stop\"]=erd.stop.astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Otx'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd[\"hol\"].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd_check=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    fst = row[1]\n",
    "    temp= erd[(erd['chr']==ch) & (erd['start']<=pos) & (erd['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        erd_check = erd_check.append({'chr' : ch, 'pos' : pos, 'hol' : temp[\"hol\"].iloc[0], \"fst\" : fst},ignore_index = True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>ATPsythase</td>\n",
       "      <td>31275084.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1058</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14678986.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1059</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14662026.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1083</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14694778.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1216</td>\n",
       "      <td>B-catenin</td>\n",
       "      <td>19685318.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1602</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30320957.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1013</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30331129.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1127</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30330987.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>0.1049</td>\n",
       "      <td>CD109</td>\n",
       "      <td>3635077.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20094418.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20094442.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20080205.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20071223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara14la</td>\n",
       "      <td>27358332.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1024</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5712045.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5721058.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1018</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5709506.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5727140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1084</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5722898.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara8LB</td>\n",
       "      <td>5781742.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1101</td>\n",
       "      <td>Ets1</td>\n",
       "      <td>41420756.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1100</td>\n",
       "      <td>NCAM</td>\n",
       "      <td>19199989.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1561</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>627088.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1149</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>669637.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1042</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>670140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>639916.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>0.1056</td>\n",
       "      <td>Wnt8</td>\n",
       "      <td>3240002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1292</td>\n",
       "      <td>calcium-sensing</td>\n",
       "      <td>12985454.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1184</td>\n",
       "      <td>cholinesterase</td>\n",
       "      <td>14505911.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>collagen</td>\n",
       "      <td>4993850.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>0.1057</td>\n",
       "      <td>collagen alpha</td>\n",
       "      <td>17756514.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>0.1268</td>\n",
       "      <td>collagen alpha</td>\n",
       "      <td>17732555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1509</td>\n",
       "      <td>collagen-type XV/XVIII</td>\n",
       "      <td>4223297.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1104</td>\n",
       "      <td>ferritin</td>\n",
       "      <td>5925928.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1086</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20737281.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1126</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20782055.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20750669.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1143</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20766314.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1278</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20741452.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>golgi precursor</td>\n",
       "      <td>14538562.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>nell1</td>\n",
       "      <td>20764925.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1021</td>\n",
       "      <td>p19</td>\n",
       "      <td>30282433.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1154</td>\n",
       "      <td>rendezvin</td>\n",
       "      <td>13356234.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>sm30</td>\n",
       "      <td>160017.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1177</td>\n",
       "      <td>sm30</td>\n",
       "      <td>160379.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920879.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1093</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920847.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1017</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920844.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145606.1</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>thrombospondin</td>\n",
       "      <td>21959397.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr     fst                     hol         pos\n",
       "12  NW_022145615.1  0.1179              ATPsythase  31275084.0\n",
       "31  NW_022145603.1  0.1058                    Alx1  14678986.0\n",
       "30  NW_022145603.1  0.1059                    Alx1  14662026.0\n",
       "26  NW_022145603.1  0.1083                    Alx1  14694778.0\n",
       "8   NW_022145613.1  0.1216               B-catenin  19685318.0\n",
       "0   NW_022145596.1  0.1602                   CA10L  30320957.0\n",
       "43  NW_022145596.1  0.1013                   CA10L  30331129.0\n",
       "18  NW_022145596.1  0.1127                   CA10L  30330987.0\n",
       "34  NW_022145599.1  0.1049                   CD109   3635077.0\n",
       "46  NW_022145597.1  0.1006              COLP3alpha  20094418.0\n",
       "40  NW_022145597.1  0.1020              COLP3alpha  20094442.0\n",
       "47  NW_022145597.1  0.1006              COLP3alpha  20080205.0\n",
       "48  NW_022145597.1  0.1001              COLP3alpha  20071223.0\n",
       "45  NW_022145613.1  0.1010                Cara14la  27358332.0\n",
       "38  NW_022145605.1  0.1024                 Cara7la   5712045.0\n",
       "11  NW_022145605.1  0.1181                 Cara7la   5721058.0\n",
       "41  NW_022145605.1  0.1018                 Cara7la   5709506.0\n",
       "27  NW_022145605.1  0.1070                 Cara7la   5727140.0\n",
       "25  NW_022145605.1  0.1084                 Cara7la   5722898.0\n",
       "44  NW_022145614.1  0.1010                 Cara8LB   5781742.0\n",
       "21  NW_022145594.1  0.1101                    Ets1  41420756.0\n",
       "22  NW_022145594.1  0.1100                    NCAM  19199989.0\n",
       "1   NW_022145596.1  0.1561                 Vegfr10    627088.0\n",
       "15  NW_022145596.1  0.1149                 Vegfr10    669637.0\n",
       "35  NW_022145596.1  0.1042                 Vegfr10    670140.0\n",
       "7   NW_022145596.1  0.1250                 Vegfr10    639916.0\n",
       "33  NW_022145595.1  0.1056                    Wnt8   3240002.0\n",
       "4   NW_022145612.1  0.1292         calcium-sensing  12985454.0\n",
       "9   NW_022145597.1  0.1184          cholinesterase  14505911.0\n",
       "36  NW_022145602.1  0.1031                collagen   4993850.0\n",
       "32  NW_022145610.1  0.1057          collagen alpha  17756514.0\n",
       "6   NW_022145610.1  0.1268          collagen alpha  17732555.0\n",
       "2   NW_022145613.1  0.1509  collagen-type XV/XVIII   4223297.0\n",
       "20  NW_022145603.1  0.1104                ferritin   5925928.0\n",
       "24  NW_022145612.1  0.1086             fibrillin-2  20737281.0\n",
       "19  NW_022145612.1  0.1126             fibrillin-2  20782055.0\n",
       "17  NW_022145612.1  0.1139             fibrillin-2  20750669.0\n",
       "16  NW_022145612.1  0.1143             fibrillin-2  20766314.0\n",
       "5   NW_022145612.1  0.1278             fibrillin-2  20741452.0\n",
       "10  NW_022145614.1  0.1181         golgi precursor  14538562.0\n",
       "29  NW_022145602.1  0.1060                   nell1  20764925.0\n",
       "39  NW_022145596.1  0.1021                     p19  30282433.0\n",
       "14  NW_022145603.1  0.1154               rendezvin  13356234.0\n",
       "28  NW_022145602.1  0.1070                    sm30    160017.0\n",
       "13  NW_022145602.1  0.1177                    sm30    160379.0\n",
       "37  NW_022145594.1  0.1031                    sst3  28920879.0\n",
       "23  NW_022145594.1  0.1093                    sst3  28920847.0\n",
       "42  NW_022145594.1  0.1017                    sst3  28920844.0\n",
       "3   NW_022145606.1  0.1319          thrombospondin  21959397.0"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd_check.sort_values(by=[\"hol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf = new_df.merge(erd_check, on=[\"chr\",\"pos\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "49"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf=cdf.sort_values(by=[\"hol_y\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#GO enrichment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Alx1:intron, exon, intron\n",
    "#COLP3alpha: intron all\n",
    "#cara7la: check geneious prime, but i think it is the lnc (the \"introns\" at least)!\n",
    "#Ets1: intron\n",
    "#wnt8: promoter!\n",
    "#ferritin: promoter! > probs p19 and rendezvin promoter region as well"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#checked alx1 exon in vcf\n",
    "BOD: hetero 13 (0/1), 4 homo (1/1) = 17\n",
    "CAP: hetero 4, 1 homo = 5\n",
    "FOG: hetero 6, 2 homo = 8\n",
    "KIB: 7, 6 = 13\n",
    "LOM: 13, 3 = 16\n",
    "SAN: 11, 0 = 11\n",
    "TER: 6, 0 = 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "alx=[8,5,13,17,6,16,11]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<matplotlib.lines.Line2D at 0x7f91c5a72040>]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXAAAAD4CAYAAAD1jb0+AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8vihELAAAACXBIWXMAAAsTAAALEwEAmpwYAAAvE0lEQVR4nO3deXxU9b3/8dd3sidkIckESAhLEiYsQVZZJbEqiq0K3lYsVdtb29K9t7Wr12pva+3t7WJ/ta21tlq1buBubbUEFxYFJEAgMAES1iSEZJKQfc98f39kQhFZksnMnDlnPs/How9gmMz5pMDbme/5nvdRWmuEEEKYj83oAYQQQnhHAlwIIUxKAlwIIUxKAlwIIUxKAlwIIUwqPJAHS01N1RMmTAjkIYUQwvR27NhRp7W2n/14QAN8woQJFBUVBfKQQghhekqpY+d6XJZQhBDCpCTAhRDCpCTAhRDCpCTAhRDCpCTAhRDCpCTAhRDCpCTAhRDCpCTARchztXSxZvtxpFpZmE1AL+QRIth097r5whNFFFc0MnVMItPHJho9khCDJu/ARUj72T9LKa5oBGBjmcvYYYQYIglwEbJe3X2Cx947yu2LJ5KXkcCGAxLgwlwkwEVIKq9t4Qcv7GHO+JHc+dHJFDjs7Dx+iubOHqNHE2LQJMBFyGnr6uVLT+4kJiKMP3xqNhFhNgocafS6Ne+V1xs9nhCDJgEuQorWmjtfLOGwq5UHVs1idGI0ALPGJTEiKpwNB2UZRZiHBLgIKX/beoxXd5/gjqUOFueknn48IszG4pwUNh50yXZCYRoS4CJk7Dp+intfc3LF5DS+cnnOh36/wJFGVWMHh1xtBkwnxNBJgIuQ0NDWzVef2smohGjuXzkDm0196Dn5jv535LKMIsxCAlxYXp9b81/P7qKutZsHb5lNUmzkOZ83dmQs2fY4NkqAC5OQABeW98CbZWwqq+N/bpjGJWOTLvjcAkcaWw/X09nTF5jhhBgGCXBhae8cqOWBt8r4j9kZrJqXedHn5ztS6ep1s+1IQwCmE2J4JMCFZVU1dvDNNcXkjornvhXTUerD695nW5CVQlS4TZZRhClIgAtL6urt4ytP7aS3T/PgLbOJiQwb1NdFR4QxPytFTmQKU7hogCulHlVK1Sql9p71+NeVUgeUUvuUUr/w34hCDN19/yhld0Ujv7rpErLsI4b0tfmTUimvbaXyVLufphPCNwbzDvwxYNmZDyilPgIsBy7RWk8DfuX70YTwzivFVTyx5Rifv2wiy/LGDPnrL8+1A7DxYJ2vRxNDIBdUXdxFA1xrvRE4+4zOl4Gfa627PM+p9cNsQgxZWU0LP3ihhEsnjOT710726jWy7SPISIqRdXAD9bk1S3+zkV+vO2D0KEHN2zVwB7BEKbVNKbVBKXXp+Z6olFqtlCpSShW5XPIPQvhPa1cvX3pyB3FRYfzeU1LlDaUU+Y5U3i2vo6fP7eMpxWAUV5yivLaVB985hPNEs9HjBC1vAzwcGAksAL4LrFXnOcWvtX5Yaz1Xaz3Xbrd7eTghLkxrzQ9e2MORujYeWDWLUQnRw3q9Aoedlq5edh1v9M2AYkjWOWsItykSYyL44csluN2ynHIu3gZ4JfCi7vc+4AZSL/I1QvjN4+8d5bU91Xz76lwWZQ//r+KinFTCbEqWUQxS6KxhflYyP/zYFHYeb2RNUYXRIwUlbwP8ZeAKAKWUA4gE5IyPMMSOY6f46T9KuXJyGl8uyPbJayZERzB7XJJsJzTAIVcrh11tLJ0yihtnZbAgK5mfv76futYuo0cLOoPZRvgMsAXIVUpVKqU+BzwKZHm2Fj4LfEbLKWNhgPrWLr729E7GJEVz/8qZ5yyp8laBw05JVZMER4Ctd9YAcNXUUSil+OmK6bR39/Kzf5YaPFnwGcwulFVa6zFa6wit9Vit9SNa626t9a1a6zyt9Wyt9VuBGFaIM/WXVBVT39bNH2+ZQ2JshE9fv8CRBsDmMvlwGUiFzhqmjklg7MhYAHLSRvDF/Gxe3FnFlkNyx6QzyZWYwrR+u/4gm8vr+MkN08jLSPT5609LTyAlLlKWUQKorrWLHcdPsXTqqA88/rUrchiXHMsPXy6hu1d2Bg2QABem9PaBWh54q5xPzBnLzZdevKTKGzabYsmkVDYedMkuiAB5q7QWrflQgEdHhPHj5dM45Grjz5sOGzRd8JEAF6ZTeaqdb60pZvLoeO5dnjeokipvFeTaqW/rxlkte5EDYZ2zhvTEaKalJ3zo9z6Sm8ZHp4/mgTfLOF4vNQcgAS5MZqCkqq9P89CtcwZdUuWtJZP6r12QZRT/6+juY3O56/TJy3O557pphNsU97y6Vy61RwJcmMy9rznZU9nEL2+awYTUOL8fL3VEFHkZCWw4IAHub5vL6+jscX9o+eRMoxOjuePqXN454OKNvScDOF1wkgAXpvHyriqe3Hqc1flZLMsbHbDjFjjs7Dx+iubOnoAdMxQVOk8SHxXO/IkpF3zeZxaOZ+qYBH78dyetXb0Bmi44SYALUzhY08KdL5Ywb0Iy37smN6DHzp9kp9etea9ctrD5S59b82ZpLZdPTiMy/MKxFB5m474b86hp6eQ3hQcDNGFwkgAXQe/fJVXh/P5Tswj3sqTKW7PHj2REVLisg/vRruOnqG/rvuDyyZlmjRvJp+aN46/vHmHfiSY/Txe8JMBFUNNa8/3n93C0ro3frZpF2jBLqrwREWZjcU4KGw+65MSZnxR6yqsKHIMvvPveNZNJjovkrpf2huw2TwlwEdQeffco/yip5rvXTGZh9oXXRv0p32GnqrGDQ642w2awssLSGhZkpZAYM/iraRNjI7jrY1Mormjkme3H/Thd8JIAF0Gr6GgD//vPUpZOHcWXCrIMnSVfthP6zenyqkEun5xpxcwMFmal8H+v78fVEnqdNRLgIijVtXbx1ad3kjEyhl/dNMOvF+sMRmZyLNn2OKmX9YPCM8qrhkopxU9vzKOzxx2SZVcS4CLo9JdU7aKxvYcHb5k9pI/V/pTvsLP1cD2dPX1Gj2Iphc4apqUnkJEU49XXZ9tH8MWCLF7aVcV7h0KreEwCXASd3xQe5N3yeu5dnse0dN+XVHmrwGGnq9fNtiNn3yJWeMvV0sXOc5RXDdVXPzJQdrWXrt7Q+Q+sBLgIKm/tr+H3b5ezcu5YVvqppMpbC7JSiAq3yTKKD721vwat4aopwwvw6IgwfrJ8GoddbTy8IXTKriTARdCoaGjnW2t2M3VMAj9Znmf0OB8SHRHGvInJciLThwqdtWQkxZyzvGqoLs9N42PTx/D7t8s5Vh8au4UkwEVQ6OzpL6lya80fb51NdIR/S6q8VeCwU17bSlVjh9GjmN7p8qopaT47SX33dVOJCLNxzyv7QmLPvgS4CAo/ec1JSVUTv75pBuNT/F9S5a3Lc/u3E8oyyvBtKnN5yqt812szOjGab1/tYMNBF/8ssX7ZlQS4MNyLOyt5ettxvliQxdXTAldS5Y1s+wjSE6OlndAHCp01xEeHMz8r2aeve9uC8UxLT+Anr+2jxeIFZBLgwlD7Tzbz3y+VMH9iMt+9OrAlVd5QSlGQa+fd8jp6+uTWXt7qc2ve2l/LR3LTiPBxt01/2dV0alu6uN/iZVcS4MIwLZ09fPnJncRHR/A7A0qqvFXgsNPS1UtxRaPRo5jWTk95lTcX7wzGzMwkbp0/nsffO8reKuuWXZnjX4ywHK01331uD8cb2vn9qlmkxQe+pMpbi3JSCbMpWUYZhvXOGiLC1OlzCv7wnWtySY6L4q6XSuizaNmVBLgwxCObj/DGvpN8f1ku87OMK6nyRkJ0BLPHJcl2wmEodPaXVyVE++8q28SYCO6+bgq7K5t4+n1rll1JgIuA2360gf99fT/XTBvFF5YYW1LlrQKHnZKqJupaQ69AabjKa1s5XOddedVQ3TAjncU5Kfzijf3UtnT6/XiBdtEAV0o9qpSqVUrtPcfvfUcppZVSqf4ZT1iNq6WLrz61k8yRMfwyCEqqvJXv6a3eXBZa3Ru+cLq8aphXXw6GUop7l+fR1ePmZ/+wXtnVYN6BPwYsO/tBpVQmsBSw5mcT4XO9fW6+8cwumjp6ePCWOX79+OxveemJJMdFyjKKFwqdJ8nLSCDdy/Kqocqyj+BLl2fzcvEJ3i231n9wLxrgWuuNwLnae34DfA+w5tkB4XP3Fx5ky+F6froij6k+uHTaSDabIn9SKpvKXCF7NxhvuFq62FXRyNIpgd3v/5XLsxmfEsvdFiu78moNXCl1A1Cltd49iOeuVkoVKaWKXC55txKq1jtrePCdQ3zy0kxumhtcJVXeynfYqWvtxlndbPQopnG6vGpqWkCPGx0Rxr3L8zhc18afLFR2NeQAV0rFAncB9wzm+Vrrh7XWc7XWc+12/20ZEsHreH07d6wtZlp6Av9zwzSjx/GZJXKXniErdNaQkRTD1DGB/wSW77Bz3SX9ZVdH66xRduXNO/BsYCKwWyl1FBgL7FRKBfc10MIQnT19fOXpHQD88ZY5QVtS5Q17fBR5GQkS4IPU3t3LprI6lk4dZdjJ67uvm0pUmI27X9lribKrIQe41rpEa52mtZ6gtZ4AVAKztdbWb44RQ/bjv+9jb1Uz96+cybiUWKPH8bn8SXZ2HjtFs8U7N3xhU1kdXb3ugGwfPJ9RCf1lV5vK6vhHSbVhc/jKYLYRPgNsAXKVUpVKqc/5fyxhBc/vqOSZ9yv48uXZfrtk2mgFDju9bs175fVGjxL0Cp01JESHM2+ib8urhuq2hROYnpHIT/7uNP1/eAezC2WV1nqM1jpCaz1Wa/3IWb8/QWttrb05YticJ5q566USFmal8O2lDqPH8ZvZ40cyIiqcjWWyjHIhp8urJvu+vGqowmyK+27Mw9Xaxf3rzF12JVdiCp9r7uzhK0/tIDEmggdWmaekyhsRYTYWZaew4YDLEmuq/rLz+Cka2roDcvHOYFwyNonbFozniS1HKak0b9mVdf9lCUNorfnO2t1UnOrgD7fMxh4fZfRIfleQa6eqsYNDLmvsbPCHwgCUVw3Vd67JJWVEFHe9bN6yKwlw4VN/3nSYdc4a7rx2MpdOMHatM1DyJ8ldei5Ea326vCo+iK6+TYiO4O7rprKnsomnth0zehyvSIALn9l2uJ7/e+MA1+aN5nOXTTR6nIDJTI4lyx4n2wnP45CrlSN1bVwdhCeyr79kDJflpPLLNw6YsuxKAlz4RG1LJ197ZhfjkmP5xScuMW1JlbcKHHa2Hq6ns8c6l2n7yrqB8qogDHClFPeuyKOrz81PXzNf2ZUEuBi23j43X396Fy2dPfzx1tlB9TE5UAocdrp63bx/5Fy1QaGt0FnD9IxExiQGprxqqCamxvGVy7N5dfcJNplsN5EEuBi2X607yLYjDdy3YjqTR5u7pMpb8yemEBluk2WUs9S2dFJc0Rg0u0/O50sF2UxIieWeV/aZ6lOUBLgYlkJnDQ9tOMSqeeP4+JyxRo9jmJjIMOZPTJYAP8tbpbVojaFXXw5GdEQY967I40hdGw9tOGT0OIMmAS68dqy+jTvWFpOXkcCPrp9q9DiGK3DYKa9tpaqxw+hRgsZAedWUMfFGj3JRSybZuWFGOg++fYgjJim7kgAXXunudfPlJ3diU8pyJVXeKnDIdsIztXf3srnc2PKqofrhdVOICrdxj0nKriTAhVfWl9bgrG7mvhvzyEy2XkmVN3LSRpCeGC13q/fYeLC/vCoYtw+eT1p8NN9dlsumsjr+vif4y64kwIVX1hZVMDohmmvzxhg9StBQSlGQa+fd8jp6+txGj2O49aX95VWXGlxeNVS3zB/PJWMTufe14C+7kgAXQ1bd1MHGgy4+MWcsYTZzfDQOlPxJdlq6eimuaDR6FEMFU3nVUIXZFPetmE59axe//tcBo8e5IHP9PyuCwgs7KnFruGlu6O46OZ9FOamE2VTIL6PsONZfXhXsu0/OZ/rYRD69cAJPbD3GnspGo8c5LwlwMSRut2ZtUSULspIZnxJn9DhBJzEmgtnjkkJ+O2Gh8yQRYer0iV0zuuNqB/YRUdz10t6gLbuSABdDsu1IA8cb2rn5UmvcmNgf8ifZKalqoq61y+hRDDFQXrUwO9XUV+UOlF2VVDXx5NbgLLuSABdDsraogviocJZNk5OX51PgqUzdXBaa9zkpr23laH27aZdPznTdJWNYMimVX/7rADXNwVd2JQEuBq25s4d/llRzw8x0YiJl3/f55KUnkhwXGbLLKIWlnvKqKWkGTzJ8SinuXZ5Hd5+be19zGj3Oh0iAi0F7tfgEXb1uVs6V5ZMLsdkUSyalsqnMhTtI1079KdjLq4ZqQmocX708h9f2VAfdRVoS4GLQniuqYPLoeC4Zm2j0KEGvwGGnrrUbZ3Wz0aME1EB5lRWWT870pcuzyEqN4+5X9gZV2ZUEuBiU/Seb2V3ZxE1zM01zWbSRlnju0hNqyyhvmqS8aqiiwvvLro7Vt/PgO8FTdiUBLgZl7fZKIsIUN87KMHoUU7DHRzEtPSHkArzQWcPYkTFMHh385VVDtTgnleUz03nonUMcdrUaPQ4gAS4Goau3j5d2VbJ06iiS4yKNHsc0Chx2dh47FfSXY/tKW5f5yquG6q6PTSEqwsbdQVJ2ddEAV0o9qpSqVUrtPeOxXyql9iul9iilXlJKJfl1SmGoN0trOdXeIycvh6jAYafXrXmvvN7oUQJiU1kd3b1uyy2fnCktPprvXZPLu+X1vLr7hNHjDOod+GPAsrMeKwTytNaXAAeBO308lwgia7ZXMCYx+vS6rhic2eNHMiIqnI0mu02XtwqdNSTGRHDpBHOVVw3Vp+aPZ4an7Kqpw9hPVxcNcK31RqDhrMfWaa17Pb/cCkgphkWdaOxgY5kUV3kjIszGouwUNhxwBcXHbX/q7XPz1v4aPpJrN1151VCF2RT33TidhrZufmVw2ZUv/p++HXjdB68jgtALOyrRGm6aI8sn3ijItVPV2MEhlznu8OKtHcdOcaq9h6VTRxs9SkDkZfSXXT257ZihzZPDCnCl1F1AL/DUBZ6zWilVpJQqcrlC46OkVbjdmrU7KliYlcK4FLlpgzfyJ4XGXXoKnTVEhtlO1wiEgm+fLrsqodeg/nevA1wp9RngOuAWfYHPh1rrh7XWc7XWc+320PnDtYKtR+qpaOiQ4qphyEyOJcseZ+nthFprCktrWJidwoiocKPHCZj46Ah+dP009p1o5m8GlV15FeBKqWXA94EbtNbtvh1JBIu12yuIjw5nWV5ofCz2lwKHna2H64PqCj5fKq9t5ZhFyquG6qPTR5PvsPPrdQc52RT4sqvBbCN8BtgC5CqlKpVSnwN+D8QDhUqpYqXUQ36eUwRYU0cPr+89yfKZ6XLD4mHKd9jp6nXz/pGGiz/ZhNY5B8qrQi/A+8uuphlWdnXRzzta61XnePgRP8wigsiru6W4ylcWTEwhMtzGhoMu8k18g4PzKXTWcMnYREYnRhs9iiHGp8Tx9Y/k8OvCg9x0oJbLcwPXwmjt/T7CawPFVdMzpLhquGIiw5g/MdmS6+C1zZ7yqhB8932m1QVZZNnjuOeVfQFdKpMAFx9SWt3MnsomVkpxlc8UOOyU17ZS1dhh9Cg+tb60FoCl00I7wKPCw/jp8jyON7Tzh7fLA3ZcCXDxIWuLKogMs0lxlQ8N3BvSatsJC50nyUyOIXeU9cqrhmpRTio3zsrgoQ2HKK8NTNmVBLj4gP7iqiqWThvFSCmu8pmctBGkJ0Zb6m71bV29vHuonqVTRssnNY///ugUYiLCuPvlwJRdSYCLD1jvrKVRiqt8TilFvsPOu+V19Bh00YevbSpzWb68aqjs8VF8b9lkthyu5+XiKr8fTwJcfMCaogrSE6O5LCfV6FEsp8Bhp6Wr19BLr31p3enyqpFGjxJUPjVvHDMzk7jvH6U0tfu37EoCXJxW1djBJimu8ptFOamE2ZQlllH6y6tquWJyGuEWL68aKptN8dMVeTS0dfOLf+3377H8+urCVE4XV8nyiV8kxkQwKzPJEvWyRcdO0djeI8sn55GXkch/LprI0+8fZ9fxU347jgS4ADzFVUUVLMpOITNZiqv8pcBhZ09lE3WtXUaPMiwD5VVWvDDJV+642sGo+Gjuemmv38quJMAFAFsP11N5Soqr/G2grW9zWZ3Bk3hPa8360hoW5YRWedVQjYgK557rp+KsbubxLf4pu5IAF0D/ycv46HCumSbFVf6Ul55IclykqfeDl4VwedVQXZs3mstz7dy/7gDVTb6/iEsCXNDU3l9ctWJmhhRX+ZnNplgyKZWNZS7cbnPepacwhMurhkopxU9uyCMpNpKjdb4vbpUAF7y6u4puKa4KmAKHnbrWbpzVzUaP4pV1zhpmjE1kVEJollcN1biUWDZ893IWZqf4/LUlwAVriyqZMiaBvIwEo0cJCQM3hzZjuVVNcye7Kxpl+WSI/LXVUgI8xDlPNFNS1cTNc8fK5dABYo+PYlp6gikD/M2B8qoQufdlsJMAD3EDxVXLZ0pxVSAVOOzsPHaKlk7/Xqnna4XOk4xLjsUxaoTRowgkwENaV28fLxdXcbUUVwVcvsNOr1vz3qF6o0cZtNPlVVNHyae1ICEBHsIKnTVSXGWQ2eNGMiIq3FTLKBsP9pdXye6T4CEBHsLWbK8gIymGxVJcFXCR4TYWZaew4YArILWjvlDorCEpVsqrgokEeIiqPNXO5vI6Pi7FVYbJd9ipauzgcF2b0aNcVG+fm7cO1HJFrpRXBRP5kwhRL+yo6i+umjPW6FFC1sBdeszQTijlVcFJAjwEud2a53ZUsDhHiquMlJkcS5Y9zhTr4IXOGiLDpbwq2EiAh6AtnuIqOXlpvAKHnW1H6gN6J/Oh0lpT6KxhcXYKcVJeFVQkwEPQmu0VJEhxVVDId9jp7HHz/pEGo0c5r4M1rRxvaOcqWT4JOhcNcKXUo0qpWqXU3jMeS1ZKFSqlyjw/ymlpk2hq7+GNfSdZMUuKq4LBgokpRIbbgnoZpdB5EpDyqmA0mHfgjwHLznrsB8CbWutJwJueXwsTeEWKq4JKTGQY8ycmB3W9bKGzhhmZSVJeFYQuGuBa643A2Z/vlgOPe37+OLDCt2MJf1lbVMHUMQnkZSQaPYrwKHDYKattparR933Rw1XT3MnuyiauluWToOTtGvgorXU1gOfHNN+NJPxl34km9lY1y113gszAdsJgfBe+vrS/+1u2DwYnv5/EVEqtVkoVKaWKXK7g+wsaSp4rqiQy3MbymelGjyLOkJM2gvTE6KAM8EJnDeNTYpmUJuVVwcjbAK9RSo0B8PxYe74naq0f1lrP1VrPtdtlD6lROnv6eGlXFddMG01SrBRXBROlFPkOO5vL6ujx081vvdHa1ct75fUsnSLlVcHK2wB/FfiM5+efAV7xzTjCXwqdNTR19LByrlx5GYwKHHZaunoprmg0epTTNh500d3nlu2DQWww2wifAbYAuUqpSqXU54CfA0uVUmXAUs+vRRBbW+QprsqW4qpgtCgnlTCbCqrL6gfKq+aOl13Cweqil1VprVed57eu9PEswk8Giqu+ccUkbFJcFZQSYyKYlZnExjIX37km1+hx+sur9tdy5RQprwpm8icTAp7fUQnATbJ8EtQKHHb2VDZR19pl9ChsP3qKpo4e2T4Y5CTALc7t1jxXVMni7FTGjpTiqmBWkNt/kn9zWZ3Bk/y7vGrgBswiOEmAW9x7h+qpauxgpez9Dnp56Ykkx0Uavp1Qa01h6Ukuy0mV8qogJwFucWuKKkiMiZCPwiZgsymWTEplY5kLt9u4u/QcqGmhoqFDuk9MQALcwhrbu/nXvpOsmJkuxVUmUeCwU9fajbO62bAZCvf1X3151RS5wDrYSYBb2CvFJ/qLq2T5xDQG1pyNbCdcX1rDzMwk0qS8KuhJgFvY2qIKpqUnMC1diqvMwh4fxbT0BMMCfKC8SrpPzEEC3KL2VjWx74QUV5lRgcPOzmOnaOnsCfixC539yydyzsQcJMAt6rmiiv7iqhkZRo8ihijfYafXrXnvUH3Aj13orGFCSiw5Ul5lChLgFtTZ08fLxSdYNm00ibERRo8jhmj2uJGMiAoP+DJKa1cvWw7Vc5WUV5mGBLgFrTtdXCXLJ2YUGW5jUXYKGw640Dpw2wkHyqtk/ds8JMAtaO32/uKqRdkpRo8ivJTvsFPV2MHhuraAHbPQWcPI2AjmSHmVaUiAW0xFQzvvHqrjprljpbjKxAbu0hOodsIeT3nVFZNHSXmViciflMX8u7hKlk/MLDM5lix7XMDWwbcfbaCpo0eWT0xGAtxC+tya53dUcllOKhlJMUaPI4Ypf5KdbUfq6ezp8/uxBsqr8h3SF28mEuAW8t6huv7iKnn3bQkFuXY6e9y8f6TBr8fRWlPorOGynFRiI6W8ykwkwC1kzfYKkmIjuHqafAy2ggUTU4gMt/l9GeVATQuVpzpk+cSEJMAtorG9m3X7algxM4OocCmusoKYyDDmT0z2e71s4b4alIIrpbzKdCTALeLlXVV097ll+cRiChx2ympbqWrs8NsxCgfKq+KlvMpsJMAtYm1RJXkZCUxNTzB6FOFDA9sJ/fUu/GRTJ3ukvMq0JMAtYG9VE87qZm6Wd9+Wk5M2gjGJ0X4L8MJSKa8yMwlwC1jrKa66QYqrLEcpRYHDzuayOnr63D5//YHyqmy7lFeZkQS4yXX29PHyriquzZPiKqsqcNhp6eqluKLRp6/b0tnDlkN1LJ0q5VVmJQFucv/ad5Lmzl45eWlhi3JSCbMpny+jbDxYR0+fZunU0T59XRE4wwpwpdS3lFL7lFJ7lVLPKKXkNHaArS2qIDM5hoVZUlxlVYkxEczKTPL5fvBC50mS4yKlvMrEvA5wpVQG8A1grtY6DwgDPumrwcTFVTS08255PTfNyZTiKosrcNgpqWqivrXLJ6/37/KqNMLk745pDXcJJRyIUUqFA7HAieGP9GH1rV1+v5zYjJ7bUYlS8PE5Y40eRfhZvsOO1rC5vM4nr7f9SAPNnb2yfdDkvA5wrXUV8CvgOFANNGmt1539PKXUaqVUkVKqyOXy7iPgT15zctsj23jnQK2341pOn1vzfFEFSybZpbgqBEzPSCQ5LtJn9bLrnDVEhdtYMknKq8xsOEsoI4HlwEQgHYhTSt169vO01g9rredqrefa7XavjvU/108jJ20Eq5/YwXrPTVdD3bvldZxo6mTlXHn3HQpsNsWSSalsLHPhdg/vLj1aa9aXSnmVFQxnCeUq4IjW2qW17gFeBBb5ZqwPGhkXydOfX8CUMfF86ckdvF5S7Y/DmMqaov7iKvkIHDryJ9mpa+3GWd08rNfZf1LKq6xiOAF+HFiglIpV/ZtIrwRKfTPWhyXGRvC3z89nRmYSX3tmF6/u9styuymcauumUIqrQs4ST1f3cHejFDoHyqskwM1uOGvg24DngZ1Aiee1HvbRXOeUEB3B47fPY874kXzz2V284Ln7TKh5uViKq0JRWnw009ITfBLgszKTsMdH+WgyYZRh7ULRWv9Iaz1Za52ntb5Na+2bPU4XMCIqnMc+eykLs1P4zvO7WbP9uL8PGVS01qzZXsH0jEQprgpB+Q47O4+doqWzx6uvr27qoKSqSS7esQhTXokZGxnOI5+5lPxJdr7/Qgl/23LU6JECZm9VM/tPtrDyUnn3HYoKHHZ63Zr3DtV79fUDmwBk/dsaTBngANERYTz86TlcNSWNu1/ZxyObjxg9UkCsLaogKtzGDTPSjR5FGGD2uJGMiAr3ehmlsLSWialxZNvjfDyZMIJpAxwgKjyMB2+Zw7V5o7n3NScPbThk9Eh+1dnTx8vFnuKqGCmuCkWR4TYWZqew8aALrYe2nVDKq6zH1AEO/X+hf7dqFtfPSOfnr+/ngTfLjB7Jb/617yQtnb2yfBLiChx2Kk91cLiubUhft+Ggy1NeJcsnVmGJXfzhYTb+380zibAp7i88SE+fmzuWOiz3LmPN9v7iqgUTpbgqlA3cpWfDAdeQerwLnTWkxEUye5yUV1mF6d+BDwizKX550wxunpvJ794q5+ev7x/yR8xgVtHQznuH6lkpxVUhLzM5lqzUODaWDX4dvKfPzdtSXmU5lngHPiDMpvjf/5hORLjiTxsP093n5p7rplrinfhzRRVSXCVOy3fYeXb7cTp7+oiOuPjFXO9LeZUlWeYd+ACbTXHv8jxuXzyRv757lLtf2Tvs7gij9bk1z+2oJH+SnXQprhJAQa6dzh73oFs6Cz3lVZdJeZWlWC7Aof8+gndfN4UvFWTz5Nbj3PliCX0mDvHN5XVUN3XKlZfitAUTU4gMtw3qLj1aawqdNSyZJOVVVmPJAIf+EP/+sly+cUUOa4oq+O5zu+n1w01hA2Ht9gpGxkZw1dQ0o0cRQSImMoz5E5MHtR+8tLqFqkYpr7IiywY49If4HVfn8u2lDl7cVcW31u72y529/amhrZt1zpOsmCXFVeKDChx2ympbOdHYccHnDZRXXTFZAtxqLB3gA75+5STuvHYyf999gq8/vYvuXvOE+Mu7qujp09wse7/FWfI92wkvtoxSWHqS2eNGSnmVBYVEgAN8sSCbe66byhv7TvKVp3bQ1dtn9EgXpbVmbVEFM8YmMnm0FFeJD5qUNoIxidEXXEapbupgb1WzLJ9YVMgEOMDtl03k3hV5rC+t5QtP7KCzJ7hDvKSqif0nW7hJTl6Kc1BKUeCws7m87rzndwbKq66S7m9LCqkAB7htwXj+7+PT2VTm4vbHttPe3Wv0SOd1urhqphRXiXPLd9hp6eyluKLxnL+/zllDVmocOWmDv2JTmEfIBTjAzZeO49c3zWDr4Xr+89HttHYFX4h39vTxSvEJPjp9DAnRUlwlzm1xTiphNnXOZZTmzh62Hq6X5RMLC8kAB/iP2WP57SdnseP4KT79yDaavSzI95c39nqKq2T5RFxAYkwEszKTzhngGw5IeZXVhWyAA1w/I50/fGoWeyqbuO0v22hqD54QX7O9gnHJscyfmGz0KCLI5TvslFQ1Ud/6wRtiDZRXzZLyKssK6QAHWJY3hodunUNpdQur/ryVhrZuo0fieH07Ww7Xs3LuWCmuEhdV4LCjdf8VuwN6+ty8faCWK6dIeZWVhXyAA1w1dRQPf3oOh1ytrHp4K3Wtfr+15wU9t6MCmxRXiUGanpFIclwkGw78exnl/SMNtHT2yu4Ti5MA97g8N41H//NSjjW08cmHt1Lb3GnIHH1uzfM7Ksl32BmTKMVV4uJsNsVlOalsLHOdLm4rdNYQHWFjySS7wdMJf5IAP8PinFQe/+w8qhs7uPnhrVQ3XfgSZX/YVOaS4ioxZAUOO3Wt3Tirm0+XV12WYycmUuoXrEwC/Czzs1J44nPzqGvpYuWftlDR0B7Q468tqiA5LlI++oohWeLor4ndcNCFs7qZqsYOrpbdJ5YnAX4Oc8Yn8+Tn59PU3sMnH97Ksfqh3XvQWw1t3RQ6a7hxVgaR4fJHIwYvLT6aqWMS2HDQ9e/yqinSXml1w0oJpVSSUup5pdR+pVSpUmqhrwYz2ozMJJ7+wgLau3tZ+actHHK1+v2YL3mKq2T5RHijINfOzmOneHX3CeaMG0nqCCmvsrrhvs37LfCG1noyMAMoHf5IwSMvI5FnVi+gz625+U9bKatp8duxtNY8V1TBjMwkckfH++04wroKHHZ63ZrDrjaukuWTkOB1gCulEoB84BEArXW31rrRR3MFjcmjE3h29QKUgk8+vJXS6ma/HGdPZX9x1cq5snVQeGf2uJHEeU5aytWXoWE478CzABfwV6XULqXUX5RScWc/SSm1WilVpJQqcrkGfxftYJKTFs+a1QuICLOx6s9b2VvV5PNjrC2qIDrCxvUzpLhKeCcy3MYVU0YxZUwC2XYprwoFwwnwcGA28Eet9SygDfjB2U/SWj+stZ6rtZ5rt5t3T2qWfQRrv7iQuMhwVv15K7uOn/LZa3d09/Fq8Qk+mifFVWJ4fvHxS1j7xQVGjyECZDgBXglUaq23eX79PP2BblnjUmJZ88UFjIyN5LZH3qfo6ODuCH4xb+yrpqWrl5Vy1x0xTDGRYcTLm4CQ4XWAa61PAhVKqVzPQ1cCTp9MFcTGjoxl7RcXkhYfxacffZ8th+qH/ZprtlcwPkWKq4QQQzPcXShfB55SSu0BZgI/G/ZEJjA6MZpnVy8gIymGzz72PpvL6i7+RedxrL6NrYcbWDk3E6WkdEgIMXjDCnCtdbFnffsSrfUKrbXvFoaDXFpCNM+sXsCElDhuf3w7b++v9ep1niuq7C+umi27T4QQQyOX+w1D6ogonvnCAhyjRrD6b0Ws23dySF8/UFxV4LAzOjHaT1MKIaxKAnyYRsZF8tTnFzA1PZGvPLWTf5ZUD/prN5a5ONncyc1y8lII4QUJcB9IjIngyc/NY0ZmEl9/ZhevFFcN6uvWbq8gJS6SKybLRRdCiKGTAPeR+OgInrh9HnPHj+Sba4p5fkflBZ9f39rF+lIprhJCeE+Sw4fiosJ57LPzWJydynef380z7x8/73NPF1fJ8okQwksS4D4WExnGXz4zlwKHnTtfLOGJLUc/9BytNWuLKpiZmYRjlBRXCSG8IwHuB9ERYfzptjlcNWUU97yyj79sOvyB399d2cTBmlapjRVCDIsEuJ9EhYfx4C2zuTZvND/9RykPvlN++vf+XVw1xsAJhRBmJwHuR5HhNn63ahY3zEjnF28c4Lfry+jo7uPvxSf46PQx0lkhhBiWcKMHsLrwMBu/uXkmEWE2frP+IJvLXbR09XKzLJ8IIYZJAjwAwmyKX37iEiLCFM9ur2BCSizzpLhKCDFMEuABYrMpfnbjdHLSRpA7Ol6Kq4QQwyYBHkA2m+LzS7KMHkMIYRFyElMIIUxKAlwIIUxKAlwIIUxKAlwIIUxKAlwIIUxKAlwIIUxKAlwIIUxKAlwIIUxKaa0DdzClXMAxL788Fajz4ThGku8l+Fjl+wD5XoLVcL6X8Vpr+9kPBjTAh0MpVaS1nmv0HL4g30vwscr3AfK9BCt/fC+yhCKEECYlAS6EECZlpgB/2OgBfEi+l+Bjle8D5HsJVj7/XkyzBi6EEOKDzPQOXAghxBkkwIUQwqRMEeBKqWVKqQNKqXKl1A+MnsdbSqlHlVK1Sqm9Rs8yHEqpTKXU20qpUqXUPqXUfxk9k7eUUtFKqfeVUrs938uPjZ5pOJRSYUqpXUqp14yeZTiUUkeVUiVKqWKlVJHR8wyHUipJKfW8Umq/59/MQp+9drCvgSulwoCDwFKgEtgOrNJaOw0dzAtKqXygFXhCa51n9DzeUkqNAcZorXcqpeKBHcAKk/6ZKCBOa92qlIoANgP/pbXeavBoXlFK3QHMBRK01tcZPY+3lFJHgblaa9NfxKOUehzYpLX+i1IqEojVWjf64rXN8A58HlCutT6ste4GngWWGzyTV7TWG4EGo+cYLq11tdZ6p+fnLUApkGHsVN7R/Vo9v4zw/C+439Wch1JqLPAx4C9GzyL6KaUSgHzgEQCtdbevwhvMEeAZQMUZv67EpGFhRUqpCcAsYJvBo3jNs+xQDNQChVprs34v/w/4HuA2eA5f0MA6pdQOpdRqo4cZhizABfzVs7T1F6VUnK9e3AwBfq7bt5vyHZLVKKVGAC8A39RaNxs9j7e01n1a65nAWGCeUsp0y1tKqeuAWq31DqNn8ZHFWuvZwLXAVz3Lj2YUDswG/qi1ngW0AT47j2eGAK8EMs/49VjghEGzCA/PevELwFNa6xeNnscXPB9t3wGWGTuJVxYDN3jWjp8FrlBKPWnsSN7TWp/w/FgLvET/UqoZVQKVZ3yqe57+QPcJMwT4dmCSUmqi5wTAJ4FXDZ4ppHlO/D0ClGqt7zd6nuFQStmVUkmen8cAVwH7DR3KC1rrO7XWY7XWE+j/N/KW1vpWg8fyilIqznNyHM9yw9WAKXduaa1PAhVKqVzPQ1cCPjvZH+6rF/IXrXWvUuprwL+AMOBRrfU+g8fyilLqGeByIFUpVQn8SGv9iLFTeWUxcBtQ4lk7BvhvrfU/jRvJa2OAxz27nWzAWq21qbfgWcAo4KX+9wmEA09rrd8wdqRh+TrwlOcN6GHgs7564aDfRiiEEOLczLCEIoQQ4hwkwIUQwqQkwIUQwqQkwIUQwqQkwIUQwqQkwIUQwqQkwIUQwqT+P6RSwZ1hYOIQAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "pyplot.plot(alx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#aa 213 (or around)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol_x</th>\n",
       "      <th>pos</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>LOC373382</td>\n",
       "      <td>exon</td>\n",
       "      <td>31275084</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>ATPsythase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>14694778</td>\n",
       "      <td>0.1083</td>\n",
       "      <td>Alx1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>14662026</td>\n",
       "      <td>0.1059</td>\n",
       "      <td>Alx1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>14678986</td>\n",
       "      <td>0.1058</td>\n",
       "      <td>Alx1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>LOC576711</td>\n",
       "      <td>exon</td>\n",
       "      <td>19685318</td>\n",
       "      <td>0.1216</td>\n",
       "      <td>B-catenin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579596</td>\n",
       "      <td>exon</td>\n",
       "      <td>30320957</td>\n",
       "      <td>0.1602</td>\n",
       "      <td>CA10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579596</td>\n",
       "      <td>intron</td>\n",
       "      <td>30330987</td>\n",
       "      <td>0.1127</td>\n",
       "      <td>CA10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579596</td>\n",
       "      <td>intron</td>\n",
       "      <td>30331129</td>\n",
       "      <td>0.1013</td>\n",
       "      <td>CA10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>LOC579544</td>\n",
       "      <td>exon</td>\n",
       "      <td>3635077</td>\n",
       "      <td>0.1049</td>\n",
       "      <td>CD109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>20071223</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>COLP3alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>20094418</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>20080205</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>20094442</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>COLP3alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>LOC591885</td>\n",
       "      <td>intron</td>\n",
       "      <td>27358332</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara14la</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC115926619</td>\n",
       "      <td>intron</td>\n",
       "      <td>5721058</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>Cara7la</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC115926619</td>\n",
       "      <td>intron</td>\n",
       "      <td>5722898</td>\n",
       "      <td>0.1084</td>\n",
       "      <td>Cara7la</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC115926619</td>\n",
       "      <td>intron</td>\n",
       "      <td>5727140</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>Cara7la</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>5712045</td>\n",
       "      <td>0.1024</td>\n",
       "      <td>Cara7la</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>5709506</td>\n",
       "      <td>0.1018</td>\n",
       "      <td>Cara7la</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC590675</td>\n",
       "      <td>intron</td>\n",
       "      <td>5781742</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara8LB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>41420756</td>\n",
       "      <td>0.1101</td>\n",
       "      <td>Ets1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC586418</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>19199989</td>\n",
       "      <td>0.1100</td>\n",
       "      <td>NCAM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579837</td>\n",
       "      <td>intron</td>\n",
       "      <td>669637</td>\n",
       "      <td>0.1149</td>\n",
       "      <td>Vegfr10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579837</td>\n",
       "      <td>intron</td>\n",
       "      <td>627088</td>\n",
       "      <td>0.1561</td>\n",
       "      <td>Vegfr10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579837</td>\n",
       "      <td>intron</td>\n",
       "      <td>639916</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>Vegfr10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>LOC579837</td>\n",
       "      <td>intron</td>\n",
       "      <td>670140</td>\n",
       "      <td>0.1042</td>\n",
       "      <td>Vegfr10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>3240002</td>\n",
       "      <td>0.1056</td>\n",
       "      <td>Wnt8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC575352</td>\n",
       "      <td>intron</td>\n",
       "      <td>12985454</td>\n",
       "      <td>0.1292</td>\n",
       "      <td>calcium-sensing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>LOC762935</td>\n",
       "      <td>intron</td>\n",
       "      <td>14505911</td>\n",
       "      <td>0.1184</td>\n",
       "      <td>cholinesterase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>LOC592088</td>\n",
       "      <td>intron</td>\n",
       "      <td>4993850</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>collagen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC589794</td>\n",
       "      <td>intron</td>\n",
       "      <td>17732555</td>\n",
       "      <td>0.1268</td>\n",
       "      <td>collagen alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC589794</td>\n",
       "      <td>intron</td>\n",
       "      <td>17756514</td>\n",
       "      <td>0.1057</td>\n",
       "      <td>collagen alpha</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>LOC576214</td>\n",
       "      <td>intron</td>\n",
       "      <td>4223297</td>\n",
       "      <td>0.1509</td>\n",
       "      <td>collagen-type XV/XVIII</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>5925928</td>\n",
       "      <td>0.1104</td>\n",
       "      <td>ferritin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105436405</td>\n",
       "      <td>exon</td>\n",
       "      <td>20741452</td>\n",
       "      <td>0.1278</td>\n",
       "      <td>fibrillin-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105436405</td>\n",
       "      <td>intron</td>\n",
       "      <td>20782055</td>\n",
       "      <td>0.1126</td>\n",
       "      <td>fibrillin-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105436405</td>\n",
       "      <td>intron</td>\n",
       "      <td>20766314</td>\n",
       "      <td>0.1143</td>\n",
       "      <td>fibrillin-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105436405</td>\n",
       "      <td>intron</td>\n",
       "      <td>20750669</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>fibrillin-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105436405</td>\n",
       "      <td>intron</td>\n",
       "      <td>20737281</td>\n",
       "      <td>0.1086</td>\n",
       "      <td>fibrillin-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC585057</td>\n",
       "      <td>intron</td>\n",
       "      <td>14538562</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>golgi precursor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>LOC589496</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>20764925</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>nell1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>30282433</td>\n",
       "      <td>0.1021</td>\n",
       "      <td>p19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>13356234</td>\n",
       "      <td>0.1154</td>\n",
       "      <td>rendezvin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>LOC581461</td>\n",
       "      <td>intron</td>\n",
       "      <td>160017</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>sm30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>LOC581461</td>\n",
       "      <td>intron</td>\n",
       "      <td>160379</td>\n",
       "      <td>0.1177</td>\n",
       "      <td>sm30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC100888644</td>\n",
       "      <td>exon</td>\n",
       "      <td>28920847</td>\n",
       "      <td>0.1093</td>\n",
       "      <td>sst3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC100888644</td>\n",
       "      <td>exon</td>\n",
       "      <td>28920844</td>\n",
       "      <td>0.1017</td>\n",
       "      <td>sst3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC100888644</td>\n",
       "      <td>exon</td>\n",
       "      <td>28920879</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>sst3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145606.1</td>\n",
       "      <td>LOC100892350</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>21959397</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>thrombospondin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr          gene          hol_x       pos     fst  \\\n",
       "12  NW_022145615.1     LOC373382           exon  31275084  0.1179   \n",
       "26  NW_022145603.1           NaN  genes_overlap  14694778  0.1083   \n",
       "30  NW_022145603.1           NaN  genes_overlap  14662026  0.1059   \n",
       "31  NW_022145603.1           NaN  genes_overlap  14678986  0.1058   \n",
       "8   NW_022145613.1     LOC576711           exon  19685318  0.1216   \n",
       "0   NW_022145596.1     LOC579596           exon  30320957  0.1602   \n",
       "18  NW_022145596.1     LOC579596         intron  30330987  0.1127   \n",
       "43  NW_022145596.1     LOC579596         intron  30331129  0.1013   \n",
       "34  NW_022145599.1     LOC579544           exon   3635077  0.1049   \n",
       "48  NW_022145597.1           NaN  genes_overlap  20071223  0.1001   \n",
       "46  NW_022145597.1           NaN  genes_overlap  20094418  0.1006   \n",
       "47  NW_022145597.1           NaN  genes_overlap  20080205  0.1006   \n",
       "40  NW_022145597.1           NaN  genes_overlap  20094442  0.1020   \n",
       "45  NW_022145613.1     LOC591885         intron  27358332  0.1010   \n",
       "11  NW_022145605.1  LOC115926619         intron   5721058  0.1181   \n",
       "25  NW_022145605.1  LOC115926619         intron   5722898  0.1084   \n",
       "27  NW_022145605.1  LOC115926619         intron   5727140  0.1070   \n",
       "38  NW_022145605.1           NaN      not_annot   5712045  0.1024   \n",
       "41  NW_022145605.1           NaN      not_annot   5709506  0.1018   \n",
       "44  NW_022145614.1     LOC590675         intron   5781742  0.1010   \n",
       "21  NW_022145594.1           NaN  genes_overlap  41420756  0.1101   \n",
       "22  NW_022145594.1     LOC586418          3'UTR  19199989  0.1100   \n",
       "15  NW_022145596.1     LOC579837         intron    669637  0.1149   \n",
       "1   NW_022145596.1     LOC579837         intron    627088  0.1561   \n",
       "7   NW_022145596.1     LOC579837         intron    639916  0.1250   \n",
       "35  NW_022145596.1     LOC579837         intron    670140  0.1042   \n",
       "33  NW_022145595.1           NaN      not_annot   3240002  0.1056   \n",
       "4   NW_022145612.1     LOC575352         intron  12985454  0.1292   \n",
       "9   NW_022145597.1     LOC762935         intron  14505911  0.1184   \n",
       "36  NW_022145602.1     LOC592088         intron   4993850  0.1031   \n",
       "6   NW_022145610.1     LOC589794         intron  17732555  0.1268   \n",
       "32  NW_022145610.1     LOC589794         intron  17756514  0.1057   \n",
       "2   NW_022145613.1     LOC576214         intron   4223297  0.1509   \n",
       "20  NW_022145603.1           NaN      not_annot   5925928  0.1104   \n",
       "5   NW_022145612.1  LOC105436405           exon  20741452  0.1278   \n",
       "19  NW_022145612.1  LOC105436405         intron  20782055  0.1126   \n",
       "16  NW_022145612.1  LOC105436405         intron  20766314  0.1143   \n",
       "17  NW_022145612.1  LOC105436405         intron  20750669  0.1139   \n",
       "24  NW_022145612.1  LOC105436405         intron  20737281  0.1086   \n",
       "10  NW_022145614.1     LOC585057         intron  14538562  0.1181   \n",
       "29  NW_022145602.1     LOC589496          3'UTR  20764925  0.1060   \n",
       "39  NW_022145596.1           NaN  genes_overlap  30282433  0.1021   \n",
       "14  NW_022145603.1           NaN  genes_overlap  13356234  0.1154   \n",
       "28  NW_022145602.1     LOC581461         intron    160017  0.1070   \n",
       "13  NW_022145602.1     LOC581461         intron    160379  0.1177   \n",
       "23  NW_022145594.1  LOC100888644           exon  28920847  0.1093   \n",
       "42  NW_022145594.1  LOC100888644           exon  28920844  0.1017   \n",
       "37  NW_022145594.1  LOC100888644           exon  28920879  0.1031   \n",
       "3   NW_022145606.1  LOC100892350          3'UTR  21959397  0.1319   \n",
       "\n",
       "                     hol_y  \n",
       "12              ATPsythase  \n",
       "26                    Alx1  \n",
       "30                    Alx1  \n",
       "31                    Alx1  \n",
       "8                B-catenin  \n",
       "0                    CA10L  \n",
       "18                   CA10L  \n",
       "43                   CA10L  \n",
       "34                   CD109  \n",
       "48              COLP3alpha  \n",
       "46              COLP3alpha  \n",
       "47              COLP3alpha  \n",
       "40              COLP3alpha  \n",
       "45                Cara14la  \n",
       "11                 Cara7la  \n",
       "25                 Cara7la  \n",
       "27                 Cara7la  \n",
       "38                 Cara7la  \n",
       "41                 Cara7la  \n",
       "44                 Cara8LB  \n",
       "21                    Ets1  \n",
       "22                    NCAM  \n",
       "15                 Vegfr10  \n",
       "1                  Vegfr10  \n",
       "7                  Vegfr10  \n",
       "35                 Vegfr10  \n",
       "33                    Wnt8  \n",
       "4          calcium-sensing  \n",
       "9           cholinesterase  \n",
       "36                collagen  \n",
       "6           collagen alpha  \n",
       "32          collagen alpha  \n",
       "2   collagen-type XV/XVIII  \n",
       "20                ferritin  \n",
       "5              fibrillin-2  \n",
       "19             fibrillin-2  \n",
       "16             fibrillin-2  \n",
       "17             fibrillin-2  \n",
       "24             fibrillin-2  \n",
       "10         golgi precursor  \n",
       "29                   nell1  \n",
       "39                     p19  \n",
       "14               rendezvin  \n",
       "28                    sm30  \n",
       "13                    sm30  \n",
       "23                    sst3  \n",
       "42                    sst3  \n",
       "37                    sst3  \n",
       "3           thrombospondin  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "int_check[\"pos\"]=int_check.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>LOC752651</td>\n",
       "      <td>lnc</td>\n",
       "      <td>15709285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC100894079</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC579367</td>\n",
       "      <td>lnc</td>\n",
       "      <td>41191507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC105444100</td>\n",
       "      <td>lnc</td>\n",
       "      <td>20905720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC105444622</td>\n",
       "      <td>lnc</td>\n",
       "      <td>1523941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC105441528</td>\n",
       "      <td>lnc</td>\n",
       "      <td>9915090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC105444132</td>\n",
       "      <td>lnc</td>\n",
       "      <td>17378166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>NW_022145576.1</td>\n",
       "      <td>LOC100891941</td>\n",
       "      <td>lnc</td>\n",
       "      <td>2132339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC100893229</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22327453</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol       pos\n",
       "0    NW_022145603.1     LOC752651   lnc  15709285\n",
       "1    NW_022145601.1  LOC100894079   lnc  22606114\n",
       "2    NW_022145612.1     LOC579367   lnc  41191507\n",
       "3    NW_022145614.1  LOC105444100   lnc  20905720\n",
       "4    NW_022145610.1  LOC105444622   lnc   1523941\n",
       "..              ...           ...   ...       ...\n",
       "139  NW_022145594.1  LOC105441528   lnc   9915090\n",
       "140  NW_022145598.1  LOC105444132   lnc  17378166\n",
       "141  NW_022145576.1  LOC100891941   lnc   2132339\n",
       "142  NW_022145612.1     LOC580713  lvar  14929793\n",
       "143  NW_022145605.1  LOC100893229   lnc  22327453\n",
       "\n",
       "[144 rows x 4 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "check = int_check.merge(cdf, on=[\"chr\",\"pos\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene_x</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene_y</th>\n",
       "      <th>hol_x</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, gene_x, hol, pos, gene_y, hol_x, fst, hol_y]\n",
       "Index: []"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
