{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter\n",
    "from matplotlib import pyplot\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop(df[df['type'] == \"region\"].index, inplace = True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gene', 'mRNA', 'exon', 'CDS', 'snRNA', 'pseudogene', 'cDNA_match',\n",
       "       'tRNA', 'lnc_RNA', 'transcript', 'snoRNA', 'rRNA',\n",
       "       'primary_transcript', 'miRNA', 'sequence_feature', 'tandem_repeat',\n",
       "       'promoter'], dtype=object)"
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['info'] = df['info'].str.replace(',',';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['+', '-'], dtype=object)"
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.s2.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci = pd.read_csv('top_goodhe_01', header=None, names=[\"chr\",\"fst\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>fst</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1_809765</td>\n",
       "      <td>0.2058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1_22684111</td>\n",
       "      <td>0.1950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1_36849648</td>\n",
       "      <td>0.1883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1_13121687</td>\n",
       "      <td>0.1882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1_20882520</td>\n",
       "      <td>0.1873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1_27269202</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1_25996126</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1_12214610</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1_3170334</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1_16380245</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          chr     fst\n",
       "0       NW_022145599.1_809765  0.2058\n",
       "1     NW_022145613.1_22684111  0.1950\n",
       "2     NW_022145600.1_36849648  0.1883\n",
       "3     NW_022145612.1_13121687  0.1882\n",
       "4     NW_022145595.1_20882520  0.1873\n",
       "...                       ...     ...\n",
       "6842  NW_022145597.1_27269202  0.1000\n",
       "6843  NW_022145597.1_25996126  0.1000\n",
       "6844  NW_022145596.1_12214610  0.1000\n",
       "6845   NW_022145595.1_3170334  0.1000\n",
       "6846  NW_022145595.1_16380245  0.1000\n",
       "\n",
       "[6847 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# if just looking at also high bayenv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {},
   "outputs": [],
   "source": [
    "hbay = pd.read_csv('high_bayenv.csv', header=None, sep='\\t',names=[\"index\",\"ave\",\"chr\",\"fst\",\"pos\",\"ch_num\",\"idx\",\"log_bf\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [],
   "source": [
    "hbay[\"comb\"]=hbay[\"chr\"] +\"_\"+ hbay[\"pos\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>ave</th>\n",
       "      <th>chr</th>\n",
       "      <th>fst</th>\n",
       "      <th>pos</th>\n",
       "      <th>ch_num</th>\n",
       "      <th>idx</th>\n",
       "      <th>log_bf</th>\n",
       "      <th>comb</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>index</td>\n",
       "      <td>ave</td>\n",
       "      <td>chr</td>\n",
       "      <td>fst</td>\n",
       "      <td>pos</td>\n",
       "      <td>ch_num</td>\n",
       "      <td>idx</td>\n",
       "      <td>log_bf</td>\n",
       "      <td>chr_pos</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6804</td>\n",
       "      <td>9.470158600000001</td>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>757345</td>\n",
       "      <td>1</td>\n",
       "      <td>668</td>\n",
       "      <td>0.9763572523430892</td>\n",
       "      <td>NW_022145612.1_757345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6818</td>\n",
       "      <td>2.86165</td>\n",
       "      <td>NW_022145611.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>36506506</td>\n",
       "      <td>2</td>\n",
       "      <td>1348</td>\n",
       "      <td>0.4566165153902805</td>\n",
       "      <td>NW_022145611.1_36506506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6840</td>\n",
       "      <td>1.2448080000000001</td>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>23380763</td>\n",
       "      <td>7</td>\n",
       "      <td>3359</td>\n",
       "      <td>0.09510237073228703</td>\n",
       "      <td>NW_022145599.1_23380763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6769</td>\n",
       "      <td>1.686525</td>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>22443666</td>\n",
       "      <td>12</td>\n",
       "      <td>4661</td>\n",
       "      <td>0.22699278328482644</td>\n",
       "      <td>NW_022145601.1_22443666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>100</td>\n",
       "      <td>1.213802</td>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1553</td>\n",
       "      <td>26343419</td>\n",
       "      <td>17</td>\n",
       "      <td>5999</td>\n",
       "      <td>0.08414784874865514</td>\n",
       "      <td>NW_022145603.1_26343419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>96</td>\n",
       "      <td>1.3058999999999998</td>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>0.1562</td>\n",
       "      <td>23070922</td>\n",
       "      <td>7</td>\n",
       "      <td>3375</td>\n",
       "      <td>0.1159099218770607</td>\n",
       "      <td>NW_022145599.1_23070922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>32</td>\n",
       "      <td>46.59352199999999</td>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>0.1668</td>\n",
       "      <td>25383710</td>\n",
       "      <td>4</td>\n",
       "      <td>2233</td>\n",
       "      <td>1.668325539968573</td>\n",
       "      <td>NW_022145615.1_25383710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>17</td>\n",
       "      <td>1.4565804000000002</td>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>0.1732</td>\n",
       "      <td>416271</td>\n",
       "      <td>16</td>\n",
       "      <td>5673</td>\n",
       "      <td>0.16333446171524763</td>\n",
       "      <td>NW_022145598.1_416271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>1</td>\n",
       "      <td>1.036662</td>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.195</td>\n",
       "      <td>22684111</td>\n",
       "      <td>3</td>\n",
       "      <td>1852</td>\n",
       "      <td>0.015637179279383465</td>\n",
       "      <td>NW_022145613.1_22684111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>254 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index                 ave             chr     fst       pos  ch_num  \\\n",
       "0    index                 ave             chr     fst       pos  ch_num   \n",
       "1     6804   9.470158600000001  NW_022145612.1     0.1    757345       1   \n",
       "2     6818             2.86165  NW_022145611.1     0.1  36506506       2   \n",
       "3     6840  1.2448080000000001  NW_022145599.1     0.1  23380763       7   \n",
       "4     6769            1.686525  NW_022145601.1  0.1001  22443666      12   \n",
       "..     ...                 ...             ...     ...       ...     ...   \n",
       "249    100            1.213802  NW_022145603.1  0.1553  26343419      17   \n",
       "250     96  1.3058999999999998  NW_022145599.1  0.1562  23070922       7   \n",
       "251     32   46.59352199999999  NW_022145615.1  0.1668  25383710       4   \n",
       "252     17  1.4565804000000002  NW_022145598.1  0.1732    416271      16   \n",
       "253      1            1.036662  NW_022145613.1   0.195  22684111       3   \n",
       "\n",
       "      idx                log_bf                     comb  \n",
       "0     idx                log_bf                  chr_pos  \n",
       "1     668    0.9763572523430892    NW_022145612.1_757345  \n",
       "2    1348    0.4566165153902805  NW_022145611.1_36506506  \n",
       "3    3359   0.09510237073228703  NW_022145599.1_23380763  \n",
       "4    4661   0.22699278328482644  NW_022145601.1_22443666  \n",
       "..    ...                   ...                      ...  \n",
       "249  5999   0.08414784874865514  NW_022145603.1_26343419  \n",
       "250  3375    0.1159099218770607  NW_022145599.1_23070922  \n",
       "251  2233     1.668325539968573  NW_022145615.1_25383710  \n",
       "252  5673   0.16333446171524763    NW_022145598.1_416271  \n",
       "253  1852  0.015637179279383465  NW_022145613.1_22684111  \n",
       "\n",
       "[254 rows x 9 columns]"
      ]
     },
     "execution_count": 299,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hbay"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci=hbay[[\"comb\",\"fst\"]].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci=loci.drop(loci.index[0])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci[\"chr\"]=loci[\"comb\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>comb</th>\n",
       "      <th>fst</th>\n",
       "      <th>chr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145612.1_757345</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NW_022145612.1_757345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145611.1_36506506</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NW_022145611.1_36506506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145599.1_23380763</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NW_022145599.1_23380763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145601.1_22443666</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>NW_022145601.1_22443666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NW_022145615.1_23388979</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>NW_022145615.1_23388979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>NW_022145603.1_26343419</td>\n",
       "      <td>0.1553</td>\n",
       "      <td>NW_022145603.1_26343419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>NW_022145599.1_23070922</td>\n",
       "      <td>0.1562</td>\n",
       "      <td>NW_022145599.1_23070922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>NW_022145615.1_25383710</td>\n",
       "      <td>0.1668</td>\n",
       "      <td>NW_022145615.1_25383710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>NW_022145598.1_416271</td>\n",
       "      <td>0.1732</td>\n",
       "      <td>NW_022145598.1_416271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>NW_022145613.1_22684111</td>\n",
       "      <td>0.195</td>\n",
       "      <td>NW_022145613.1_22684111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>253 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        comb     fst                      chr\n",
       "1      NW_022145612.1_757345     0.1    NW_022145612.1_757345\n",
       "2    NW_022145611.1_36506506     0.1  NW_022145611.1_36506506\n",
       "3    NW_022145599.1_23380763     0.1  NW_022145599.1_23380763\n",
       "4    NW_022145601.1_22443666  0.1001  NW_022145601.1_22443666\n",
       "5    NW_022145615.1_23388979  0.1001  NW_022145615.1_23388979\n",
       "..                       ...     ...                      ...\n",
       "249  NW_022145603.1_26343419  0.1553  NW_022145603.1_26343419\n",
       "250  NW_022145599.1_23070922  0.1562  NW_022145599.1_23070922\n",
       "251  NW_022145615.1_25383710  0.1668  NW_022145615.1_25383710\n",
       "252    NW_022145598.1_416271  0.1732    NW_022145598.1_416271\n",
       "253  NW_022145613.1_22684111   0.195  NW_022145613.1_22684111\n",
       "\n",
       "[253 rows x 3 columns]"
      ]
     },
     "execution_count": 303,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#t=loci.chr.to_numpy()\n",
    "#chrs=[i.split(\"_\")[:2] for i in t ]\n",
    "#chrs =[\"_\".join(i) for i in chrs]\n",
    "#pos=[i.split(\"_\")[2] for i in t ]\n",
    "#fixed_loci = pd.DataFrame(list(zip(chrs, pos)), columns =['chr', 'pos'])\n",
    "#np.savetxt(\"fixed_loci.csv\", t, delimiter=',', fmt=\"%s\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#---------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "76\n",
      "96\n"
     ]
    }
   ],
   "source": [
    "new_df=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    temp=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    if len(temp) == 1:\n",
    "        print(\"only one\")\n",
    "        print(temp['type'])\n",
    "    if len(temp) > 1:\n",
    "        #print(\"more than 1\")\n",
    "        #checking for gene overlap\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    ids.append(ID)\n",
    "                #else:\n",
    "                    #print(temp[\"type\"].iloc[i])\n",
    "                    #if temp[\"type\"].iloc[i] == \"cDNA_match\":\n",
    "                        #print(\"cDNA_match\")\n",
    "                    #else:\n",
    "                        #print(\"what about ids???\")\n",
    "                        #print(t)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        if uni > 1:\n",
    "            #print(\"genes overlap\")\n",
    "            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"genes_overlap\", \"gene\": \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "        \n",
    "        if uni == 1:\n",
    "            if temp['type'].str.contains('gene').any():\n",
    "                if temp['type'].str.contains('pseudogene').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                # snRNA, tRNA, lnc_RNA, snoRNA, rRNA, miRNA also have line with \"gene\"\n",
    "                elif temp['type'].str.contains('snRNA').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"snRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('tRNA').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"tRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('lnc_RNA').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc_RNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('snoRNA').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"snoRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('rRNA').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"rRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('miRNA').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"miRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                else:\n",
    "                    if temp['type'].str.contains('exon').any(): \n",
    "                        if temp['type'].str.contains('CDS').any():\n",
    "                            #print(\"in exon\")\n",
    "                            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"exon\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                        else:\n",
    "                            direction=temp['s2'].iloc[0]\n",
    "                            e_start=temp[temp[\"type\"]==\"exon\"].iloc[0].start\n",
    "                            e_stop=temp[temp[\"type\"]==\"exon\"].iloc[0].stop\n",
    "                            g_start=temp[temp[\"type\"]==\"gene\"].iloc[0].start\n",
    "                            g_stop=temp[temp[\"type\"]==\"gene\"].iloc[0].stop\n",
    "                            if direction == \"+\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('5prime1')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"3prime1\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                else:\n",
    "                                    print(index) #alternative UTR?\n",
    "                                    #print(temp)\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            elif direction == \"-\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('3prime2')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"5prime2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True)\n",
    "                                else:\n",
    "                                    #print(\"ERROR2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            else:\n",
    "                                print(index)\n",
    "                                #print(temp)\n",
    "\n",
    "                    else:\n",
    "                        #print(\"in intron\")\n",
    "                        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"intron\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "\n",
    "            else:\n",
    "                print(\"not gene\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {},
   "outputs": [],
   "source": [
    "overl = new_df[new_df['hol']==\"genes_overlap\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>18968401.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr gene            hol         pos\n",
       "202  NW_022145614.1   NA  genes_overlap  18968401.0"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df2=pd.DataFrame()\n",
    "for index, row in overl.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = int(row.pos)\n",
    "    temp=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    if len(temp) == 1:\n",
    "        print(\"only one\")\n",
    "        print(temp['type'])\n",
    "    if len(temp) > 1:\n",
    "        #print(\"more than 1\")\n",
    "        #checking for gene overlap\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    ids.append(ID)\n",
    "                #else:\n",
    "                    #print(temp[\"type\"].iloc[i])\n",
    "                    #if temp[\"type\"].iloc[i] == \"cDNA_match\":\n",
    "                        #print(\"cDNA_match\")\n",
    "                    #else:\n",
    "                        #print(\"what about ids???\")\n",
    "                        #print(t)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        all_loc=Counter(ids)\n",
    "    \n",
    "        for i in all_loc:\n",
    "            #print(i) #gets loci names\n",
    "            gt=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos) & (df['info'].str.contains(i)) ]\n",
    "            if gt['type'].str.contains('gene').any():\n",
    "                if gt['type'].str.contains('pseudogene').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                # snRNA, tRNA, lnc_RNA, snoRNA, rRNA, miRNA also have line with \"gene\"\n",
    "                elif gt['type'].str.contains('snRNA').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"snRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('tRNA').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"tRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('lnc_RNA').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc_RNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('snoRNA').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"snoRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('rRNA').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"rRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('miRNA').any():\n",
    "                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"miRNA\", \"gene\":i },ignore_index = True)\n",
    "                else:\n",
    "                    if gt['type'].str.contains('exon').any(): \n",
    "                        if gt['type'].str.contains('CDS').any():\n",
    "                            #print(\"in exon\")\n",
    "                            new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"exon\", \"gene\":f\"LOC{i}\" },ignore_index = True) \n",
    "                        else:\n",
    "                            direction=gt['s2'].iloc[0]\n",
    "                            e_start=gt[gt[\"type\"]==\"exon\"].iloc[0].start\n",
    "                            e_stop=gt[gt[\"type\"]==\"exon\"].iloc[0].stop\n",
    "                            g_start=gt[gt[\"type\"]==\"gene\"].iloc[0].start\n",
    "                            g_stop=gt[gt[\"type\"]==\"gene\"].iloc[0].stop\n",
    "                            if direction == \"+\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('5prime1')\n",
    "                                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{i}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"3prime1\")\n",
    "                                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{i}\" },ignore_index = True) \n",
    "                                else:\n",
    "                                    print(index) #alternative UTR?\n",
    "                                    #print(gt)\n",
    "                                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            elif direction == \"-\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('3prime2')\n",
    "                                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{i}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"5prime2\")\n",
    "                                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{i}\" },ignore_index = True)\n",
    "                                else:\n",
    "                                    #print(\"ERROR2\")\n",
    "                                    new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            else:\n",
    "                                print(index)\n",
    "                                #print(gt)\n",
    "\n",
    "                    else:\n",
    "                        #print(\"in intron\")\n",
    "                        new_df2 = new_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"intron\", \"gene\":f\"LOC{i}\" },ignore_index = True) \n",
    "\n",
    "            else:\n",
    "                print(\"not gene\")\n",
    "                    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.pos = new_df.pos.astype(int)\n",
    "new_df2.pos = new_df2.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.to_csv(\"annotated_top_goodhe_01.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df2.to_csv(\"annotated_top_goodhe_01_overlap.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "111"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_df[new_df[\"hol\"]==\"not_annot\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#---------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.read_csv('annotated_top_goodhe_01.csv')\n",
    "new_df2 = pd.read_csv('annotated_top_goodhe_01_overlap.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 457,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])\n",
    "#promoters = promoters[promoters['type']==\"gene\"]\n",
    "\n",
    "promoters.drop(promoters[promoters['type'] == \"region\"].index, inplace = True)\n",
    "\n",
    "\n",
    "imp_list=[\"gene\",\"snRNA\", \"tRNA\", \"lnc_RNA\", \"snoRNA\", \"rRNA\", \"miRNA\"]\n",
    "for l in imp_list:\n",
    "    promoters.loc[(promoters[\"type\"]==l) & (promoters[\"s2\"]==\"+\"), \"start\"] -= 1000\n",
    "    promoters.loc[(promoters[\"type\"]==l) & (promoters[\"s2\"]==\"-\"), \"stop\"] += 1000\n",
    "\n",
    "promoters['info'] = promoters['info'].str.replace(',',';')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 458,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"not_annot\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 459,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>809765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>22684111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>36849648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>13121687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>20882520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6839</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>16153542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6840</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>23380763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>27269202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>25996126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>3170334</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2934 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr gene        hol       pos\n",
       "0     NW_022145599.1  NaN  not_annot    809765\n",
       "1     NW_022145613.1  NaN  not_annot  22684111\n",
       "2     NW_022145600.1  NaN  not_annot  36849648\n",
       "3     NW_022145612.1  NaN  not_annot  13121687\n",
       "4     NW_022145595.1  NaN  not_annot  20882520\n",
       "...              ...  ...        ...       ...\n",
       "6839  NW_022145600.1  NaN  not_annot  16153542\n",
       "6840  NW_022145599.1  NaN  not_annot  23380763\n",
       "6842  NW_022145597.1  NaN  not_annot  27269202\n",
       "6843  NW_022145597.1  NaN  not_annot  25996126\n",
       "6845  NW_022145595.1  NaN  not_annot   3170334\n",
       "\n",
       "[2934 rows x 4 columns]"
      ]
     },
     "execution_count": 459,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 460,
   "metadata": {},
   "outputs": [],
   "source": [
    "prom_df=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = int(row.pos)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    else:\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    #print(ID)\n",
    "                    ids.append(ID)\n",
    "               # else:\n",
    "                    #print(temp[\"type\"].iloc[i])\n",
    "                    #if temp[\"type\"].iloc[i] == \"cDNA_match\":\n",
    "                        #print(\"cDNA_match\")\n",
    "                        #print(ID)\n",
    "                    #else:\n",
    "                        #print(\"what about ids???\")\n",
    "                        #print(t)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        if uni > 1:\n",
    "            #print(\"genes overlap\")\n",
    "            prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"genes_overlap\", \"gene\": \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "        \n",
    "        if uni == 1:\n",
    "            #print(\"new\")\n",
    "            if temp['type'].str.contains('gene').any():\n",
    "                if temp['type'].str.contains('pseudogene').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                # snRNA, tRNA, lnc_RNA, snoRNA, rRNA, miRNA also have line with \"gene\"\n",
    "                elif temp['type'].str.contains('snRNA').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"snRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('tRNA').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"tRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('lnc_RNA').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc_RNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('snoRNA').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"snoRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('rRNA').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"rRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                elif temp['type'].str.contains('miRNA').any():\n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"miRNA\", \"gene\":ids[0] },ignore_index = True)\n",
    "                else:    \n",
    "                    prom_df = prom_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"protein_coding\", \"gene\" : f\"LOC{gene_name}\"},  \n",
    "                                    ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 461,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>NW_022145606.1</td>\n",
       "      <td>115927542</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>51617618.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>105439156</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>4464029.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>115919527</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>5147427.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>4431805.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>NW_022145609.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>38496531.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>115919686</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>29121669.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1145</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>115921769</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>26982685.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>18694201.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1693</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>105436976</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>29688422.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1705</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>1832731.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1813</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>7003765.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1927</th>\n",
       "      <td>NW_022145607.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>39221355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1992</th>\n",
       "      <td>NW_022145606.1</td>\n",
       "      <td>115927542</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>51617558.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2080</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>115921049</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>32277738.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2371</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>29277296.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2433</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>7003884.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2572</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>13279159.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2674</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>115920689</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>16394964.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr       gene            hol         pos\n",
       "154   NW_022145606.1  115927542        lnc_RNA  51617618.0\n",
       "299   NW_022145610.1  105439156        lnc_RNA   4464029.0\n",
       "377   NW_022145612.1  115919527        lnc_RNA   5147427.0\n",
       "380   NW_022145604.1         NA  genes_overlap   4431805.0\n",
       "541   NW_022145609.1         NA  genes_overlap  38496531.0\n",
       "579   NW_022145612.1  115919686           tRNA  29121669.0\n",
       "1145  NW_022145595.1  115921769        lnc_RNA  26982685.0\n",
       "1274  NW_022145597.1         NA  genes_overlap  18694201.0\n",
       "1693  NW_022145597.1  105436976        lnc_RNA  29688422.0\n",
       "1705  NW_022145597.1         NA  genes_overlap   1832731.0\n",
       "1813  NW_022145615.1         NA  genes_overlap   7003765.0\n",
       "1927  NW_022145607.1         NA  genes_overlap  39221355.0\n",
       "1992  NW_022145606.1  115927542        lnc_RNA  51617558.0\n",
       "2080  NW_022145594.1  115921049        lnc_RNA  32277738.0\n",
       "2371  NW_022145596.1         NA  genes_overlap  29277296.0\n",
       "2433  NW_022145615.1         NA  genes_overlap   7003884.0\n",
       "2572  NW_022145615.1         NA  genes_overlap  13279159.0\n",
       "2674  NW_022145615.1  115920689        lnc_RNA  16394964.0"
      ]
     },
     "execution_count": 461,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_df[(prom_df[\"hol\"]!=\"protein_coding\") & (prom_df[\"hol\"]!=\"not_annot\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 462,
   "metadata": {},
   "outputs": [],
   "source": [
    "prom_overl = prom_df[prom_df['hol']==\"genes_overlap\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 463,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>4431805.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>NW_022145609.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>38496531.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>18694201.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1705</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>1832731.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1813</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>7003765.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1927</th>\n",
       "      <td>NW_022145607.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>39221355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2371</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>29277296.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2433</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>7003884.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2572</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>13279159.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr gene            hol         pos\n",
       "380   NW_022145604.1   NA  genes_overlap   4431805.0\n",
       "541   NW_022145609.1   NA  genes_overlap  38496531.0\n",
       "1274  NW_022145597.1   NA  genes_overlap  18694201.0\n",
       "1705  NW_022145597.1   NA  genes_overlap   1832731.0\n",
       "1813  NW_022145615.1   NA  genes_overlap   7003765.0\n",
       "1927  NW_022145607.1   NA  genes_overlap  39221355.0\n",
       "2371  NW_022145596.1   NA  genes_overlap  29277296.0\n",
       "2433  NW_022145615.1   NA  genes_overlap   7003884.0\n",
       "2572  NW_022145615.1   NA  genes_overlap  13279159.0"
      ]
     },
     "execution_count": 463,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_overl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 464,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "434607  NW_022145604.1  tRNAscan-SE  gene  4430754  4431826  .  -  .   \n",
      "434608  NW_022145604.1  tRNAscan-SE  tRNA  4430754  4431826  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "434607  ID=gene-TRNAR-UCG-22;Dbxref=GeneID:115926272;N...  \n",
      "434608  ID=rna-TRNAR-UCG-22;Parent=gene-TRNAR-UCG-22;D...  \n",
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "434610  NW_022145604.1  tRNAscan-SE  gene  4430931  4432003  .  -  .   \n",
      "434611  NW_022145604.1  tRNAscan-SE  tRNA  4430931  4432003  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "434610  ID=gene-TRNAR-UCG-23;Dbxref=GeneID:115926273;N...  \n",
      "434611  ID=rna-TRNAR-UCG-23;Parent=gene-TRNAR-UCG-23;D...  \n",
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "434613  NW_022145604.1  tRNAscan-SE  gene  4431107  4432178  .  -  .   \n",
      "434614  NW_022145604.1  tRNAscan-SE  tRNA  4431107  4432178  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "434613  ID=gene-TRNAR-UCG-24;Dbxref=GeneID:115926274;N...  \n",
      "434614  ID=rna-TRNAR-UCG-24;Parent=gene-TRNAR-UCG-24;D...  \n",
      "                   chr       source  type     start      stop s1 s2 s3  \\\n",
      "613569  NW_022145609.1  tRNAscan-SE  gene  38495778  38496850  .  +  .   \n",
      "613570  NW_022145609.1  tRNAscan-SE  tRNA  38495778  38496850  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "613569  ID=gene-TRNAI-AAU-60;Dbxref=GeneID:115928942;N...  \n",
      "613570  ID=rna-TRNAI-AAU-60;Parent=gene-TRNAI-AAU-60;D...  \n",
      "                   chr       source  type     start      stop s1 s2 s3  \\\n",
      "613572  NW_022145609.1  tRNAscan-SE  gene  38495973  38497045  .  +  .   \n",
      "613573  NW_022145609.1  tRNAscan-SE  tRNA  38495973  38497045  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "613572  ID=gene-TRNAI-AAU-61;Dbxref=GeneID:115928943;N...  \n",
      "613573  ID=rna-TRNAI-AAU-61;Parent=gene-TRNAI-AAU-61;D...  \n",
      "                   chr  source  type     start      stop s1 s2 s3  \\\n",
      "201361  NW_022145597.1  Gnomon  gene  18684772  18695091  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "201361  ID=gene-LOC593302;Dbxref=GeneID:593302;Name=LO...  \n",
      "                   chr  source  type     start      stop s1 s2 s3  \\\n",
      "201403  NW_022145597.1  Gnomon  gene  18693264  18711911  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "201403  ID=gene-LOC593294;Dbxref=GeneID:593294;Name=LO...  \n",
      "                   chr  source     type    start     stop s1 s2 s3  \\\n",
      "186351  NW_022145597.1  Gnomon     gene  1822415  1832904  .  -  .   \n",
      "186352  NW_022145597.1  Gnomon  lnc_RNA  1822415  1832904  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "186351  ID=gene-LOC105437611;Dbxref=GeneID:105437611;N...  \n",
      "186352  ID=rna-XR_970659.2;Parent=gene-LOC105437611;Db...  \n",
      "                   chr  source  type    start     stop s1 s2 s3  \\\n",
      "186359  NW_022145597.1  Gnomon  gene  1831981  1833460  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "186359  ID=gene-LOC105437610;Dbxref=GeneID:105437610;N...  \n",
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "901647  NW_022145615.1  tRNAscan-SE  gene  7003017  7004088  .  +  .   \n",
      "901648  NW_022145615.1  tRNAscan-SE  tRNA  7003017  7004088  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "901647  ID=gene-TRNAC-GCA-16;Dbxref=GeneID:115920799;N...  \n",
      "901648  ID=rna-TRNAC-GCA-16;Parent=gene-TRNAC-GCA-16;D...  \n",
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "901650  NW_022145615.1  tRNAscan-SE  gene  7003719  7004790  .  +  .   \n",
      "901651  NW_022145615.1  tRNAscan-SE  tRNA  7003719  7004790  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "901650  ID=gene-TRNAC-GCA-17;Dbxref=GeneID:115920800;N...  \n",
      "901651  ID=rna-TRNAC-GCA-17;Parent=gene-TRNAC-GCA-17;D...  \n",
      "                   chr  source  type     start      stop s1 s2 s3  \\\n",
      "573943  NW_022145607.1  Gnomon  gene  39214409  39221941  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "573943  ID=gene-LOC578469;Dbxref=GeneID:578469;Name=LO...  \n",
      "                   chr  source  type     start      stop s1 s2 s3  \\\n",
      "573974  NW_022145607.1  Gnomon  gene  39220551  39227728  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "573974  ID=gene-LOC578536;Dbxref=GeneID:578536;Name=LO...  \n",
      "                   chr  source  type     start      stop s1 s2 s3  \\\n",
      "180727  NW_022145596.1  Gnomon  gene  29271950  29278293  .  -  .   \n",
      "\n",
      "                                                     info  \n",
      "180727  ID=gene-LOC587288;Dbxref=GeneID:587288;Name=LO...  \n",
      "                   chr  source  type     start      stop s1 s2 s3  \\\n",
      "180733  NW_022145596.1  Gnomon  gene  29276573  29312287  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "180733  ID=gene-LOC587249;Dbxref=GeneID:587249;Name=LO...  \n",
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "901647  NW_022145615.1  tRNAscan-SE  gene  7003017  7004088  .  +  .   \n",
      "901648  NW_022145615.1  tRNAscan-SE  tRNA  7003017  7004088  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "901647  ID=gene-TRNAC-GCA-16;Dbxref=GeneID:115920799;N...  \n",
      "901648  ID=rna-TRNAC-GCA-16;Parent=gene-TRNAC-GCA-16;D...  \n",
      "                   chr       source  type    start     stop s1 s2 s3  \\\n",
      "901650  NW_022145615.1  tRNAscan-SE  gene  7003719  7004790  .  +  .   \n",
      "901651  NW_022145615.1  tRNAscan-SE  tRNA  7003719  7004790  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "901650  ID=gene-TRNAC-GCA-17;Dbxref=GeneID:115920800;N...  \n",
      "901651  ID=rna-TRNAC-GCA-17;Parent=gene-TRNAC-GCA-17;D...  \n",
      "                   chr       source  type     start      stop s1 s2 s3  \\\n",
      "908046  NW_022145615.1  tRNAscan-SE  gene  13278407  13279479  .  +  .   \n",
      "908047  NW_022145615.1  tRNAscan-SE  tRNA  13278407  13279479  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "908046  ID=gene-TRNAT-AGU-29;Dbxref=GeneID:115920822;N...  \n",
      "908047  ID=rna-TRNAT-AGU-29;Parent=gene-TRNAT-AGU-29;D...  \n",
      "                   chr       source  type     start      stop s1 s2 s3  \\\n",
      "908049  NW_022145615.1  tRNAscan-SE  gene  13278861  13279933  .  +  .   \n",
      "908050  NW_022145615.1  tRNAscan-SE  tRNA  13278861  13279933  .  +  .   \n",
      "\n",
      "                                                     info  \n",
      "908049  ID=gene-TRNAT-AGU-30;Dbxref=GeneID:115920823;N...  \n",
      "908050  ID=rna-TRNAT-AGU-30;Parent=gene-TRNAT-AGU-30;D...  \n"
     ]
    }
   ],
   "source": [
    "prom_df2=pd.DataFrame()\n",
    "for index, row in prom_overl.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = int(row.pos)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    else:\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    #print(ID)\n",
    "                    ids.append(ID)\n",
    "                #else:\n",
    "                    #print(temp[\"type\"].iloc[i])\n",
    "                    #if temp[\"type\"].iloc[i] == \"cDNA_match\":\n",
    "                       # print(\"cDNA_match\")\n",
    "                        #print(ID)\n",
    "                   # else:\n",
    "                       # print(\"what about ids???\")\n",
    "                       # print(t)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        all_loc=Counter(ids)\n",
    "        \n",
    "        for i in all_loc:\n",
    "            #print(i) #gets loci names\n",
    "            gt=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos) & (promoters['info'].str.contains(i)) ]\n",
    "            print(gt)\n",
    "            if gt['type'].str.contains('gene').any():\n",
    "                if gt['type'].str.contains('pseudogene').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                # snRNA, tRNA, lnc_RNA, snoRNA, rRNA, miRNA also have line with \"gene\"\n",
    "                elif gt['type'].str.contains('snRNA').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"snRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('tRNA').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"tRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('lnc_RNA').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc_RNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('snoRNA').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"snoRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('rRNA').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"rRNA\", \"gene\":i },ignore_index = True)\n",
    "                elif gt['type'].str.contains('miRNA').any():\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"miRNA\", \"gene\":i },ignore_index = True)\n",
    "                else:\n",
    "                    prom_df2 = prom_df2.append({'chr' : ch, 'pos' : pos, 'hol' : \"protein_coding\", \"gene\":f\"LOC{i}\" },ignore_index = True)\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 465,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['tRNA', 'protein_coding', 'lnc_RNA'], dtype=object)"
      ]
     },
     "execution_count": 465,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_df2.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 466,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "142"
      ]
     },
     "execution_count": 466,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(prom_df[prom_df[\"hol\"]==\"protein_coding\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 467,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 467,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(prom_df2[prom_df2[\"hol\"]==\"lnc_RNA\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>4431805.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>NW_022145609.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>38496531.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>18694201.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1705</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>1832731.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1813</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>7003765.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1927</th>\n",
       "      <td>NW_022145607.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>39221355.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2371</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>29277296.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2433</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>7003884.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2572</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>NA</td>\n",
       "      <td>genes_overlap</td>\n",
       "      <td>13279159.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr gene            hol         pos\n",
       "380   NW_022145604.1   NA  genes_overlap   4431805.0\n",
       "541   NW_022145609.1   NA  genes_overlap  38496531.0\n",
       "1274  NW_022145597.1   NA  genes_overlap  18694201.0\n",
       "1705  NW_022145597.1   NA  genes_overlap   1832731.0\n",
       "1813  NW_022145615.1   NA  genes_overlap   7003765.0\n",
       "1927  NW_022145607.1   NA  genes_overlap  39221355.0\n",
       "2371  NW_022145596.1   NA  genes_overlap  29277296.0\n",
       "2433  NW_022145615.1   NA  genes_overlap   7003884.0\n",
       "2572  NW_022145615.1   NA  genes_overlap  13279159.0"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_df[prom_df[\"hol\"]==\"genes_overlap\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "151"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(prom_df[(prom_df[\"hol\"]!=\"genes_overlap\") & (prom_df[\"hol\"]!=\"not_annot\")])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, outside of intron"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "are intronic ones located in:\n",
    "    1 ATAC\n",
    "    2 lvar\n",
    "    3 chip \n",
    "    4 lnc\n",
    "    5 promoter\n",
    "    6 TFBS\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])\n",
    "promoters = promoters[promoters['type']==\"gene\"]\n",
    "\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"+\"), \"start\"] -= 1000\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"-\"), \"stop\"] += 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = pd.read_csv(\"atac_mapped_translatedto5.0.csv\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "chip = pd.read_csv('AlxChip_mapped_translatedto5.0.txt', sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lvar = pd.read_csv('lvar_spur_48_50_mapped_translatedto50.csv', header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lnc = pd.read_csv(\"lncRNA_mapped_translatedto5.0.csv\", header=1, names=[\"chr\",\"start\",\"stop\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"not_annot\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "enh_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in promoters\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"promoter\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['lnc', 'promoter'], dtype=object)"
      ]
     },
     "execution_count": 340,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 342,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(enh_check[enh_check[\"hol\"]==\"lnc\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en = enh_check[(enh_check[\"hol\"]==\"lvar\") | (enh_check[\"hol\"]==\"chip\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en=pot_en.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pot_en)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reminder: chosen atac supp (or lvar or chip) too strict?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, inside of intron"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"intron\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "int_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    gene=row.gene\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\",'gene':gene },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\",'gene':gene },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\",'gene':gene },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\",'gene':gene },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['lnc'], dtype=object)"
      ]
     },
     "execution_count": 345,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC100894079</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC579367</td>\n",
       "      <td>lnc</td>\n",
       "      <td>41191507.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC105444100</td>\n",
       "      <td>lnc</td>\n",
       "      <td>20905720.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC105444622</td>\n",
       "      <td>lnc</td>\n",
       "      <td>1523941.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145543.1</td>\n",
       "      <td>LOC115917929</td>\n",
       "      <td>lnc</td>\n",
       "      <td>956594.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>LOC100892961</td>\n",
       "      <td>lnc</td>\n",
       "      <td>29040288.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC762981</td>\n",
       "      <td>lnc</td>\n",
       "      <td>5104468.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC105444132</td>\n",
       "      <td>lnc</td>\n",
       "      <td>17378166.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>NW_022145576.1</td>\n",
       "      <td>LOC100891941</td>\n",
       "      <td>lnc</td>\n",
       "      <td>2132339.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC100893229</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22327453.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>118 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene  hol         pos\n",
       "0    NW_022145601.1  LOC100894079  lnc  22606114.0\n",
       "1    NW_022145612.1     LOC579367  lnc  41191507.0\n",
       "2    NW_022145614.1  LOC105444100  lnc  20905720.0\n",
       "3    NW_022145610.1  LOC105444622  lnc   1523941.0\n",
       "4    NW_022145543.1  LOC115917929  lnc    956594.0\n",
       "..              ...           ...  ...         ...\n",
       "124  NW_022145615.1  LOC100892961  lnc  29040288.0\n",
       "125  NW_022145612.1     LOC762981  lnc   5104468.0\n",
       "126  NW_022145598.1  LOC105444132  lnc  17378166.0\n",
       "127  NW_022145576.1  LOC100891941  lnc   2132339.0\n",
       "129  NW_022145605.1  LOC100893229  lnc  22327453.0\n",
       "\n",
       "[118 rows x 4 columns]"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check[int_check[\"hol\"]==\"lnc\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in = int_check[(int_check[\"hol\"]==\"lvar\") | (int_check[\"hol\"]==\"chip\") | (int_check[\"hol\"]==\"atac\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in=pot_in.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>LOC577823</td>\n",
       "      <td>lvar</td>\n",
       "      <td>18662627.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC754783</td>\n",
       "      <td>chip</td>\n",
       "      <td>5003800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>LOC582776</td>\n",
       "      <td>lvar</td>\n",
       "      <td>8855042.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC588549</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16695182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NW_022145609.1</td>\n",
       "      <td>LOC105445730</td>\n",
       "      <td>chip</td>\n",
       "      <td>1841502.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC100894087</td>\n",
       "      <td>lvar</td>\n",
       "      <td>35316800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC105447203</td>\n",
       "      <td>chip</td>\n",
       "      <td>11733562.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC582326</td>\n",
       "      <td>chip</td>\n",
       "      <td>17759074.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580332</td>\n",
       "      <td>lvar</td>\n",
       "      <td>47720769.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>LOC581963</td>\n",
       "      <td>atac</td>\n",
       "      <td>38079900.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>LOC581963</td>\n",
       "      <td>atac</td>\n",
       "      <td>38080603.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol         pos\n",
       "86   NW_022145600.1     LOC577823  lvar  18662627.0\n",
       "34   NW_022145601.1     LOC754783  chip   5003800.0\n",
       "75   NW_022145604.1     LOC582776  lvar   8855042.0\n",
       "106  NW_022145605.1     LOC588549  lvar  16695182.0\n",
       "37   NW_022145609.1  LOC105445730  chip   1841502.0\n",
       "91   NW_022145610.1  LOC100894087  lvar  35316800.0\n",
       "55   NW_022145612.1  LOC105447203  chip  11733562.0\n",
       "128  NW_022145612.1     LOC580713  lvar  14929793.0\n",
       "13   NW_022145612.1     LOC582326  chip  17759074.0\n",
       "88   NW_022145612.1     LOC580332  lvar  47720769.0\n",
       "23   NW_022145613.1     LOC581963  atac  38079900.0\n",
       "104  NW_022145613.1     LOC581963  atac  38080603.0"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pot_in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>809765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>22684111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>36849648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>13121687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>20882520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>27269202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>25996126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alternative UTR</td>\n",
       "      <td>12214610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>3170334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>LOC582877</td>\n",
       "      <td>intron</td>\n",
       "      <td>16380245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr       gene              hol       pos\n",
       "0     NW_022145599.1        NaN        not_annot    809765\n",
       "1     NW_022145613.1        NaN        not_annot  22684111\n",
       "2     NW_022145600.1        NaN        not_annot  36849648\n",
       "3     NW_022145612.1        NaN        not_annot  13121687\n",
       "4     NW_022145595.1        NaN        not_annot  20882520\n",
       "...              ...        ...              ...       ...\n",
       "6842  NW_022145597.1        NaN        not_annot  27269202\n",
       "6843  NW_022145597.1        NaN        not_annot  25996126\n",
       "6844  NW_022145596.1        NaN  alternative UTR  12214610\n",
       "6845  NW_022145595.1        NaN        not_annot   3170334\n",
       "6846  NW_022145595.1  LOC582877           intron  16380245\n",
       "\n",
       "[6847 rows x 4 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Biomin genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 475,
   "metadata": {},
   "outputs": [],
   "source": [
    "#new_dff = pd.read_csv('annotated_top_goodhe_01_overlap.csv')\n",
    "new_df =new_df.dropna()\n",
    "new_df2 =new_df2.dropna()\n",
    "#prom_df =prom_df.dropna()\n",
    "#prom_df2 =prom_df2.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 454,
   "metadata": {},
   "outputs": [],
   "source": [
    "#comb_set=set(list(new_dff[new_dff[\"gene\"].str.contains(\"LOC\")].gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 476,
   "metadata": {},
   "outputs": [],
   "source": [
    "oloc=list(new_df[new_df[\"gene\"].str.contains(\"LOC\")].gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 477,
   "metadata": {},
   "outputs": [],
   "source": [
    "nloc=list(new_df2[new_df2[\"gene\"].str.contains(\"LOC\")].gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 507,
   "metadata": {},
   "outputs": [],
   "source": [
    "oploc=list(prom_df[prom_df[\"gene\"].str.contains(\"LOC\")].gene)\n",
    "nploc=list(prom_df2[prom_df2[\"gene\"].str.contains(\"LOC\")].gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 527,
   "metadata": {},
   "outputs": [],
   "source": [
    "comb_set=set(oloc + nloc)# + oploc + nploc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'LOC100888258',\n",
       " 'LOC100888559',\n",
       " 'LOC100888919',\n",
       " 'LOC100888943',\n",
       " 'LOC100889081',\n",
       " 'LOC100889178',\n",
       " 'LOC100889636',\n",
       " 'LOC100890044',\n",
       " 'LOC100890291',\n",
       " 'LOC100890901',\n",
       " 'LOC100891054',\n",
       " 'LOC100891151',\n",
       " 'LOC100891656',\n",
       " 'LOC100892015',\n",
       " 'LOC100892245',\n",
       " 'LOC100893334',\n",
       " 'LOC100893463',\n",
       " 'LOC100893580',\n",
       " 'LOC100893707',\n",
       " 'LOC100893741',\n",
       " 'LOC100893814',\n",
       " 'LOC105436405',\n",
       " 'LOC105436540',\n",
       " 'LOC105436858',\n",
       " 'LOC105437164',\n",
       " 'LOC105437450',\n",
       " 'LOC105437610',\n",
       " 'LOC105437744',\n",
       " 'LOC105440755',\n",
       " 'LOC105441802',\n",
       " 'LOC105443575',\n",
       " 'LOC105444109',\n",
       " 'LOC105444601',\n",
       " 'LOC105445286',\n",
       " 'LOC105445349',\n",
       " 'LOC105446050',\n",
       " 'LOC105446450',\n",
       " 'LOC115918170',\n",
       " 'LOC115918473',\n",
       " 'LOC115920308',\n",
       " 'LOC115920337',\n",
       " 'LOC115921812',\n",
       " 'LOC115926117',\n",
       " 'LOC115927664',\n",
       " 'LOC115927899',\n",
       " 'LOC115928085',\n",
       " 'LOC115929921',\n",
       " 'LOC575136',\n",
       " 'LOC575196',\n",
       " 'LOC575433',\n",
       " 'LOC575587',\n",
       " 'LOC575771',\n",
       " 'LOC575816',\n",
       " 'LOC576048',\n",
       " 'LOC576142',\n",
       " 'LOC576394',\n",
       " 'LOC576425',\n",
       " 'LOC576459',\n",
       " 'LOC576527',\n",
       " 'LOC576943',\n",
       " 'LOC576978',\n",
       " 'LOC577037',\n",
       " 'LOC577291',\n",
       " 'LOC577353',\n",
       " 'LOC577797',\n",
       " 'LOC578295',\n",
       " 'LOC578457',\n",
       " 'LOC578924',\n",
       " 'LOC579262',\n",
       " 'LOC579418',\n",
       " 'LOC579505',\n",
       " 'LOC579509',\n",
       " 'LOC579661',\n",
       " 'LOC579746',\n",
       " 'LOC579946',\n",
       " 'LOC579960',\n",
       " 'LOC580090',\n",
       " 'LOC580370',\n",
       " 'LOC580453',\n",
       " 'LOC580556',\n",
       " 'LOC580566',\n",
       " 'LOC580579',\n",
       " 'LOC581313',\n",
       " 'LOC581503',\n",
       " 'LOC582167',\n",
       " 'LOC582362',\n",
       " 'LOC582800',\n",
       " 'LOC582830',\n",
       " 'LOC582837',\n",
       " 'LOC582896',\n",
       " 'LOC582937',\n",
       " 'LOC583108',\n",
       " 'LOC583243',\n",
       " 'LOC583524',\n",
       " 'LOC583754',\n",
       " 'LOC584509',\n",
       " 'LOC584837',\n",
       " 'LOC585739',\n",
       " 'LOC585848',\n",
       " 'LOC586560',\n",
       " 'LOC586604',\n",
       " 'LOC586642',\n",
       " 'LOC586665',\n",
       " 'LOC586731',\n",
       " 'LOC587374',\n",
       " 'LOC587649',\n",
       " 'LOC587947',\n",
       " 'LOC588097',\n",
       " 'LOC588172',\n",
       " 'LOC588497',\n",
       " 'LOC588570',\n",
       " 'LOC589799',\n",
       " 'LOC589863',\n",
       " 'LOC590634',\n",
       " 'LOC590949',\n",
       " 'LOC591243',\n",
       " 'LOC591406',\n",
       " 'LOC591851',\n",
       " 'LOC592049',\n",
       " 'LOC592512',\n",
       " 'LOC592592',\n",
       " 'LOC592866',\n",
       " 'LOC592919',\n",
       " 'LOC593090',\n",
       " 'LOC593397',\n",
       " 'LOC594670',\n",
       " 'LOC594807',\n",
       " 'LOC752636',\n",
       " 'LOC753159',\n",
       " 'LOC753542',\n",
       " 'LOC753746',\n",
       " 'LOC753988',\n",
       " 'LOC754339',\n",
       " 'LOC754687',\n",
       " 'LOC755676',\n",
       " 'LOC756342',\n",
       " 'LOC762753',\n",
       " 'LOC762780',\n",
       " 'LOC763313',\n",
       " 'LOC764011',\n",
       " 'LOC765114'}"
      ]
     },
     "execution_count": 350,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 480,
   "metadata": {},
   "outputs": [],
   "source": [
    "mappin = pd.read_csv('SPU_LOC_echino.csv', sep=\",\", names=['genepage','locname', 'info', 'vmi1','vmi2','vmi3','vmi4'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 481,
   "metadata": {},
   "outputs": [],
   "source": [
    "mappin=mappin.drop(columns=['vmi1',\"vmi3\",\"vmi4\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 482,
   "metadata": {},
   "outputs": [],
   "source": [
    "mappin.drop(mappin[mappin['vmi2'] == \"Manually curated\"].index, inplace = True)\n",
    "mappin.drop(mappin[mappin['vmi2'] == \"Algorithmically generated\"].index, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 483,
   "metadata": {},
   "outputs": [],
   "source": [
    "mappin=mappin.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 484,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_spus = mappin[mappin['vmi2'].str.contains('SPU')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 485,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "mapping = defaultdict(list)\n",
    "\n",
    "for _, (locname, vmi) in my_spus[['locname','vmi2']].iterrows():\n",
    "    \n",
    "    candidates = vmi.split(\"|\")\n",
    "    spus = [el for el in candidates if \"SPU\" in el]\n",
    "    locs = [el for el in candidates if \"LOC\" in el]\n",
    "    if \"LOC\" in locname:\n",
    "        locs.append(locname)\n",
    "        \n",
    "    for spu in spus:\n",
    "        for loc in locs:\n",
    "            mapping[spu].append(loc)\n",
    "        \n",
    "mapping = dict(mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 486,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['LOC582406', 'LOC110987996']"
      ]
     },
     "execution_count": 486,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping[\"SPU_023201\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#i was here with checking high bayenv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 554,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd = pd.read_csv('regulator.csv', header=None, names=[\"chr\",\"start\",\"stop\",\"hol\"],dtype=object)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 555,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>hol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100888859</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100889578</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100889729</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100889993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100891144</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>758</th>\n",
       "      <td>100891319</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>759</th>\n",
       "      <td>100891398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>100893779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>761</th>\n",
       "      <td>373229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>762</th>\n",
       "      <td>373404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>763 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr start stop  hol\n",
       "0    100888859   NaN  NaN  NaN\n",
       "1    100889578   NaN  NaN  NaN\n",
       "2    100889729   NaN  NaN  NaN\n",
       "3    100889993   NaN  NaN  NaN\n",
       "4    100891144   NaN  NaN  NaN\n",
       "..         ...   ...  ...  ...\n",
       "758  100891319   NaN  NaN  NaN\n",
       "759  100891398   NaN  NaN  NaN\n",
       "760  100893779   NaN  NaN  NaN\n",
       "761     373229   NaN  NaN  NaN\n",
       "762     373404   NaN  NaN  NaN\n",
       "\n",
       "[763 rows x 4 columns]"
      ]
     },
     "execution_count": 555,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 556,
   "metadata": {},
   "outputs": [],
   "source": [
    "#erd[\"f\"]=\"LOC\" + erd[\"chr\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 557,
   "metadata": {},
   "outputs": [],
   "source": [
    "#alt_genes=set(erd.f.to_numpy())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 551,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes=erd.chr.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 552,
   "metadata": {},
   "outputs": [],
   "source": [
    "alt_genes=set()\n",
    "for gene in genes:\n",
    "    if gene in mapping:\n",
    "        alt = mapping[gene]\n",
    "        for el in alt:\n",
    "            alt_genes.add(el)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 558,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "763"
      ]
     },
     "execution_count": 558,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(alt_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'LOC100887874',\n",
       " 'LOC100887881',\n",
       " 'LOC100887895',\n",
       " 'LOC100887976',\n",
       " 'LOC100887987',\n",
       " 'LOC100888100',\n",
       " 'LOC100888106',\n",
       " 'LOC100888144',\n",
       " 'LOC100888163',\n",
       " 'LOC100888173',\n",
       " 'LOC100888193',\n",
       " 'LOC100888229',\n",
       " 'LOC100888264',\n",
       " 'LOC100888276',\n",
       " 'LOC100888309',\n",
       " 'LOC100888349',\n",
       " 'LOC100888350',\n",
       " 'LOC100888404',\n",
       " 'LOC100888418',\n",
       " 'LOC100888427',\n",
       " 'LOC100888432',\n",
       " 'LOC100888460',\n",
       " 'LOC100888466',\n",
       " 'LOC100888482',\n",
       " 'LOC100888492',\n",
       " 'LOC100888504',\n",
       " 'LOC100888536',\n",
       " 'LOC100888596',\n",
       " 'LOC100888619',\n",
       " 'LOC100888640',\n",
       " 'LOC100888664',\n",
       " 'LOC100888674',\n",
       " 'LOC100888705',\n",
       " 'LOC100888710',\n",
       " 'LOC100888724',\n",
       " 'LOC100888741',\n",
       " 'LOC100888758',\n",
       " 'LOC100888777',\n",
       " 'LOC100888783',\n",
       " 'LOC100888798',\n",
       " 'LOC100888859',\n",
       " 'LOC100888872',\n",
       " 'LOC100888880',\n",
       " 'LOC100888889',\n",
       " 'LOC100888898',\n",
       " 'LOC100888945',\n",
       " 'LOC100889015',\n",
       " 'LOC100889018',\n",
       " 'LOC100889019',\n",
       " 'LOC100889107',\n",
       " 'LOC100889161',\n",
       " 'LOC100889197',\n",
       " 'LOC100889201',\n",
       " 'LOC100889230',\n",
       " 'LOC100889261',\n",
       " 'LOC100889274',\n",
       " 'LOC100889344',\n",
       " 'LOC100889357',\n",
       " 'LOC100889417',\n",
       " 'LOC100889422',\n",
       " 'LOC100889425',\n",
       " 'LOC100889430',\n",
       " 'LOC100889453',\n",
       " 'LOC100889482',\n",
       " 'LOC100889528',\n",
       " 'LOC100889533',\n",
       " 'LOC100889578',\n",
       " 'LOC100889583',\n",
       " 'LOC100889604',\n",
       " 'LOC100889605',\n",
       " 'LOC100889627',\n",
       " 'LOC100889645',\n",
       " 'LOC100889650',\n",
       " 'LOC100889729',\n",
       " 'LOC100889756',\n",
       " 'LOC100889764',\n",
       " 'LOC100889794',\n",
       " 'LOC100889858',\n",
       " 'LOC100889859',\n",
       " 'LOC100889890',\n",
       " 'LOC100889900',\n",
       " 'LOC100889907',\n",
       " 'LOC100889942',\n",
       " 'LOC100889993',\n",
       " 'LOC100890022',\n",
       " 'LOC100890028',\n",
       " 'LOC100890074',\n",
       " 'LOC100890077',\n",
       " 'LOC100890109',\n",
       " 'LOC100890110',\n",
       " 'LOC100890169',\n",
       " 'LOC100890317',\n",
       " 'LOC100890407',\n",
       " 'LOC100890432',\n",
       " 'LOC100890437',\n",
       " 'LOC100890486',\n",
       " 'LOC100890511',\n",
       " 'LOC100890529',\n",
       " 'LOC100890571',\n",
       " 'LOC100890594',\n",
       " 'LOC100890626',\n",
       " 'LOC100890678',\n",
       " 'LOC100890684',\n",
       " 'LOC100890770',\n",
       " 'LOC100890782',\n",
       " 'LOC100890792',\n",
       " 'LOC100890799',\n",
       " 'LOC100890811',\n",
       " 'LOC100890826',\n",
       " 'LOC100890837',\n",
       " 'LOC100890845',\n",
       " 'LOC100890848',\n",
       " 'LOC100890854',\n",
       " 'LOC100890887',\n",
       " 'LOC100890901',\n",
       " 'LOC100890913',\n",
       " 'LOC100890919',\n",
       " 'LOC100890928',\n",
       " 'LOC100890949',\n",
       " 'LOC100890961',\n",
       " 'LOC100890973',\n",
       " 'LOC100890976',\n",
       " 'LOC100890990',\n",
       " 'LOC100891000',\n",
       " 'LOC100891013',\n",
       " 'LOC100891014',\n",
       " 'LOC100891025',\n",
       " 'LOC100891037',\n",
       " 'LOC100891105',\n",
       " 'LOC100891124',\n",
       " 'LOC100891144',\n",
       " 'LOC100891166',\n",
       " 'LOC100891208',\n",
       " 'LOC100891209',\n",
       " 'LOC100891243',\n",
       " 'LOC100891286',\n",
       " 'LOC100891317',\n",
       " 'LOC100891319',\n",
       " 'LOC100891325',\n",
       " 'LOC100891349',\n",
       " 'LOC100891357',\n",
       " 'LOC100891362',\n",
       " 'LOC100891388',\n",
       " 'LOC100891398',\n",
       " 'LOC100891415',\n",
       " 'LOC100891430',\n",
       " 'LOC100891470',\n",
       " 'LOC100891481',\n",
       " 'LOC100891494',\n",
       " 'LOC100891572',\n",
       " 'LOC100891592',\n",
       " 'LOC100891606',\n",
       " 'LOC100891622',\n",
       " 'LOC100891643',\n",
       " 'LOC100891660',\n",
       " 'LOC100891716',\n",
       " 'LOC100891720',\n",
       " 'LOC100891792',\n",
       " 'LOC100891814',\n",
       " 'LOC100891822',\n",
       " 'LOC100891891',\n",
       " 'LOC100891922',\n",
       " 'LOC100891941',\n",
       " 'LOC100891948',\n",
       " 'LOC100891950',\n",
       " 'LOC100891956',\n",
       " 'LOC100892015',\n",
       " 'LOC100892035',\n",
       " 'LOC100892078',\n",
       " 'LOC100892146',\n",
       " 'LOC100892165',\n",
       " 'LOC100892192',\n",
       " 'LOC100892207',\n",
       " 'LOC100892220',\n",
       " 'LOC100892353',\n",
       " 'LOC100892461',\n",
       " 'LOC100892462',\n",
       " 'LOC100892523',\n",
       " 'LOC100892524',\n",
       " 'LOC100892543',\n",
       " 'LOC100892559',\n",
       " 'LOC100892620',\n",
       " 'LOC100892642',\n",
       " 'LOC100892650',\n",
       " 'LOC100892685',\n",
       " 'LOC100892688',\n",
       " 'LOC100892692',\n",
       " 'LOC100892697',\n",
       " 'LOC100892706',\n",
       " 'LOC100892714',\n",
       " 'LOC100892730',\n",
       " 'LOC100892779',\n",
       " 'LOC100892872',\n",
       " 'LOC100892885',\n",
       " 'LOC100892889',\n",
       " 'LOC100892915',\n",
       " 'LOC100892992',\n",
       " 'LOC100893012',\n",
       " 'LOC100893020',\n",
       " 'LOC100893042',\n",
       " 'LOC100893126',\n",
       " 'LOC100893127',\n",
       " 'LOC100893131',\n",
       " 'LOC100893148',\n",
       " 'LOC100893177',\n",
       " 'LOC100893252',\n",
       " 'LOC100893271',\n",
       " 'LOC100893285',\n",
       " 'LOC100893306',\n",
       " 'LOC100893352',\n",
       " 'LOC100893353',\n",
       " 'LOC100893365',\n",
       " 'LOC100893394',\n",
       " 'LOC100893402',\n",
       " 'LOC100893433',\n",
       " 'LOC100893443',\n",
       " 'LOC100893457',\n",
       " 'LOC100893459',\n",
       " 'LOC100893501',\n",
       " 'LOC100893580',\n",
       " 'LOC100893600',\n",
       " 'LOC100893603',\n",
       " 'LOC100893627',\n",
       " 'LOC100893638',\n",
       " 'LOC100893693',\n",
       " 'LOC100893750',\n",
       " 'LOC100893758',\n",
       " 'LOC100893772',\n",
       " 'LOC100893779',\n",
       " 'LOC100893788',\n",
       " 'LOC100893803',\n",
       " 'LOC100893836',\n",
       " 'LOC100893848',\n",
       " 'LOC100893937',\n",
       " 'LOC100894034',\n",
       " 'LOC100894039',\n",
       " 'LOC100894050',\n",
       " 'LOC100894087',\n",
       " 'LOC100894088',\n",
       " 'LOC100894148',\n",
       " 'LOC373197',\n",
       " 'LOC373200',\n",
       " 'LOC373201',\n",
       " 'LOC373202',\n",
       " 'LOC373203',\n",
       " 'LOC373229',\n",
       " 'LOC373243',\n",
       " 'LOC373244',\n",
       " 'LOC373266',\n",
       " 'LOC373306',\n",
       " 'LOC373307',\n",
       " 'LOC373316',\n",
       " 'LOC373332',\n",
       " 'LOC373356',\n",
       " 'LOC373357',\n",
       " 'LOC373358',\n",
       " 'LOC373359',\n",
       " 'LOC373385',\n",
       " 'LOC373398',\n",
       " 'LOC373400',\n",
       " 'LOC373403',\n",
       " 'LOC373404',\n",
       " 'LOC373462',\n",
       " 'LOC373467',\n",
       " 'LOC373468',\n",
       " 'LOC373470',\n",
       " 'LOC373486',\n",
       " 'LOC373496',\n",
       " 'LOC373506',\n",
       " 'LOC373521',\n",
       " 'LOC373522',\n",
       " 'LOC373527',\n",
       " 'LOC373528',\n",
       " 'LOC373529',\n",
       " 'LOC373531',\n",
       " 'LOC373533',\n",
       " 'LOC373538',\n",
       " 'LOC378468',\n",
       " 'LOC378469',\n",
       " 'LOC378470',\n",
       " 'LOC397069',\n",
       " 'LOC408052',\n",
       " 'LOC414294',\n",
       " 'LOC548625',\n",
       " 'LOC574072',\n",
       " 'LOC574563',\n",
       " 'LOC574567',\n",
       " 'LOC574577',\n",
       " 'LOC574588',\n",
       " 'LOC574674',\n",
       " 'LOC574677',\n",
       " 'LOC574751',\n",
       " 'LOC574797',\n",
       " 'LOC574804',\n",
       " 'LOC574841',\n",
       " 'LOC574894',\n",
       " 'LOC575016',\n",
       " 'LOC575060',\n",
       " 'LOC575083',\n",
       " 'LOC575094',\n",
       " 'LOC575179',\n",
       " 'LOC575191',\n",
       " 'LOC575196',\n",
       " 'LOC575199',\n",
       " 'LOC575200',\n",
       " 'LOC575213',\n",
       " 'LOC575230',\n",
       " 'LOC575231',\n",
       " 'LOC575247',\n",
       " 'LOC575262',\n",
       " 'LOC575292',\n",
       " 'LOC575350',\n",
       " 'LOC575361',\n",
       " 'LOC575435',\n",
       " 'LOC575443',\n",
       " 'LOC575445',\n",
       " 'LOC575472',\n",
       " 'LOC575549',\n",
       " 'LOC575572',\n",
       " 'LOC575591',\n",
       " 'LOC575703',\n",
       " 'LOC575706',\n",
       " 'LOC575745',\n",
       " 'LOC575826',\n",
       " 'LOC575879',\n",
       " 'LOC575920',\n",
       " 'LOC575938',\n",
       " 'LOC575958',\n",
       " 'LOC576033',\n",
       " 'LOC576057',\n",
       " 'LOC576102',\n",
       " 'LOC576117',\n",
       " 'LOC576148',\n",
       " 'LOC576170',\n",
       " 'LOC576184',\n",
       " 'LOC576191',\n",
       " 'LOC576197',\n",
       " 'LOC576228',\n",
       " 'LOC576257',\n",
       " 'LOC576281',\n",
       " 'LOC576303',\n",
       " 'LOC576309',\n",
       " 'LOC576329',\n",
       " 'LOC576339',\n",
       " 'LOC576365',\n",
       " 'LOC576385',\n",
       " 'LOC576422',\n",
       " 'LOC576427',\n",
       " 'LOC576430',\n",
       " 'LOC576433',\n",
       " 'LOC576487',\n",
       " 'LOC576502',\n",
       " 'LOC576535',\n",
       " 'LOC576580',\n",
       " 'LOC576618',\n",
       " 'LOC576632',\n",
       " 'LOC576656',\n",
       " 'LOC576774',\n",
       " 'LOC576795',\n",
       " 'LOC576799',\n",
       " 'LOC576814',\n",
       " 'LOC576870',\n",
       " 'LOC576939',\n",
       " 'LOC576952',\n",
       " 'LOC576971',\n",
       " 'LOC576991',\n",
       " 'LOC577009',\n",
       " 'LOC577033',\n",
       " 'LOC577108',\n",
       " 'LOC577120',\n",
       " 'LOC577135',\n",
       " 'LOC577214',\n",
       " 'LOC577345',\n",
       " 'LOC577397',\n",
       " 'LOC577423',\n",
       " 'LOC577426',\n",
       " 'LOC577456',\n",
       " 'LOC577485',\n",
       " 'LOC577516',\n",
       " 'LOC577601',\n",
       " 'LOC577616',\n",
       " 'LOC577625',\n",
       " 'LOC577661',\n",
       " 'LOC577676',\n",
       " 'LOC577677',\n",
       " 'LOC577702',\n",
       " 'LOC577747',\n",
       " 'LOC577766',\n",
       " 'LOC577786',\n",
       " 'LOC577801',\n",
       " 'LOC577815',\n",
       " 'LOC577823',\n",
       " 'LOC577837',\n",
       " 'LOC577839',\n",
       " 'LOC577863',\n",
       " 'LOC577880',\n",
       " 'LOC577883',\n",
       " 'LOC577924',\n",
       " 'LOC577994',\n",
       " 'LOC578054',\n",
       " 'LOC578072',\n",
       " 'LOC578316',\n",
       " 'LOC578336',\n",
       " 'LOC578373',\n",
       " 'LOC578414',\n",
       " 'LOC578473',\n",
       " 'LOC578584',\n",
       " 'LOC578589',\n",
       " 'LOC578618',\n",
       " 'LOC578637',\n",
       " 'LOC578691',\n",
       " 'LOC578784',\n",
       " 'LOC578854',\n",
       " 'LOC578894',\n",
       " 'LOC579018',\n",
       " 'LOC579092',\n",
       " 'LOC579112',\n",
       " 'LOC579133',\n",
       " 'LOC579182',\n",
       " 'LOC579208',\n",
       " 'LOC579226',\n",
       " 'LOC579264',\n",
       " 'LOC579267',\n",
       " 'LOC579286',\n",
       " 'LOC579306',\n",
       " 'LOC579348',\n",
       " 'LOC579386',\n",
       " 'LOC579474',\n",
       " 'LOC579483',\n",
       " 'LOC579532',\n",
       " 'LOC579568',\n",
       " 'LOC579718',\n",
       " 'LOC579772',\n",
       " 'LOC579838',\n",
       " 'LOC579936',\n",
       " 'LOC579965',\n",
       " 'LOC579988',\n",
       " 'LOC580021',\n",
       " 'LOC580047',\n",
       " 'LOC580050',\n",
       " 'LOC580069',\n",
       " 'LOC580113',\n",
       " 'LOC580133',\n",
       " 'LOC580177',\n",
       " 'LOC580181',\n",
       " 'LOC580200',\n",
       " 'LOC580208',\n",
       " 'LOC580246',\n",
       " 'LOC580358',\n",
       " 'LOC580363',\n",
       " 'LOC580371',\n",
       " 'LOC580404',\n",
       " 'LOC580412',\n",
       " 'LOC580498',\n",
       " 'LOC580518',\n",
       " 'LOC580526',\n",
       " 'LOC580556',\n",
       " 'LOC580656',\n",
       " 'LOC580686',\n",
       " 'LOC580704',\n",
       " 'LOC580734',\n",
       " 'LOC580818',\n",
       " 'LOC580828',\n",
       " 'LOC580856',\n",
       " 'LOC580859',\n",
       " 'LOC580883',\n",
       " 'LOC580897',\n",
       " 'LOC581084',\n",
       " 'LOC581152',\n",
       " 'LOC581160',\n",
       " 'LOC581162',\n",
       " 'LOC581239',\n",
       " 'LOC581266',\n",
       " 'LOC581269',\n",
       " 'LOC581359',\n",
       " 'LOC581446',\n",
       " 'LOC581613',\n",
       " 'LOC581618',\n",
       " 'LOC581625',\n",
       " 'LOC581646',\n",
       " 'LOC581708',\n",
       " 'LOC581716',\n",
       " 'LOC581721',\n",
       " 'LOC581729',\n",
       " 'LOC581783',\n",
       " 'LOC581867',\n",
       " 'LOC581970',\n",
       " 'LOC581992',\n",
       " 'LOC581998',\n",
       " 'LOC582056',\n",
       " 'LOC582139',\n",
       " 'LOC582157',\n",
       " 'LOC582205',\n",
       " 'LOC582266',\n",
       " 'LOC582267',\n",
       " 'LOC582289',\n",
       " 'LOC582334',\n",
       " 'LOC582387',\n",
       " 'LOC582443',\n",
       " 'LOC582509',\n",
       " 'LOC582522',\n",
       " 'LOC582550',\n",
       " 'LOC582683',\n",
       " 'LOC582779',\n",
       " 'LOC582802',\n",
       " 'LOC582832',\n",
       " 'LOC582943',\n",
       " 'LOC583022',\n",
       " 'LOC583063',\n",
       " 'LOC583129',\n",
       " 'LOC583158',\n",
       " 'LOC583193',\n",
       " 'LOC583226',\n",
       " 'LOC583635',\n",
       " 'LOC583668',\n",
       " 'LOC583678',\n",
       " 'LOC583688',\n",
       " 'LOC583705',\n",
       " 'LOC583735',\n",
       " 'LOC583802',\n",
       " 'LOC583865',\n",
       " 'LOC583936',\n",
       " 'LOC583948',\n",
       " 'LOC584138',\n",
       " 'LOC584191',\n",
       " 'LOC584374',\n",
       " 'LOC584513',\n",
       " 'LOC584535',\n",
       " 'LOC584539',\n",
       " 'LOC584640',\n",
       " 'LOC584892',\n",
       " 'LOC585151',\n",
       " 'LOC585260',\n",
       " 'LOC585274',\n",
       " 'LOC585334',\n",
       " 'LOC585491',\n",
       " 'LOC585495',\n",
       " 'LOC585554',\n",
       " 'LOC585561',\n",
       " 'LOC585635',\n",
       " 'LOC585636',\n",
       " 'LOC585655',\n",
       " 'LOC585766',\n",
       " 'LOC585895',\n",
       " 'LOC585918',\n",
       " 'LOC586014',\n",
       " 'LOC586051',\n",
       " 'LOC586069',\n",
       " 'LOC586106',\n",
       " 'LOC586389',\n",
       " 'LOC586414',\n",
       " 'LOC586455',\n",
       " 'LOC586543',\n",
       " 'LOC586691',\n",
       " 'LOC586893',\n",
       " 'LOC586896',\n",
       " 'LOC587009',\n",
       " 'LOC587072',\n",
       " 'LOC587121',\n",
       " 'LOC587141',\n",
       " 'LOC587156',\n",
       " 'LOC587174',\n",
       " 'LOC587204',\n",
       " 'LOC587286',\n",
       " 'LOC587463',\n",
       " 'LOC587495',\n",
       " 'LOC587667',\n",
       " 'LOC587670',\n",
       " 'LOC587679',\n",
       " 'LOC587681',\n",
       " 'LOC587946',\n",
       " 'LOC587959',\n",
       " 'LOC587991',\n",
       " 'LOC588020',\n",
       " 'LOC588024',\n",
       " 'LOC588048',\n",
       " 'LOC588137',\n",
       " 'LOC588186',\n",
       " 'LOC588193',\n",
       " 'LOC588294',\n",
       " 'LOC588358',\n",
       " 'LOC588463',\n",
       " 'LOC588522',\n",
       " 'LOC588533',\n",
       " 'LOC588553',\n",
       " 'LOC588569',\n",
       " 'LOC588583',\n",
       " 'LOC588662',\n",
       " 'LOC588775',\n",
       " 'LOC588816',\n",
       " 'LOC589006',\n",
       " 'LOC589017',\n",
       " 'LOC589099',\n",
       " 'LOC589142',\n",
       " 'LOC589156',\n",
       " 'LOC589211',\n",
       " 'LOC589213',\n",
       " 'LOC589247',\n",
       " 'LOC589273',\n",
       " 'LOC589274',\n",
       " 'LOC589388',\n",
       " 'LOC589453',\n",
       " 'LOC589504',\n",
       " 'LOC589616',\n",
       " 'LOC589806',\n",
       " 'LOC589879',\n",
       " 'LOC589987',\n",
       " 'LOC590038',\n",
       " 'LOC590249',\n",
       " 'LOC590306',\n",
       " 'LOC590322',\n",
       " 'LOC590379',\n",
       " 'LOC590448',\n",
       " 'LOC590592',\n",
       " 'LOC590715',\n",
       " 'LOC590789',\n",
       " 'LOC590807',\n",
       " 'LOC590815',\n",
       " 'LOC590859',\n",
       " 'LOC590866',\n",
       " 'LOC590867',\n",
       " 'LOC590876',\n",
       " 'LOC591088',\n",
       " 'LOC591116',\n",
       " 'LOC591124',\n",
       " 'LOC591167',\n",
       " 'LOC591205',\n",
       " 'LOC591326',\n",
       " 'LOC591431',\n",
       " 'LOC591457',\n",
       " 'LOC591467',\n",
       " 'LOC591524',\n",
       " 'LOC591553',\n",
       " 'LOC591571',\n",
       " 'LOC591657',\n",
       " 'LOC591662',\n",
       " 'LOC591696',\n",
       " 'LOC591714',\n",
       " 'LOC591730',\n",
       " 'LOC591744',\n",
       " 'LOC591795',\n",
       " 'LOC591879',\n",
       " 'LOC591922',\n",
       " 'LOC592051',\n",
       " 'LOC592057',\n",
       " 'LOC592160',\n",
       " 'LOC592389',\n",
       " 'LOC592435',\n",
       " 'LOC592483',\n",
       " 'LOC592517',\n",
       " 'LOC592563',\n",
       " 'LOC592596',\n",
       " 'LOC592735',\n",
       " 'LOC592751',\n",
       " 'LOC592869',\n",
       " 'LOC592877',\n",
       " 'LOC592902',\n",
       " 'LOC592936',\n",
       " 'LOC593001',\n",
       " 'LOC593063',\n",
       " 'LOC593067',\n",
       " 'LOC593151',\n",
       " 'LOC593175',\n",
       " 'LOC593193',\n",
       " 'LOC593304',\n",
       " 'LOC593379',\n",
       " 'LOC593387',\n",
       " 'LOC593417',\n",
       " 'LOC593496',\n",
       " 'LOC593520',\n",
       " 'LOC593587',\n",
       " 'LOC593676',\n",
       " 'LOC593693',\n",
       " 'LOC593781',\n",
       " 'LOC593964',\n",
       " 'LOC593983',\n",
       " 'LOC594176',\n",
       " 'LOC594243',\n",
       " 'LOC594477',\n",
       " 'LOC594492',\n",
       " 'LOC594525',\n",
       " 'LOC594572',\n",
       " 'LOC594596',\n",
       " 'LOC594598',\n",
       " 'LOC594776',\n",
       " 'LOC594820',\n",
       " 'LOC751833',\n",
       " 'LOC751834',\n",
       " 'LOC751845',\n",
       " 'LOC751846',\n",
       " 'LOC751847',\n",
       " 'LOC751860',\n",
       " 'LOC752043',\n",
       " 'LOC752256',\n",
       " 'LOC752324',\n",
       " 'LOC752355',\n",
       " 'LOC752418',\n",
       " 'LOC752563',\n",
       " 'LOC752772',\n",
       " 'LOC752819',\n",
       " 'LOC752837',\n",
       " 'LOC752906',\n",
       " 'LOC752948',\n",
       " 'LOC752963',\n",
       " 'LOC753166',\n",
       " 'LOC753299',\n",
       " 'LOC753325',\n",
       " 'LOC753540',\n",
       " 'LOC753938',\n",
       " 'LOC753978',\n",
       " 'LOC754015',\n",
       " 'LOC754022',\n",
       " 'LOC754050',\n",
       " 'LOC754096',\n",
       " 'LOC754237',\n",
       " 'LOC754454',\n",
       " 'LOC754498',\n",
       " 'LOC754648',\n",
       " 'LOC754664',\n",
       " 'LOC754692',\n",
       " 'LOC754693',\n",
       " 'LOC754978',\n",
       " 'LOC755102',\n",
       " 'LOC755253',\n",
       " 'LOC755325',\n",
       " 'LOC755342',\n",
       " 'LOC755893',\n",
       " 'LOC756187',\n",
       " 'LOC756237',\n",
       " 'LOC756515',\n",
       " 'LOC756604',\n",
       " 'LOC756755',\n",
       " 'LOC756859',\n",
       " 'LOC756925',\n",
       " 'LOC756940',\n",
       " 'LOC757216',\n",
       " 'LOC762402',\n",
       " 'LOC762443',\n",
       " 'LOC762550',\n",
       " 'LOC762552',\n",
       " 'LOC762555',\n",
       " 'LOC762571',\n",
       " 'LOC762765',\n",
       " 'LOC762890',\n",
       " 'LOC763033',\n",
       " 'LOC763056',\n",
       " 'LOC763076',\n",
       " 'LOC763081',\n",
       " 'LOC763090',\n",
       " 'LOC763101',\n",
       " 'LOC763239',\n",
       " 'LOC763325',\n",
       " 'LOC763676',\n",
       " 'LOC763703',\n",
       " 'LOC763973',\n",
       " 'LOC764243',\n",
       " 'LOC764303',\n",
       " 'LOC764728',\n",
       " 'LOC764762',\n",
       " 'LOC764791',\n",
       " 'LOC764966',\n",
       " 'LOC765036',\n",
       " 'LOC765057'}"
      ]
     },
     "execution_count": 399,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alt_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'LOC100888721',\n",
       " 'LOC100892305',\n",
       " 'LOC585380',\n",
       " 'LOC579152',\n",
       " 'LOC100888805',\n",
       " 'LOC756379',\n",
       " 'LOC577683',\n",
       " 'LOC576228',\n",
       " 'LOC763804',\n",
       " 'LOC100889593',\n",
       " 'LOC584537',\n",
       " 'LOC577766',\n",
       " 'LOC100891134',\n",
       " 'LOC576508',\n",
       " 'LOC583289',\n",
       " 'LOC578469',\n",
       " 'LOC100893146',\n",
       " 'LOC577291',\n",
       " 'LOC577545',\n",
       " 'LOC100892160',\n",
       " 'LOC763369',\n",
       " 'LOC764101',\n",
       " 'LOC577313',\n",
       " 'LOC100890361',\n",
       " 'LOC105446972',\n",
       " 'LOC105447312',\n",
       " 'LOC590237',\n",
       " 'LOC105443094',\n",
       " 'LOC115922139',\n",
       " 'LOC593302',\n",
       " 'LOC105441802',\n",
       " 'LOC105446450',\n",
       " 'LOC577155',\n",
       " 'LOC757176',\n",
       " 'LOC576978',\n",
       " 'LOC592598',\n",
       " 'LOC105442195',\n",
       " 'LOC754388',\n",
       " 'LOC100892211',\n",
       " 'LOC584564',\n",
       " 'LOC548623',\n",
       " 'LOC585283',\n",
       " 'LOC580955',\n",
       " 'LOC580066',\n",
       " 'LOC583510',\n",
       " 'LOC594712',\n",
       " 'LOC594342',\n",
       " 'LOC576998',\n",
       " 'LOC115927876',\n",
       " 'LOC100888744',\n",
       " 'LOC586138',\n",
       " 'LOC583749',\n",
       " 'LOC594043',\n",
       " 'LOC115921855',\n",
       " 'LOC582885',\n",
       " 'LOC115920679',\n",
       " 'LOC574747',\n",
       " 'LOC105444822',\n",
       " 'LOC105437963',\n",
       " 'LOC591610',\n",
       " 'LOC578639',\n",
       " 'LOC105438999',\n",
       " 'LOC100892823',\n",
       " 'LOC115921764',\n",
       " 'LOC105441997',\n",
       " 'LOC587252',\n",
       " 'LOC580359',\n",
       " 'LOC100892785',\n",
       " 'LOC754268',\n",
       " 'LOC578593',\n",
       " 'LOC586872',\n",
       " 'LOC100887974',\n",
       " 'LOC594709',\n",
       " 'LOC577479',\n",
       " 'LOC763076',\n",
       " 'LOC586216',\n",
       " 'LOC575835',\n",
       " 'LOC581461',\n",
       " 'LOC578952',\n",
       " 'LOC100889891',\n",
       " 'LOC578924',\n",
       " 'LOC755676',\n",
       " 'LOC592216',\n",
       " 'LOC574667',\n",
       " 'LOC588154',\n",
       " 'LOC586705',\n",
       " 'LOC583308',\n",
       " 'LOC105444929',\n",
       " 'LOC579511',\n",
       " 'LOC594022',\n",
       " 'LOC753880',\n",
       " 'LOC579286',\n",
       " 'LOC587893',\n",
       " 'LOC593558',\n",
       " 'LOC105439363',\n",
       " 'LOC373486',\n",
       " 'LOC577639',\n",
       " 'LOC764585',\n",
       " 'LOC100889485',\n",
       " 'LOC592317',\n",
       " 'LOC115918196',\n",
       " 'LOC115920233',\n",
       " 'LOC755600',\n",
       " 'LOC591308',\n",
       " 'LOC100890901',\n",
       " 'LOC587772',\n",
       " 'LOC588119',\n",
       " 'LOC582326',\n",
       " 'LOC577161',\n",
       " 'LOC576854',\n",
       " 'LOC584816',\n",
       " 'LOC105438481',\n",
       " 'LOC756684',\n",
       " 'LOC589728',\n",
       " 'LOC100888678',\n",
       " 'LOC115920061',\n",
       " 'LOC579202',\n",
       " 'LOC373434',\n",
       " 'LOC580814',\n",
       " 'LOC764821',\n",
       " 'LOC592922',\n",
       " 'LOC582658',\n",
       " 'LOC581386',\n",
       " 'LOC581020',\n",
       " 'LOC576080',\n",
       " 'LOC589590',\n",
       " 'LOC100892171',\n",
       " 'LOC115923684',\n",
       " 'LOC578116',\n",
       " 'LOC575818',\n",
       " 'LOC592845',\n",
       " 'LOC754275',\n",
       " 'LOC588004',\n",
       " 'LOC581330',\n",
       " 'LOC576278',\n",
       " 'LOC592667',\n",
       " 'LOC105440379',\n",
       " 'LOC584380',\n",
       " 'LOC754729',\n",
       " 'LOC587664',\n",
       " 'LOC576588',\n",
       " 'LOC753176',\n",
       " 'LOC105443534',\n",
       " 'LOC105436589',\n",
       " 'LOC583575',\n",
       " 'LOC577678',\n",
       " 'LOC574553',\n",
       " 'LOC578365',\n",
       " 'LOC105436983',\n",
       " 'LOC373288',\n",
       " 'LOC100887920',\n",
       " 'LOC583825',\n",
       " 'LOC575851',\n",
       " 'LOC578108',\n",
       " 'LOC755757',\n",
       " 'LOC576491',\n",
       " 'LOC578787',\n",
       " 'LOC582944',\n",
       " 'LOC592049',\n",
       " 'LOC594484',\n",
       " 'LOC115928085',\n",
       " 'LOC100889101',\n",
       " 'LOC100889307',\n",
       " 'LOC100890552',\n",
       " 'LOC579746',\n",
       " 'LOC590337',\n",
       " 'LOC100892375',\n",
       " 'LOC575600',\n",
       " 'LOC580296',\n",
       " 'LOC373409',\n",
       " 'LOC591009',\n",
       " 'LOC578742',\n",
       " 'LOC100888305',\n",
       " 'LOC100889316',\n",
       " 'LOC115928333',\n",
       " 'LOC378945',\n",
       " 'LOC587464',\n",
       " 'LOC100892833',\n",
       " 'LOC762935',\n",
       " 'LOC577659',\n",
       " 'LOC100890023',\n",
       " 'LOC579289',\n",
       " 'LOC585277',\n",
       " 'LOC575094',\n",
       " 'LOC581926',\n",
       " 'LOC594236',\n",
       " 'LOC576704',\n",
       " 'LOC100891706',\n",
       " 'LOC591190',\n",
       " 'LOC115929738',\n",
       " 'LOC591448',\n",
       " 'LOC100892077',\n",
       " 'LOC577909',\n",
       " 'LOC586488',\n",
       " 'LOC591914',\n",
       " 'LOC763187',\n",
       " 'LOC584514',\n",
       " 'LOC100889515',\n",
       " 'LOC589875',\n",
       " 'LOC575831',\n",
       " 'LOC593601',\n",
       " 'LOC100891707',\n",
       " 'LOC579712',\n",
       " 'LOC574782',\n",
       " 'LOC100892670',\n",
       " 'LOC584551',\n",
       " 'LOC755499',\n",
       " 'LOC580462',\n",
       " 'LOC105437450',\n",
       " 'LOC593589',\n",
       " 'LOC576983',\n",
       " 'LOC754634',\n",
       " 'LOC579416',\n",
       " 'LOC115926126',\n",
       " 'LOC574892',\n",
       " 'LOC581439',\n",
       " 'LOC580246',\n",
       " 'LOC100890536',\n",
       " 'LOC762801',\n",
       " 'LOC105438652',\n",
       " 'LOC577037',\n",
       " 'LOC577626',\n",
       " 'LOC577755',\n",
       " 'LOC763427',\n",
       " 'LOC577316',\n",
       " 'LOC762375',\n",
       " 'LOC575196',\n",
       " 'LOC588997',\n",
       " 'LOC577629',\n",
       " 'LOC592464',\n",
       " 'LOC586665',\n",
       " 'LOC580653',\n",
       " 'LOC581539',\n",
       " 'LOC100893745',\n",
       " 'LOC580355',\n",
       " 'LOC579780',\n",
       " 'LOC586332',\n",
       " 'LOC592492',\n",
       " 'LOC580674',\n",
       " 'LOC100889253',\n",
       " 'LOC588169',\n",
       " 'LOC592055',\n",
       " 'LOC575829',\n",
       " 'LOC100889506',\n",
       " 'LOC105438146',\n",
       " 'LOC100890426',\n",
       " 'LOC580856',\n",
       " 'LOC580517',\n",
       " 'LOC575999',\n",
       " 'LOC577193',\n",
       " 'LOC584368',\n",
       " 'LOC580991',\n",
       " 'LOC105441647',\n",
       " 'LOC578597',\n",
       " 'LOC105441787',\n",
       " 'LOC583524',\n",
       " 'LOC100888046',\n",
       " 'LOC581546',\n",
       " 'LOC397068',\n",
       " 'LOC578304',\n",
       " 'LOC105446077',\n",
       " 'LOC585092',\n",
       " 'LOC100892981',\n",
       " 'LOC584098',\n",
       " 'LOC591201',\n",
       " 'LOC755431',\n",
       " 'LOC580703',\n",
       " 'LOC115927502',\n",
       " 'LOC575558',\n",
       " 'LOC592919',\n",
       " 'LOC579182',\n",
       " 'LOC577995',\n",
       " 'LOC593809',\n",
       " 'LOC585491',\n",
       " 'LOC100893061',\n",
       " 'LOC100890044',\n",
       " 'LOC585096',\n",
       " 'LOC580381',\n",
       " 'LOC588134',\n",
       " 'LOC105445042',\n",
       " 'LOC580804',\n",
       " 'LOC105444850',\n",
       " 'LOC582521',\n",
       " 'LOC580818',\n",
       " 'LOC589120',\n",
       " 'LOC589744',\n",
       " 'LOC574743',\n",
       " 'LOC576943',\n",
       " 'LOC576505',\n",
       " 'LOC581761',\n",
       " 'LOC578834',\n",
       " 'LOC763684',\n",
       " 'LOC593157',\n",
       " 'LOC100890275',\n",
       " 'LOC582488',\n",
       " 'LOC105438416',\n",
       " 'LOC100890776',\n",
       " 'LOC580902',\n",
       " 'LOC755141',\n",
       " 'LOC579837',\n",
       " 'LOC583086',\n",
       " 'LOC115918175',\n",
       " 'LOC755612',\n",
       " 'LOC105440151',\n",
       " 'LOC100890236',\n",
       " 'LOC105443206',\n",
       " 'LOC591247',\n",
       " 'LOC100889845',\n",
       " 'LOC587687',\n",
       " 'LOC578673',\n",
       " 'LOC578485',\n",
       " 'LOC762780',\n",
       " 'LOC581568',\n",
       " 'LOC105444100',\n",
       " 'LOC574791',\n",
       " 'LOC579053',\n",
       " 'LOC105436617',\n",
       " 'LOC587456',\n",
       " 'LOC100889596',\n",
       " 'LOC577801',\n",
       " 'LOC587107',\n",
       " 'LOC763563',\n",
       " 'LOC115929911',\n",
       " 'LOC100889017',\n",
       " 'LOC587010',\n",
       " 'LOC580010',\n",
       " 'LOC581503',\n",
       " 'LOC578247',\n",
       " 'LOC590908',\n",
       " 'LOC754353',\n",
       " 'LOC580694',\n",
       " 'LOC100891124',\n",
       " 'LOC100889313',\n",
       " 'LOC115922235',\n",
       " 'LOC100887881',\n",
       " 'LOC752031',\n",
       " 'LOC105444109',\n",
       " 'LOC582032',\n",
       " 'LOC100893465',\n",
       " 'LOC582361',\n",
       " 'LOC105440739',\n",
       " 'LOC586531',\n",
       " 'LOC577301',\n",
       " 'LOC581823',\n",
       " 'LOC580090',\n",
       " 'LOC105436828',\n",
       " 'LOC105437209',\n",
       " 'LOC593286',\n",
       " 'LOC585628',\n",
       " 'LOC373182',\n",
       " 'LOC577924',\n",
       " 'LOC580579',\n",
       " 'LOC582310',\n",
       " 'LOC115922948',\n",
       " 'LOC583176',\n",
       " 'LOC592189',\n",
       " 'LOC105444860',\n",
       " 'LOC576797',\n",
       " 'LOC115920157',\n",
       " 'LOC579389',\n",
       " 'LOC575561',\n",
       " 'LOC115919856',\n",
       " 'LOC574611',\n",
       " 'LOC105437164',\n",
       " 'LOC575180',\n",
       " 'LOC580494',\n",
       " 'LOC764891',\n",
       " 'LOC577171',\n",
       " 'LOC100888667',\n",
       " 'LOC592585',\n",
       " 'LOC586523',\n",
       " 'LOC105439057',\n",
       " 'LOC582056',\n",
       " 'LOC579112',\n",
       " 'LOC579132',\n",
       " 'LOC105438852',\n",
       " 'LOC583473',\n",
       " 'LOC582140',\n",
       " 'LOC752806',\n",
       " 'LOC585202',\n",
       " 'LOC754797',\n",
       " 'LOC764505',\n",
       " 'LOC105437610',\n",
       " 'LOC580824',\n",
       " 'LOC373200',\n",
       " 'LOC115929616',\n",
       " 'LOC764874',\n",
       " 'LOC100893074',\n",
       " 'LOC576824',\n",
       " 'LOC583996',\n",
       " 'LOC586838',\n",
       " 'LOC100893967',\n",
       " 'LOC100889386',\n",
       " 'LOC100887925',\n",
       " 'LOC115919355',\n",
       " 'LOC586105',\n",
       " 'LOC115918427',\n",
       " 'LOC100890051',\n",
       " 'LOC100887875',\n",
       " 'LOC575602',\n",
       " 'LOC100888902',\n",
       " 'LOC100893051',\n",
       " 'LOC576365',\n",
       " 'LOC592705',\n",
       " 'LOC592342',\n",
       " 'LOC764194',\n",
       " 'LOC574629',\n",
       " 'LOC100894079',\n",
       " 'LOC577417',\n",
       " 'LOC590418',\n",
       " 'LOC575373',\n",
       " 'LOC100891668',\n",
       " 'LOC100888574',\n",
       " 'LOC105442979',\n",
       " 'LOC105436674',\n",
       " 'LOC580263',\n",
       " 'LOC755833',\n",
       " 'LOC587199',\n",
       " 'LOC576783',\n",
       " 'LOC373178',\n",
       " 'LOC586721',\n",
       " 'LOC414054',\n",
       " 'LOC588190',\n",
       " 'LOC579469',\n",
       " 'LOC105436978',\n",
       " 'LOC579784',\n",
       " 'LOC105446097',\n",
       " 'LOC586938',\n",
       " 'LOC100889866',\n",
       " 'LOC115929607',\n",
       " 'LOC577702',\n",
       " 'LOC589864',\n",
       " 'LOC100889828',\n",
       " 'LOC581873',\n",
       " 'LOC756234',\n",
       " 'LOC763730',\n",
       " 'LOC105445960',\n",
       " 'LOC756647',\n",
       " 'LOC752300',\n",
       " 'LOC115917893',\n",
       " 'LOC580383',\n",
       " 'LOC100889859',\n",
       " 'LOC764715',\n",
       " 'LOC754582',\n",
       " 'LOC115919774',\n",
       " 'LOC589053',\n",
       " 'LOC100888461',\n",
       " 'LOC100890274',\n",
       " 'LOC100894146',\n",
       " 'LOC115918318',\n",
       " 'LOC105441943',\n",
       " 'LOC585959',\n",
       " 'LOC587672',\n",
       " 'LOC587864',\n",
       " 'LOC585848',\n",
       " 'LOC578181',\n",
       " 'LOC755699',\n",
       " 'LOC578083',\n",
       " 'LOC100890625',\n",
       " 'LOC593769',\n",
       " 'LOC100888361',\n",
       " 'LOC100891867',\n",
       " 'LOC584788',\n",
       " 'LOC100888632',\n",
       " 'LOC594565',\n",
       " 'LOC100891748',\n",
       " 'LOC578960',\n",
       " 'LOC100890679',\n",
       " 'LOC100893224',\n",
       " 'LOC590092',\n",
       " 'LOC582116',\n",
       " 'LOC583848',\n",
       " 'LOC100890286',\n",
       " 'LOC100891657',\n",
       " 'LOC580883',\n",
       " 'LOC105443939',\n",
       " 'LOC575887',\n",
       " 'LOC115927087',\n",
       " 'LOC100890930',\n",
       " 'LOC115925506',\n",
       " 'LOC100893962',\n",
       " 'LOC579274',\n",
       " 'LOC378470',\n",
       " 'LOC580049',\n",
       " 'LOC580086',\n",
       " 'LOC583334',\n",
       " 'LOC591243',\n",
       " 'LOC115929764',\n",
       " 'LOC592814',\n",
       " 'LOC100894039',\n",
       " 'LOC580181',\n",
       " 'LOC575072',\n",
       " 'LOC100890609',\n",
       " 'LOC100892657',\n",
       " 'LOC584518',\n",
       " 'LOC584303',\n",
       " 'LOC100889099',\n",
       " 'LOC100890574',\n",
       " 'LOC577098',\n",
       " 'LOC583832',\n",
       " 'LOC100893429',\n",
       " 'LOC100888210',\n",
       " 'LOC580777',\n",
       " 'LOC115927899',\n",
       " 'LOC594726',\n",
       " 'LOC584681',\n",
       " 'LOC100890539',\n",
       " 'LOC585408',\n",
       " 'LOC584038',\n",
       " 'LOC100890029',\n",
       " 'LOC100888294',\n",
       " 'LOC100888534',\n",
       " 'LOC591026',\n",
       " 'LOC373524',\n",
       " 'LOC587404',\n",
       " 'LOC579579',\n",
       " 'LOC581006',\n",
       " 'LOC576450',\n",
       " 'LOC100888295',\n",
       " 'LOC100893810',\n",
       " 'LOC579411',\n",
       " 'LOC100893467',\n",
       " 'LOC575673',\n",
       " 'LOC100892534',\n",
       " 'LOC589201',\n",
       " 'LOC584940',\n",
       " 'LOC763532',\n",
       " 'LOC580994',\n",
       " 'LOC752325',\n",
       " 'LOC577634',\n",
       " 'LOC584718',\n",
       " 'LOC588497',\n",
       " 'LOC589579',\n",
       " 'LOC764044',\n",
       " 'LOC105437126',\n",
       " 'LOC576717',\n",
       " 'LOC588183',\n",
       " 'LOC105438491',\n",
       " 'LOC578345',\n",
       " 'LOC593971',\n",
       " 'LOC577272',\n",
       " 'LOC577567',\n",
       " 'LOC100889994',\n",
       " 'LOC593238',\n",
       " 'LOC577443',\n",
       " 'LOC115917881',\n",
       " 'LOC100891710',\n",
       " 'LOC582643',\n",
       " 'LOC577156',\n",
       " 'LOC592682',\n",
       " 'LOC586173',\n",
       " 'LOC115920164',\n",
       " 'LOC100893739',\n",
       " 'LOC575010',\n",
       " 'LOC100888493',\n",
       " 'LOC576711',\n",
       " 'LOC576911',\n",
       " 'LOC586471',\n",
       " 'LOC105440182',\n",
       " 'LOC585798',\n",
       " 'LOC594829',\n",
       " 'LOC586083',\n",
       " 'LOC754774',\n",
       " 'LOC105441736',\n",
       " 'LOC592209',\n",
       " 'LOC578824',\n",
       " 'LOC763326',\n",
       " 'LOC756696',\n",
       " 'LOC583936',\n",
       " 'LOC763680',\n",
       " 'LOC588126',\n",
       " 'LOC115926475',\n",
       " 'LOC587255',\n",
       " 'LOC115924742',\n",
       " 'LOC578586',\n",
       " 'LOC755522',\n",
       " 'LOC763114',\n",
       " 'LOC579510',\n",
       " 'LOC590488',\n",
       " 'LOC590455',\n",
       " 'LOC764487',\n",
       " 'LOC590226',\n",
       " 'LOC105445026',\n",
       " 'LOC100894098',\n",
       " 'LOC585805',\n",
       " 'LOC581041',\n",
       " 'LOC100893839',\n",
       " 'LOC575102',\n",
       " 'LOC580388',\n",
       " 'LOC586909',\n",
       " 'LOC100893701',\n",
       " 'LOC100889094',\n",
       " 'LOC587396',\n",
       " 'LOC105446366',\n",
       " 'LOC100891014',\n",
       " 'LOC582557',\n",
       " 'LOC100890329',\n",
       " 'LOC582192',\n",
       " 'LOC580304',\n",
       " 'LOC590369',\n",
       " 'LOC580984',\n",
       " 'LOC593191',\n",
       " 'LOC591154',\n",
       " 'LOC105439745',\n",
       " 'LOC589799',\n",
       " 'LOC580152',\n",
       " 'LOC100893601',\n",
       " 'LOC591807',\n",
       " 'LOC593225',\n",
       " 'LOC581163',\n",
       " 'LOC581763',\n",
       " 'LOC581393',\n",
       " 'LOC587798',\n",
       " 'LOC373241',\n",
       " 'LOC105439056',\n",
       " 'LOC583506',\n",
       " 'LOC100891375',\n",
       " 'LOC574589',\n",
       " 'LOC585164',\n",
       " 'LOC589769',\n",
       " 'LOC583929',\n",
       " 'LOC576535',\n",
       " 'LOC589354',\n",
       " 'LOC576169',\n",
       " 'LOC587910',\n",
       " 'LOC100893334',\n",
       " 'LOC100889363',\n",
       " 'LOC592432',\n",
       " 'LOC100892870',\n",
       " 'LOC100888159',\n",
       " 'LOC105447203',\n",
       " 'LOC578295',\n",
       " 'LOC756624',\n",
       " 'LOC581154',\n",
       " 'LOC100890824',\n",
       " 'LOC576102',\n",
       " 'LOC584274',\n",
       " 'LOC100892905',\n",
       " 'LOC100890367',\n",
       " 'LOC593427',\n",
       " 'LOC100888006',\n",
       " 'LOC593885',\n",
       " 'LOC577137',\n",
       " 'LOC754554',\n",
       " 'LOC100892024',\n",
       " 'LOC575431',\n",
       " 'LOC105443840',\n",
       " 'LOC105445134',\n",
       " 'LOC581667',\n",
       " 'LOC115921849',\n",
       " 'LOC757189',\n",
       " 'LOC115929631',\n",
       " 'LOC576617',\n",
       " 'LOC591166',\n",
       " 'LOC100890677',\n",
       " 'LOC583722',\n",
       " 'LOC100891808',\n",
       " 'LOC586410',\n",
       " 'LOC579914',\n",
       " 'LOC590195',\n",
       " 'LOC575022',\n",
       " 'LOC100890615',\n",
       " 'LOC105440766',\n",
       " 'LOC583872',\n",
       " 'LOC756540',\n",
       " 'LOC100889178',\n",
       " 'LOC581605',\n",
       " 'LOC579243',\n",
       " 'LOC100888258',\n",
       " 'LOC100889092',\n",
       " 'LOC587451',\n",
       " 'LOC100888109',\n",
       " 'LOC589927',\n",
       " 'LOC115925851',\n",
       " 'LOC115920214',\n",
       " 'LOC756550',\n",
       " 'LOC100889821',\n",
       " 'LOC100890558',\n",
       " 'LOC593924',\n",
       " 'LOC574673',\n",
       " 'LOC577286',\n",
       " 'LOC577102',\n",
       " 'LOC100893160',\n",
       " 'LOC579946',\n",
       " 'LOC589644',\n",
       " 'LOC100889010',\n",
       " 'LOC581930',\n",
       " 'LOC575858',\n",
       " 'LOC576953',\n",
       " 'LOC115919090',\n",
       " 'LOC575643',\n",
       " 'LOC100892687',\n",
       " 'LOC592490',\n",
       " 'LOC105436540',\n",
       " 'LOC752367',\n",
       " 'LOC585395',\n",
       " 'LOC752246',\n",
       " 'LOC576048',\n",
       " 'LOC578150',\n",
       " 'LOC589130',\n",
       " 'LOC594807',\n",
       " 'LOC575206',\n",
       " 'LOC100892401',\n",
       " 'LOC589496',\n",
       " 'LOC581609',\n",
       " 'LOC100889246',\n",
       " 'LOC100891059',\n",
       " 'LOC578590',\n",
       " 'LOC764755',\n",
       " 'LOC100888702',\n",
       " 'LOC105442235',\n",
       " 'LOC584097',\n",
       " 'LOC591003',\n",
       " 'LOC115929676',\n",
       " 'LOC587014',\n",
       " 'LOC100893280',\n",
       " 'LOC592346',\n",
       " 'LOC575381',\n",
       " 'LOC105440970',\n",
       " 'LOC100894103',\n",
       " 'LOC582167',\n",
       " 'LOC100893139',\n",
       " 'LOC373287',\n",
       " 'LOC105436450',\n",
       " 'LOC105440161',\n",
       " 'LOC582058',\n",
       " 'LOC591109',\n",
       " 'LOC100891591',\n",
       " 'LOC100888387',\n",
       " 'LOC575148',\n",
       " 'LOC105446106',\n",
       " 'LOC577395',\n",
       " 'LOC100893515',\n",
       " 'LOC582052',\n",
       " 'LOC100888242',\n",
       " 'LOC577668',\n",
       " 'LOC763050',\n",
       " 'LOC576448',\n",
       " 'LOC580710',\n",
       " 'LOC100892002',\n",
       " 'LOC583980',\n",
       " 'LOC593111',\n",
       " 'LOC589658',\n",
       " 'LOC583252',\n",
       " 'LOC100889636',\n",
       " 'LOC585223',\n",
       " 'LOC575146',\n",
       " 'LOC579722',\n",
       " 'LOC582291',\n",
       " 'LOC100893600',\n",
       " 'LOC754777',\n",
       " 'LOC105439231',\n",
       " 'LOC575472',\n",
       " 'LOC577780',\n",
       " 'LOC578509',\n",
       " 'LOC100890874',\n",
       " 'LOC580747',\n",
       " 'LOC577688',\n",
       " 'LOC583206',\n",
       " 'LOC590007',\n",
       " 'LOC576982',\n",
       " 'LOC105444326',\n",
       " 'LOC100888362',\n",
       " 'LOC584296',\n",
       " 'LOC576914',\n",
       " 'LOC762514',\n",
       " 'LOC755988',\n",
       " 'LOC594646',\n",
       " 'LOC105445959',\n",
       " 'LOC591618',\n",
       " 'LOC591372',\n",
       " 'LOC105445886',\n",
       " 'LOC574894',\n",
       " 'LOC580854',\n",
       " 'LOC762418',\n",
       " 'LOC100891791',\n",
       " 'LOC583616',\n",
       " 'LOC578774',\n",
       " 'LOC583108',\n",
       " 'LOC373363',\n",
       " 'LOC100890586',\n",
       " 'LOC584014',\n",
       " 'LOC579490',\n",
       " 'LOC115927263',\n",
       " 'LOC589264',\n",
       " 'LOC586731',\n",
       " 'LOC578227',\n",
       " 'LOC586547',\n",
       " 'LOC577121',\n",
       " 'LOC105444816',\n",
       " 'LOC582903',\n",
       " 'LOC755287',\n",
       " 'LOC754028',\n",
       " 'LOC100891447',\n",
       " 'LOC105443908',\n",
       " 'LOC100892714',\n",
       " 'LOC577957',\n",
       " 'LOC100891399',\n",
       " 'LOC105439145',\n",
       " 'LOC585896',\n",
       " 'LOC100891054',\n",
       " 'LOC594750',\n",
       " 'LOC582635',\n",
       " 'LOC578767',\n",
       " 'LOC100889521',\n",
       " 'LOC100893314',\n",
       " 'LOC105443889',\n",
       " 'LOC588105',\n",
       " 'LOC586036',\n",
       " 'LOC100888814',\n",
       " 'LOC576939',\n",
       " 'LOC579423',\n",
       " 'LOC582206',\n",
       " 'LOC578141',\n",
       " 'LOC583396',\n",
       " 'LOC100893136',\n",
       " 'LOC100890383',\n",
       " 'LOC581833',\n",
       " 'LOC762451',\n",
       " 'LOC580566',\n",
       " 'LOC757011',\n",
       " 'LOC105445399',\n",
       " 'LOC577783',\n",
       " 'LOC100890668',\n",
       " 'LOC582114',\n",
       " 'LOC582112',\n",
       " 'LOC591161',\n",
       " 'LOC100890076',\n",
       " 'LOC100894143',\n",
       " 'LOC100892428',\n",
       " 'LOC577428',\n",
       " 'LOC105447140',\n",
       " 'LOC752237',\n",
       " 'LOC575625',\n",
       " 'LOC587030',\n",
       " 'LOC575370',\n",
       " 'LOC100888699',\n",
       " 'LOC580173',\n",
       " 'LOC589758',\n",
       " 'LOC586737',\n",
       " 'LOC100888466',\n",
       " 'LOC589737',\n",
       " 'LOC581284',\n",
       " 'LOC587591',\n",
       " 'LOC578737',\n",
       " 'LOC576221',\n",
       " 'LOC115929932',\n",
       " 'LOC581723',\n",
       " 'LOC105436999',\n",
       " 'LOC580452',\n",
       " 'LOC579021',\n",
       " 'LOC100891396',\n",
       " 'LOC581686',\n",
       " 'LOC752867',\n",
       " 'LOC590679',\n",
       " 'LOC588358',\n",
       " 'LOC589021',\n",
       " 'LOC590854',\n",
       " 'LOC581018',\n",
       " 'LOC578457',\n",
       " 'LOC583469',\n",
       " 'LOC373316',\n",
       " 'LOC105439711',\n",
       " 'LOC576034',\n",
       " 'LOC763505',\n",
       " 'LOC584369',\n",
       " 'LOC763009',\n",
       " 'LOC100889512',\n",
       " 'LOC763805',\n",
       " 'LOC115919459',\n",
       " 'LOC589156',\n",
       " 'LOC592747',\n",
       " 'LOC579944',\n",
       " 'LOC576399',\n",
       " 'LOC578429',\n",
       " 'LOC582641',\n",
       " 'LOC100890449',\n",
       " 'LOC100891941',\n",
       " 'LOC584790',\n",
       " 'LOC580472',\n",
       " 'LOC575025',\n",
       " 'LOC590672',\n",
       " 'LOC763325',\n",
       " 'LOC580684',\n",
       " 'LOC586779',\n",
       " 'LOC115918356',\n",
       " 'LOC100892317',\n",
       " 'LOC100893463',\n",
       " 'LOC100892920',\n",
       " 'LOC578044',\n",
       " 'LOC590861',\n",
       " 'LOC105442777',\n",
       " 'LOC100891820',\n",
       " 'LOC100891646',\n",
       " 'LOC100892041',\n",
       " 'LOC100889942',\n",
       " 'LOC754213',\n",
       " 'LOC591844',\n",
       " 'LOC589106',\n",
       " 'LOC576394',\n",
       " 'LOC756628',\n",
       " 'LOC579661',\n",
       " 'LOC582578',\n",
       " 'LOC752951',\n",
       " 'LOC100891456',\n",
       " 'LOC582857',\n",
       " 'LOC105440851',\n",
       " 'LOC115929614',\n",
       " 'LOC752227',\n",
       " 'LOC763355',\n",
       " 'LOC579609',\n",
       " 'LOC100892761',\n",
       " 'LOC586687',\n",
       " 'LOC100891566',\n",
       " 'LOC100893707',\n",
       " 'LOC582837',\n",
       " 'LOC100893885',\n",
       " 'LOC575483',\n",
       " 'LOC105444466',\n",
       " 'LOC764665',\n",
       " 'LOC100890113',\n",
       " 'LOC373228',\n",
       " 'LOC577823',\n",
       " 'LOC100891417',\n",
       " 'LOC105438000',\n",
       " 'LOC100890886',\n",
       " 'LOC576654',\n",
       " 'LOC591779',\n",
       " 'LOC100890651',\n",
       " 'LOC576645',\n",
       " 'LOC588536',\n",
       " 'LOC578628',\n",
       " 'LOC100892350',\n",
       " 'LOC589151',\n",
       " 'LOC589708',\n",
       " 'LOC105438134',\n",
       " 'LOC584660',\n",
       " 'LOC575819',\n",
       " 'LOC578115',\n",
       " 'LOC594488',\n",
       " 'LOC580600',\n",
       " 'LOC576882',\n",
       " 'LOC115929347',\n",
       " 'LOC589395',\n",
       " 'LOC579069',\n",
       " 'LOC590507',\n",
       " 'LOC594620',\n",
       " 'LOC577039',\n",
       " 'LOC100892105',\n",
       " 'LOC105440894',\n",
       " 'LOC548625',\n",
       " 'LOC115925342',\n",
       " 'LOC100893604',\n",
       " 'LOC583670',\n",
       " 'LOC753109',\n",
       " 'LOC579367',\n",
       " 'LOC577598',\n",
       " 'LOC105436985',\n",
       " 'LOC583353',\n",
       " 'LOC115929081',\n",
       " 'LOC100888221',\n",
       " 'LOC115928086',\n",
       " 'LOC586120',\n",
       " 'LOC578026',\n",
       " 'LOC105441791',\n",
       " 'LOC577423',\n",
       " 'LOC105439181',\n",
       " 'LOC593847',\n",
       " 'LOC593587',\n",
       " 'LOC100890784',\n",
       " 'LOC593397',\n",
       " 'LOC591940',\n",
       " 'LOC752636',\n",
       " 'LOC590907',\n",
       " 'LOC105441737',\n",
       " 'LOC100890478',\n",
       " 'LOC754661',\n",
       " 'LOC115919017',\n",
       " 'LOC592718',\n",
       " 'LOC100889171',\n",
       " 'LOC100888724',\n",
       " 'LOC588089',\n",
       " 'LOC576114',\n",
       " 'LOC580666',\n",
       " 'LOC594592',\n",
       " 'LOC576124',\n",
       " 'LOC105441205',\n",
       " 'LOC576500',\n",
       " 'LOC577465',\n",
       " 'LOC762747',\n",
       " 'LOC580668',\n",
       " 'LOC586739',\n",
       " 'LOC752999',\n",
       " 'LOC585636',\n",
       " 'LOC763378',\n",
       " 'LOC581083',\n",
       " 'LOC100889733',\n",
       " 'LOC577224',\n",
       " 'LOC581884',\n",
       " 'LOC594516',\n",
       " ...}"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 559,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "128"
      ]
     },
     "execution_count": 559,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(comb_set.intersection(alt_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 529,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'LOC100890645',\n",
       " 'LOC100892375',\n",
       " 'LOC100893839',\n",
       " 'LOC576535',\n",
       " 'LOC576799',\n",
       " 'LOC577426',\n",
       " 'LOC577766',\n",
       " 'LOC581625',\n",
       " 'LOC581867',\n",
       " 'LOC585260',\n",
       " 'LOC585636',\n",
       " 'LOC587946',\n",
       " 'LOC588089',\n",
       " 'LOC589006',\n",
       " 'LOC591691',\n",
       " 'LOC591914',\n",
       " 'LOC756958'}"
      ]
     },
     "execution_count": 529,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb_set.intersection(alt_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 546,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6238</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC756958</td>\n",
       "      <td>intron</td>\n",
       "      <td>3918069</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr       gene     hol      pos\n",
       "6238  NW_022145610.1  LOC756958  intron  3918069"
      ]
     },
     "execution_count": 546,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df[new_df['gene']==\"LOC756958\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 505,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC575445</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>20730134.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr       gene             hol         pos\n",
       "485  NW_022145598.1  LOC575445  protein_coding  20730134.0"
      ]
     },
     "execution_count": 505,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prom_df[prom_df['gene']==\"LOC575445\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'LOC587946'"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from random import choice\n",
    "\n",
    "\n",
    "choice(list(comb_set.intersection(alt_genes)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## let's check for my old list (what i wanted to get probes for)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd = pd.read_csv('my_old_list.csv', header=None, names=[\"chr\",\"start\",\"stop\",\"hol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>hol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Chromosome</td>\n",
       "      <td>Start</td>\n",
       "      <td>End</td>\n",
       "      <td>Gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145611.1</td>\n",
       "      <td>20239836.0</td>\n",
       "      <td>20262988.0</td>\n",
       "      <td>Otx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>19666380.0</td>\n",
       "      <td>19695392.0</td>\n",
       "      <td>B-catenin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145607.1</td>\n",
       "      <td>33413785.0</td>\n",
       "      <td>33447745.0</td>\n",
       "      <td>Blimp1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>3234303.0</td>\n",
       "      <td>3240831.0</td>\n",
       "      <td>Wnt8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr       start        stop        hol\n",
       "0        Chromosome       Start         End       Gene\n",
       "1    NW_022145611.1  20239836.0  20262988.0        Otx\n",
       "2    NW_022145613.1  19666380.0  19695392.0  B-catenin\n",
       "3    NW_022145607.1  33413785.0  33447745.0     Blimp1\n",
       "4    NW_022145595.1   3234303.0   3240831.0       Wnt8\n",
       "..              ...         ...         ...        ...\n",
       "125             NaN         NaN         NaN        NaN\n",
       "126             NaN         NaN         NaN        NaN\n",
       "127             NaN         NaN         NaN        NaN\n",
       "128             NaN         NaN         NaN        NaN\n",
       "129             NaN         NaN         NaN        NaN\n",
       "\n",
       "[130 rows x 4 columns]"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd.drop([0], inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd=erd.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-233-fc4b1870e431>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  erd[\"start\"]=erd.start.astype(float)\n",
      "<ipython-input-233-fc4b1870e431>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  erd[\"stop\"]=erd.stop.astype(float)\n"
     ]
    }
   ],
   "source": [
    "erd[\"start\"]=erd.start.astype(float)\n",
    "erd[\"stop\"]=erd.stop.astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Otx'"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd[\"hol\"].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd_check=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    fst = row[1]\n",
    "    temp= erd[(erd['chr']==ch) & (erd['start']<=pos) & (erd['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        erd_check = erd_check.append({'chr' : ch, 'pos' : pos, 'hol' : temp[\"hol\"].iloc[0], \"fst\" : fst},ignore_index = True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>ATPsythase</td>\n",
       "      <td>31275084.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1058</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14678986.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1059</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14662026.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1083</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14694778.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1216</td>\n",
       "      <td>B-catenin</td>\n",
       "      <td>19685318.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1602</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30320957.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1013</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30331129.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1127</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30330987.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>0.1049</td>\n",
       "      <td>CD109</td>\n",
       "      <td>3635077.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20094418.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20094442.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20080205.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20071223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara14la</td>\n",
       "      <td>27358332.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1024</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5712045.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5721058.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1018</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5709506.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5727140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1084</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5722898.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara8LB</td>\n",
       "      <td>5781742.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1101</td>\n",
       "      <td>Ets1</td>\n",
       "      <td>41420756.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1100</td>\n",
       "      <td>NCAM</td>\n",
       "      <td>19199989.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1561</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>627088.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1149</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>669637.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1042</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>670140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>639916.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>0.1056</td>\n",
       "      <td>Wnt8</td>\n",
       "      <td>3240002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1292</td>\n",
       "      <td>calcium-sensing</td>\n",
       "      <td>12985454.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1184</td>\n",
       "      <td>cholinesterase</td>\n",
       "      <td>14505911.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>collagen</td>\n",
       "      <td>4993850.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>0.1057</td>\n",
       "      <td>collagen alpha</td>\n",
       "      <td>17756514.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>0.1268</td>\n",
       "      <td>collagen alpha</td>\n",
       "      <td>17732555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1509</td>\n",
       "      <td>collagen-type XV/XVIII</td>\n",
       "      <td>4223297.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1104</td>\n",
       "      <td>ferritin</td>\n",
       "      <td>5925928.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1086</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20737281.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1126</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20782055.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20750669.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1143</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20766314.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1278</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20741452.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>golgi precursor</td>\n",
       "      <td>14538562.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>nell1</td>\n",
       "      <td>20764925.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1021</td>\n",
       "      <td>p19</td>\n",
       "      <td>30282433.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1154</td>\n",
       "      <td>rendezvin</td>\n",
       "      <td>13356234.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>sm30</td>\n",
       "      <td>160017.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1177</td>\n",
       "      <td>sm30</td>\n",
       "      <td>160379.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920879.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1093</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920847.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1017</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920844.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145606.1</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>thrombospondin</td>\n",
       "      <td>21959397.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr     fst                     hol         pos\n",
       "12  NW_022145615.1  0.1179              ATPsythase  31275084.0\n",
       "31  NW_022145603.1  0.1058                    Alx1  14678986.0\n",
       "30  NW_022145603.1  0.1059                    Alx1  14662026.0\n",
       "26  NW_022145603.1  0.1083                    Alx1  14694778.0\n",
       "8   NW_022145613.1  0.1216               B-catenin  19685318.0\n",
       "0   NW_022145596.1  0.1602                   CA10L  30320957.0\n",
       "43  NW_022145596.1  0.1013                   CA10L  30331129.0\n",
       "18  NW_022145596.1  0.1127                   CA10L  30330987.0\n",
       "34  NW_022145599.1  0.1049                   CD109   3635077.0\n",
       "46  NW_022145597.1  0.1006              COLP3alpha  20094418.0\n",
       "40  NW_022145597.1  0.1020              COLP3alpha  20094442.0\n",
       "47  NW_022145597.1  0.1006              COLP3alpha  20080205.0\n",
       "48  NW_022145597.1  0.1001              COLP3alpha  20071223.0\n",
       "45  NW_022145613.1  0.1010                Cara14la  27358332.0\n",
       "38  NW_022145605.1  0.1024                 Cara7la   5712045.0\n",
       "11  NW_022145605.1  0.1181                 Cara7la   5721058.0\n",
       "41  NW_022145605.1  0.1018                 Cara7la   5709506.0\n",
       "27  NW_022145605.1  0.1070                 Cara7la   5727140.0\n",
       "25  NW_022145605.1  0.1084                 Cara7la   5722898.0\n",
       "44  NW_022145614.1  0.1010                 Cara8LB   5781742.0\n",
       "21  NW_022145594.1  0.1101                    Ets1  41420756.0\n",
       "22  NW_022145594.1  0.1100                    NCAM  19199989.0\n",
       "1   NW_022145596.1  0.1561                 Vegfr10    627088.0\n",
       "15  NW_022145596.1  0.1149                 Vegfr10    669637.0\n",
       "35  NW_022145596.1  0.1042                 Vegfr10    670140.0\n",
       "7   NW_022145596.1  0.1250                 Vegfr10    639916.0\n",
       "33  NW_022145595.1  0.1056                    Wnt8   3240002.0\n",
       "4   NW_022145612.1  0.1292         calcium-sensing  12985454.0\n",
       "9   NW_022145597.1  0.1184          cholinesterase  14505911.0\n",
       "36  NW_022145602.1  0.1031                collagen   4993850.0\n",
       "32  NW_022145610.1  0.1057          collagen alpha  17756514.0\n",
       "6   NW_022145610.1  0.1268          collagen alpha  17732555.0\n",
       "2   NW_022145613.1  0.1509  collagen-type XV/XVIII   4223297.0\n",
       "20  NW_022145603.1  0.1104                ferritin   5925928.0\n",
       "24  NW_022145612.1  0.1086             fibrillin-2  20737281.0\n",
       "19  NW_022145612.1  0.1126             fibrillin-2  20782055.0\n",
       "17  NW_022145612.1  0.1139             fibrillin-2  20750669.0\n",
       "16  NW_022145612.1  0.1143             fibrillin-2  20766314.0\n",
       "5   NW_022145612.1  0.1278             fibrillin-2  20741452.0\n",
       "10  NW_022145614.1  0.1181         golgi precursor  14538562.0\n",
       "29  NW_022145602.1  0.1060                   nell1  20764925.0\n",
       "39  NW_022145596.1  0.1021                     p19  30282433.0\n",
       "14  NW_022145603.1  0.1154               rendezvin  13356234.0\n",
       "28  NW_022145602.1  0.1070                    sm30    160017.0\n",
       "13  NW_022145602.1  0.1177                    sm30    160379.0\n",
       "37  NW_022145594.1  0.1031                    sst3  28920879.0\n",
       "23  NW_022145594.1  0.1093                    sst3  28920847.0\n",
       "42  NW_022145594.1  0.1017                    sst3  28920844.0\n",
       "3   NW_022145606.1  0.1319          thrombospondin  21959397.0"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd_check.sort_values(by=[\"hol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf = new_df.merge(erd_check, on=[\"chr\",\"pos\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf=cdf.sort_values(by=[\"hol_y\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "49"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [],
   "source": [
    "int_check[\"pos\"]=int_check.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC100894079</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC579367</td>\n",
       "      <td>lnc</td>\n",
       "      <td>41191507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC105444100</td>\n",
       "      <td>lnc</td>\n",
       "      <td>20905720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC105444622</td>\n",
       "      <td>lnc</td>\n",
       "      <td>1523941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145543.1</td>\n",
       "      <td>LOC115917929</td>\n",
       "      <td>lnc</td>\n",
       "      <td>956594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC762981</td>\n",
       "      <td>lnc</td>\n",
       "      <td>5104468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC105444132</td>\n",
       "      <td>lnc</td>\n",
       "      <td>17378166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>NW_022145576.1</td>\n",
       "      <td>LOC100891941</td>\n",
       "      <td>lnc</td>\n",
       "      <td>2132339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC100893229</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22327453</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>130 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol       pos\n",
       "0    NW_022145601.1  LOC100894079   lnc  22606114\n",
       "1    NW_022145612.1     LOC579367   lnc  41191507\n",
       "2    NW_022145614.1  LOC105444100   lnc  20905720\n",
       "3    NW_022145610.1  LOC105444622   lnc   1523941\n",
       "4    NW_022145543.1  LOC115917929   lnc    956594\n",
       "..              ...           ...   ...       ...\n",
       "125  NW_022145612.1     LOC762981   lnc   5104468\n",
       "126  NW_022145598.1  LOC105444132   lnc  17378166\n",
       "127  NW_022145576.1  LOC100891941   lnc   2132339\n",
       "128  NW_022145612.1     LOC580713  lvar  14929793\n",
       "129  NW_022145605.1  LOC100893229   lnc  22327453\n",
       "\n",
       "[130 rows x 4 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "check = int_check.merge(cdf, on=[\"chr\",\"pos\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene_x</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene_y</th>\n",
       "      <th>hol_x</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, gene_x, hol, pos, gene_y, hol_x, fst, hol_y]\n",
       "Index: []"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Go terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['not_annot', 'exon', \"3'UTR\", 'intron', 'genes_overlap',\n",
       "       'alternative UTR', \"5'UTR\", 'pseudogene'], dtype=object)"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "for_go =new_df[(new_df[\"hol\"]==\"exon\") | (new_df[\"hol\"]==\"3'UTR\") | (new_df[\"hol\"]==\"intron\") | (new_df[\"hol\"]==\"5'UTR\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "for_go.to_csv(\"loci_for_go.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes=for_go.gene.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['protein_external_id', 'preferred_name']\n"
     ]
    }
   ],
   "source": [
    "with open(\"SPU_LOC_conv.txt\") as f:\n",
    "    mapping = list(map(lambda x: x.strip().split(\"\\t\")[:2],f.readlines()))\n",
    "    print(mapping[0])\n",
    "    mapping = {key:val for val,key in mapping[1:]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "alt_gene=[]\n",
    "for gene in genes:\n",
    "    if gene in mapping.keys():\n",
    "        alt = mapping[gene]\n",
    "        x = alt.split(\".\")\n",
    "        y = x[1].split(\"-\")\n",
    "        alt_gene.append(y[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.savetxt(\"for_go_spus.csv\", alt_gene, delimiter=',', fmt=\"%s\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "980"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(alt_gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
