{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from collections import Counter\n",
    "from matplotlib import pyplot\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop(df[df['type'] == \"region\"].index, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gene', 'mRNA', 'exon', 'CDS', 'snRNA', 'pseudogene', 'cDNA_match',\n",
       "       'tRNA', 'lnc_RNA', 'transcript', 'snoRNA', 'rRNA',\n",
       "       'primary_transcript', 'miRNA', 'sequence_feature', 'tandem_repeat',\n",
       "       'promoter'], dtype=object)"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "loci = pd.read_csv('top_goodhe_01', header=None, names=[\"chr\",\"fst\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>fst</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1_809765</td>\n",
       "      <td>0.2058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1_22684111</td>\n",
       "      <td>0.1950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1_36849648</td>\n",
       "      <td>0.1883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1_13121687</td>\n",
       "      <td>0.1882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1_20882520</td>\n",
       "      <td>0.1873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1_27269202</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1_25996126</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1_12214610</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1_3170334</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1_16380245</td>\n",
       "      <td>0.1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          chr     fst\n",
       "0       NW_022145599.1_809765  0.2058\n",
       "1     NW_022145613.1_22684111  0.1950\n",
       "2     NW_022145600.1_36849648  0.1883\n",
       "3     NW_022145612.1_13121687  0.1882\n",
       "4     NW_022145595.1_20882520  0.1873\n",
       "...                       ...     ...\n",
       "6842  NW_022145597.1_27269202  0.1000\n",
       "6843  NW_022145597.1_25996126  0.1000\n",
       "6844  NW_022145596.1_12214610  0.1000\n",
       "6845   NW_022145595.1_3170334  0.1000\n",
       "6846  NW_022145595.1_16380245  0.1000\n",
       "\n",
       "[6847 rows x 2 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "#t=loci.chr.to_numpy()\n",
    "#chrs=[i.split(\"_\")[:2] for i in t ]\n",
    "#chrs =[\"_\".join(i) for i in chrs]\n",
    "#pos=[i.split(\"_\")[2] for i in t ]\n",
    "#fixed_loci = pd.DataFrame(list(zip(chrs, pos)), columns =['chr', 'pos'])\n",
    "#np.savetxt(\"fixed_loci.csv\", t, delimiter=',', fmt=\"%s\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#---------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR1\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-2c392f1a2651>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"_\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mch\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mpos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"_\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m     \u001b[0mtemp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'chr'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0mch\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m&\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'start'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m<=\u001b[0m\u001b[0mpos\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m&\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'stop'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m>=\u001b[0m\u001b[0mpos\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtemp\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m         \u001b[0;31m#print(\"not in annotation\")\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/ops/common.py\u001b[0m in \u001b[0;36mnew_method\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     63\u001b[0m         \u001b[0mother\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mitem_from_zerodim\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     64\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 65\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     66\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     67\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mnew_method\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/arraylike.py\u001b[0m in \u001b[0;36m__eq__\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     27\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0munpack_zerodim_and_defer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"__eq__\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__eq__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 29\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cmp_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moperator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0meq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     30\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0munpack_zerodim_and_defer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"__ne__\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_cmp_method\u001b[0;34m(self, other, op)\u001b[0m\n\u001b[1;32m   4937\u001b[0m         \u001b[0mrvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mextract_numpy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4938\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4939\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcomparison_op\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4940\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4941\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_construct_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres_values\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mres_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomparison_op\u001b[0;34m(left, right, op)\u001b[0m\n\u001b[1;32m    241\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    242\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mis_object_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 243\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcomp_method_OBJECT_ARRAY\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    244\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    245\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/petak/lib/python3.8/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomp_method_OBJECT_ARRAY\u001b[0;34m(op, x, y)\u001b[0m\n\u001b[1;32m     53\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvec_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 55\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscalar_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     56\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "new_df=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    temp=df[(df['chr']==ch) & (df['start']<=pos) & (df['stop']>=pos)]\n",
    "    if len(temp) == 0:\n",
    "        #print(\"not in annotation\")\n",
    "        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"not_annot\", \"gene\" : \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "    if len(temp) == 1:\n",
    "        print(\"only one\")\n",
    "        print(temp['type'])\n",
    "    if len(temp) > 1:\n",
    "        #print(\"more than 1\")\n",
    "        #checking for gene overlap\n",
    "        ids=[]\n",
    "        for i in range(len(temp)):\n",
    "            t=temp[\"info\"].iloc[i].split(\";\")\n",
    "            res = list(filter(lambda x: \"LOC\" in x, t))\n",
    "            if len(res) == 0:\n",
    "                res = list(filter(lambda x: \"GeneID:\" in x, t))\n",
    "                if len(res) != 0:\n",
    "                    ID = res[0].split(\"GeneID:\",1)[1]\n",
    "                    ids.append(ID)\n",
    "            else:\n",
    "                ID =res[0].split(\"LOC\",1)[1]\n",
    "                ids.append(ID)\n",
    "        uni=len(Counter(ids))\n",
    "        gene_name = ids[0]\n",
    "        if uni > 1:\n",
    "            #print(\"genes overlap\")\n",
    "            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"genes_overlap\", \"gene\": \"NA\"},  \n",
    "                        ignore_index = True) \n",
    "        \n",
    "        if uni == 1:\n",
    "            if temp['type'].str.contains('gene').any():\n",
    "                if temp['type'].str.contains('pseudogene').any():\n",
    "                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"pseudogene\", \"gene\":\"NA\" },ignore_index = True)\n",
    "                else:\n",
    "                    if temp['type'].str.contains('exon').any(): \n",
    "                        if temp['type'].str.contains('CDS').any():\n",
    "                            #print(\"in exon\")\n",
    "                            new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"exon\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                        else:\n",
    "                            direction=temp['s2'].iloc[0]\n",
    "                            e_start=temp[temp[\"type\"]==\"exon\"].iloc[0].start\n",
    "                            e_stop=temp[temp[\"type\"]==\"exon\"].iloc[0].stop\n",
    "                            g_start=temp[temp[\"type\"]==\"gene\"].iloc[0].start\n",
    "                            g_stop=temp[temp[\"type\"]==\"gene\"].iloc[0].stop\n",
    "                            if direction == \"+\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('5prime1')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"3prime1\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                else:\n",
    "                                    print(\"ERROR1\") #alternative UTR?\n",
    "                                    #print(temp)\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            elif direction == \"-\":\n",
    "                                if e_start == g_start:\n",
    "                                    #print('3prime2')\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"3'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "                                elif e_stop == g_stop:\n",
    "                                    #print(\"5prime2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"5'UTR\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True)\n",
    "                                else:\n",
    "                                    print(\"ERROR2\")\n",
    "                                    new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"alternative UTR\", \"gene\":\"NA\" },ignore_index = True) \n",
    "                            else:\n",
    "                                print(\"ERROR3\")\n",
    "                                #print(temp)\n",
    "\n",
    "                    else:\n",
    "                        #print(\"in intron\")\n",
    "                        new_df = new_df.append({'chr' : ch, 'pos' : pos, 'hol' : \"intron\", \"gene\":f\"LOC{ids[0]}\" },ignore_index = True) \n",
    "\n",
    "            else:\n",
    "                print(\"not gene\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.pos = new_df.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.to_csv(\"annotated_top_goodhe_01.csv\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#------------------------------------------------"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df = pd.read_csv('annotated_top_goodhe_01.csv', sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>809765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>22684111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>36849648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>13121687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>20882520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6842</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>27269202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6843</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>25996126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6844</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alternative UTR</td>\n",
       "      <td>12214610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6845</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_annot</td>\n",
       "      <td>3170334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6846</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>LOC582877</td>\n",
       "      <td>intron</td>\n",
       "      <td>16380245</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6847 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr       gene              hol       pos\n",
       "0     NW_022145599.1        NaN        not_annot    809765\n",
       "1     NW_022145613.1        NaN        not_annot  22684111\n",
       "2     NW_022145600.1        NaN        not_annot  36849648\n",
       "3     NW_022145612.1        NaN        not_annot  13121687\n",
       "4     NW_022145595.1        NaN        not_annot  20882520\n",
       "...              ...        ...              ...       ...\n",
       "6842  NW_022145597.1        NaN        not_annot  27269202\n",
       "6843  NW_022145597.1        NaN        not_annot  25996126\n",
       "6844  NW_022145596.1        NaN  alternative UTR  12214610\n",
       "6845  NW_022145595.1        NaN        not_annot   3170334\n",
       "6846  NW_022145595.1  LOC582877           intron  16380245\n",
       "\n",
       "[6847 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_df[new_df[\"hol\"]==\"pseudogene\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['not_annot', 'exon', \"3'UTR\", 'intron', 'genes_overlap',\n",
       "       'alternative UTR', \"5'UTR\", 'pseudogene'], dtype=object)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.hol.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, outside of intron"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "are intronic ones located in:\n",
    "    1 ATAC\n",
    "    2 lvar\n",
    "    3 chip \n",
    "    4 lnc\n",
    "    5 promoter\n",
    "    6 TFBS\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoters = pd.read_csv('all_annotations.txt', sep=\"\\t\", header=None, names=[\"chr\",\"source\",\"type\",\"start\",\"stop\",\"s1\",\"s2\",\"s3\",\"info\"])\n",
    "promoters = promoters[promoters['type']==\"gene\"]\n",
    "\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"+\"), \"start\"] -= 1000\n",
    "promoters.loc[(promoters[\"type\"]==\"gene\") & (promoters[\"s2\"]==\"-\"), \"stop\"] += 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = pd.read_csv(\"atac_mapped_translatedto5.0.csv\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "chip = pd.read_csv('AlxChip_mapped_translatedto5.0.txt', sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lvar = pd.read_csv('lvar_spur_48_50_mapped_translatedto50.csv', header=None, names=[\"chr\",\"start\",\"stop\"])\n",
    "lnc = pd.read_csv(\"lncRNA_mapped_translatedto5.0.csv\", header=1, names=[\"chr\",\"start\",\"stop\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"not_annot\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in chip\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lvar\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in chip\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in chip\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in chip\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lvar\n",
      "in chip\n",
      "in chip\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lvar\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in promoters\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n",
      "in promoters\n",
      "in promoters\n",
      "in lnc\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "enh_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\" },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\" },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\" },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\" },ignore_index = True)\n",
    "    temp=promoters[(promoters['chr']==ch) & (promoters['start']<=pos) & (promoters['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in promoters\")\n",
    "        enh_check = enh_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"promoter\" },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['promoter', 'lnc', 'lvar', 'chip'], dtype=object)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enh_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_promoters = enh_check[enh_check[\"hol\"]==\"promoter\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en = enh_check[(enh_check[\"hol\"]==\"lvar\") | (enh_check[\"hol\"]==\"chip\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_en=pot_en.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pot_en)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reminder: chosen atac supp (or lvar or chip) too strict?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## now let's see if there is any in enhancer, inside of intron"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "mis = new_df[new_df['hol']==\"intron\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in atac\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in chip\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in atac\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lnc\n",
      "in lvar\n",
      "in lnc\n"
     ]
    }
   ],
   "source": [
    "int_check=pd.DataFrame()\n",
    "for index, row in mis.iterrows():\n",
    "    ch = row.chr\n",
    "    pos = row.pos\n",
    "    gene=row.gene\n",
    "    temp=atac[(atac['chr']==ch) & (atac['start']<=pos) & (atac['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in atac\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"atac\",'gene':gene },ignore_index = True) \n",
    "    temp=chip[(chip['chr']==ch) & (chip['start']<=pos) & (chip['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in chip\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"chip\",'gene':gene },ignore_index = True)\n",
    "    temp=lvar[(lvar['chr']==ch) & (lvar['start']<=pos) & (lvar['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lvar\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lvar\",'gene':gene },ignore_index = True)\n",
    "    temp=lnc[(lnc['chr']==ch) & (lnc['start']<=pos) & (lnc['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        print(\"in lnc\")\n",
    "        int_check = int_check.append({'chr' : ch, 'pos' : pos, 'hol' : \"lnc\",'gene':gene },ignore_index = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['lnc', 'chip', 'atac', 'lvar'], dtype=object)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>NW_022145604.1</td>\n",
       "      <td>LOC582776</td>\n",
       "      <td>lvar</td>\n",
       "      <td>8855042.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>NW_022145600.1</td>\n",
       "      <td>LOC577823</td>\n",
       "      <td>lvar</td>\n",
       "      <td>18662627.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580332</td>\n",
       "      <td>lvar</td>\n",
       "      <td>47720769.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC100894087</td>\n",
       "      <td>lvar</td>\n",
       "      <td>35316800.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>LOC115921692</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16140373.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC588549</td>\n",
       "      <td>lvar</td>\n",
       "      <td>16695182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol         pos\n",
       "83   NW_022145604.1     LOC582776  lvar   8855042.0\n",
       "96   NW_022145600.1     LOC577823  lvar  18662627.0\n",
       "97   NW_022145612.1     LOC580332  lvar  47720769.0\n",
       "100  NW_022145610.1  LOC100894087  lvar  35316800.0\n",
       "114  NW_022145595.1  LOC115921692  lvar  16140373.0\n",
       "118  NW_022145605.1     LOC588549  lvar  16695182.0\n",
       "142  NW_022145612.1     LOC580713  lvar  14929793.0"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check[int_check[\"hol\"]==\"lvar\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in = int_check[(int_check[\"hol\"]==\"lvar\") | (int_check[\"hol\"]==\"chip\") | (int_check[\"hol\"]==\"atac\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "pot_in=pot_in.sort_values(by=[\"chr\",'pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>LOC752651</td>\n",
       "      <td>lnc</td>\n",
       "      <td>15709285.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC100894079</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC579367</td>\n",
       "      <td>lnc</td>\n",
       "      <td>41191507.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC105444100</td>\n",
       "      <td>lnc</td>\n",
       "      <td>20905720.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC105444622</td>\n",
       "      <td>lnc</td>\n",
       "      <td>1523941.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC105441528</td>\n",
       "      <td>lnc</td>\n",
       "      <td>9915090.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC105444132</td>\n",
       "      <td>lnc</td>\n",
       "      <td>17378166.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>NW_022145576.1</td>\n",
       "      <td>LOC100891941</td>\n",
       "      <td>lnc</td>\n",
       "      <td>2132339.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC100893229</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22327453.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol         pos\n",
       "0    NW_022145603.1     LOC752651   lnc  15709285.0\n",
       "1    NW_022145601.1  LOC100894079   lnc  22606114.0\n",
       "2    NW_022145612.1     LOC579367   lnc  41191507.0\n",
       "3    NW_022145614.1  LOC105444100   lnc  20905720.0\n",
       "4    NW_022145610.1  LOC105444622   lnc   1523941.0\n",
       "..              ...           ...   ...         ...\n",
       "139  NW_022145594.1  LOC105441528   lnc   9915090.0\n",
       "140  NW_022145598.1  LOC105444132   lnc  17378166.0\n",
       "141  NW_022145576.1  LOC100891941   lnc   2132339.0\n",
       "142  NW_022145612.1     LOC580713  lvar  14929793.0\n",
       "143  NW_022145605.1  LOC100893229   lnc  22327453.0\n",
       "\n",
       "[144 rows x 4 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## let's check for my old list (what i wanted to get probes for)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd = pd.read_csv('my_old_list.csv', header=None, names=[\"chr\",\"start\",\"stop\",\"hol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd.drop([0], inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd=erd.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd[\"start\"]=erd.start.astype(float)\n",
    "erd[\"stop\"]=erd.stop.astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Otx'"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd[\"hol\"].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "erd_check=pd.DataFrame()\n",
    "for index, row in loci.iterrows():\n",
    "    ch = row[0].split(\"_\")[:2]\n",
    "    ch = \"_\".join(ch)\n",
    "    pos = int(row[0].split(\"_\")[2])\n",
    "    fst = row[1]\n",
    "    temp= erd[(erd['chr']==ch) & (erd['start']<=pos) & (erd['stop']>=pos)]\n",
    "    if len(temp) > 0:\n",
    "        erd_check = erd_check.append({'chr' : ch, 'pos' : pos, 'hol' : temp[\"hol\"].iloc[0], \"fst\" : fst},ignore_index = True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NW_022145615.1</td>\n",
       "      <td>0.1179</td>\n",
       "      <td>ATPsythase</td>\n",
       "      <td>31275084.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1058</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14678986.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1059</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14662026.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1083</td>\n",
       "      <td>Alx1</td>\n",
       "      <td>14694778.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1216</td>\n",
       "      <td>B-catenin</td>\n",
       "      <td>19685318.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1602</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30320957.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1013</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30331129.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1127</td>\n",
       "      <td>CA10L</td>\n",
       "      <td>30330987.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NW_022145599.1</td>\n",
       "      <td>0.1049</td>\n",
       "      <td>CD109</td>\n",
       "      <td>3635077.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20094418.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1020</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20094442.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1006</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20080205.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1001</td>\n",
       "      <td>COLP3alpha</td>\n",
       "      <td>20071223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara14la</td>\n",
       "      <td>27358332.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1024</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5712045.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5721058.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1018</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5709506.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5727140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>0.1084</td>\n",
       "      <td>Cara7la</td>\n",
       "      <td>5722898.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>Cara8LB</td>\n",
       "      <td>5781742.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1101</td>\n",
       "      <td>Ets1</td>\n",
       "      <td>41420756.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1100</td>\n",
       "      <td>NCAM</td>\n",
       "      <td>19199989.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1561</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>627088.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1149</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>669637.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1042</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>670140.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1250</td>\n",
       "      <td>Vegfr10</td>\n",
       "      <td>639916.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NW_022145595.1</td>\n",
       "      <td>0.1056</td>\n",
       "      <td>Wnt8</td>\n",
       "      <td>3240002.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1292</td>\n",
       "      <td>calcium-sensing</td>\n",
       "      <td>12985454.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NW_022145597.1</td>\n",
       "      <td>0.1184</td>\n",
       "      <td>cholinesterase</td>\n",
       "      <td>14505911.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>collagen</td>\n",
       "      <td>4993850.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>0.1057</td>\n",
       "      <td>collagen alpha</td>\n",
       "      <td>17756514.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>0.1268</td>\n",
       "      <td>collagen alpha</td>\n",
       "      <td>17732555.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145613.1</td>\n",
       "      <td>0.1509</td>\n",
       "      <td>collagen-type XV/XVIII</td>\n",
       "      <td>4223297.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1104</td>\n",
       "      <td>ferritin</td>\n",
       "      <td>5925928.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1086</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20737281.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1126</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20782055.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1139</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20750669.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1143</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20766314.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>0.1278</td>\n",
       "      <td>fibrillin-2</td>\n",
       "      <td>20741452.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>0.1181</td>\n",
       "      <td>golgi precursor</td>\n",
       "      <td>14538562.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>nell1</td>\n",
       "      <td>20764925.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NW_022145596.1</td>\n",
       "      <td>0.1021</td>\n",
       "      <td>p19</td>\n",
       "      <td>30282433.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>0.1154</td>\n",
       "      <td>rendezvin</td>\n",
       "      <td>13356234.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1070</td>\n",
       "      <td>sm30</td>\n",
       "      <td>160017.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NW_022145602.1</td>\n",
       "      <td>0.1177</td>\n",
       "      <td>sm30</td>\n",
       "      <td>160379.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1031</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920879.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1093</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920847.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>0.1017</td>\n",
       "      <td>sst3</td>\n",
       "      <td>28920844.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145606.1</td>\n",
       "      <td>0.1319</td>\n",
       "      <td>thrombospondin</td>\n",
       "      <td>21959397.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               chr     fst                     hol         pos\n",
       "12  NW_022145615.1  0.1179              ATPsythase  31275084.0\n",
       "31  NW_022145603.1  0.1058                    Alx1  14678986.0\n",
       "30  NW_022145603.1  0.1059                    Alx1  14662026.0\n",
       "26  NW_022145603.1  0.1083                    Alx1  14694778.0\n",
       "8   NW_022145613.1  0.1216               B-catenin  19685318.0\n",
       "0   NW_022145596.1  0.1602                   CA10L  30320957.0\n",
       "43  NW_022145596.1  0.1013                   CA10L  30331129.0\n",
       "18  NW_022145596.1  0.1127                   CA10L  30330987.0\n",
       "34  NW_022145599.1  0.1049                   CD109   3635077.0\n",
       "46  NW_022145597.1  0.1006              COLP3alpha  20094418.0\n",
       "40  NW_022145597.1  0.1020              COLP3alpha  20094442.0\n",
       "47  NW_022145597.1  0.1006              COLP3alpha  20080205.0\n",
       "48  NW_022145597.1  0.1001              COLP3alpha  20071223.0\n",
       "45  NW_022145613.1  0.1010                Cara14la  27358332.0\n",
       "38  NW_022145605.1  0.1024                 Cara7la   5712045.0\n",
       "11  NW_022145605.1  0.1181                 Cara7la   5721058.0\n",
       "41  NW_022145605.1  0.1018                 Cara7la   5709506.0\n",
       "27  NW_022145605.1  0.1070                 Cara7la   5727140.0\n",
       "25  NW_022145605.1  0.1084                 Cara7la   5722898.0\n",
       "44  NW_022145614.1  0.1010                 Cara8LB   5781742.0\n",
       "21  NW_022145594.1  0.1101                    Ets1  41420756.0\n",
       "22  NW_022145594.1  0.1100                    NCAM  19199989.0\n",
       "1   NW_022145596.1  0.1561                 Vegfr10    627088.0\n",
       "15  NW_022145596.1  0.1149                 Vegfr10    669637.0\n",
       "35  NW_022145596.1  0.1042                 Vegfr10    670140.0\n",
       "7   NW_022145596.1  0.1250                 Vegfr10    639916.0\n",
       "33  NW_022145595.1  0.1056                    Wnt8   3240002.0\n",
       "4   NW_022145612.1  0.1292         calcium-sensing  12985454.0\n",
       "9   NW_022145597.1  0.1184          cholinesterase  14505911.0\n",
       "36  NW_022145602.1  0.1031                collagen   4993850.0\n",
       "32  NW_022145610.1  0.1057          collagen alpha  17756514.0\n",
       "6   NW_022145610.1  0.1268          collagen alpha  17732555.0\n",
       "2   NW_022145613.1  0.1509  collagen-type XV/XVIII   4223297.0\n",
       "20  NW_022145603.1  0.1104                ferritin   5925928.0\n",
       "24  NW_022145612.1  0.1086             fibrillin-2  20737281.0\n",
       "19  NW_022145612.1  0.1126             fibrillin-2  20782055.0\n",
       "17  NW_022145612.1  0.1139             fibrillin-2  20750669.0\n",
       "16  NW_022145612.1  0.1143             fibrillin-2  20766314.0\n",
       "5   NW_022145612.1  0.1278             fibrillin-2  20741452.0\n",
       "10  NW_022145614.1  0.1181         golgi precursor  14538562.0\n",
       "29  NW_022145602.1  0.1060                   nell1  20764925.0\n",
       "39  NW_022145596.1  0.1021                     p19  30282433.0\n",
       "14  NW_022145603.1  0.1154               rendezvin  13356234.0\n",
       "28  NW_022145602.1  0.1070                    sm30    160017.0\n",
       "13  NW_022145602.1  0.1177                    sm30    160379.0\n",
       "37  NW_022145594.1  0.1031                    sst3  28920879.0\n",
       "23  NW_022145594.1  0.1093                    sst3  28920847.0\n",
       "42  NW_022145594.1  0.1017                    sst3  28920844.0\n",
       "3   NW_022145606.1  0.1319          thrombospondin  21959397.0"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "erd_check.sort_values(by=[\"hol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf = new_df.merge(erd_check, on=[\"chr\",\"pos\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdf=cdf.sort_values(by=[\"hol_y\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "49"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "int_check[\"pos\"]=int_check.pos.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_022145603.1</td>\n",
       "      <td>LOC752651</td>\n",
       "      <td>lnc</td>\n",
       "      <td>15709285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_022145601.1</td>\n",
       "      <td>LOC100894079</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22606114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC579367</td>\n",
       "      <td>lnc</td>\n",
       "      <td>41191507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_022145614.1</td>\n",
       "      <td>LOC105444100</td>\n",
       "      <td>lnc</td>\n",
       "      <td>20905720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_022145610.1</td>\n",
       "      <td>LOC105444622</td>\n",
       "      <td>lnc</td>\n",
       "      <td>1523941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>NW_022145594.1</td>\n",
       "      <td>LOC105441528</td>\n",
       "      <td>lnc</td>\n",
       "      <td>9915090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>NW_022145598.1</td>\n",
       "      <td>LOC105444132</td>\n",
       "      <td>lnc</td>\n",
       "      <td>17378166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>NW_022145576.1</td>\n",
       "      <td>LOC100891941</td>\n",
       "      <td>lnc</td>\n",
       "      <td>2132339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>NW_022145612.1</td>\n",
       "      <td>LOC580713</td>\n",
       "      <td>lvar</td>\n",
       "      <td>14929793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>NW_022145605.1</td>\n",
       "      <td>LOC100893229</td>\n",
       "      <td>lnc</td>\n",
       "      <td>22327453</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                chr          gene   hol       pos\n",
       "0    NW_022145603.1     LOC752651   lnc  15709285\n",
       "1    NW_022145601.1  LOC100894079   lnc  22606114\n",
       "2    NW_022145612.1     LOC579367   lnc  41191507\n",
       "3    NW_022145614.1  LOC105444100   lnc  20905720\n",
       "4    NW_022145610.1  LOC105444622   lnc   1523941\n",
       "..              ...           ...   ...       ...\n",
       "139  NW_022145594.1  LOC105441528   lnc   9915090\n",
       "140  NW_022145598.1  LOC105444132   lnc  17378166\n",
       "141  NW_022145576.1  LOC100891941   lnc   2132339\n",
       "142  NW_022145612.1     LOC580713  lvar  14929793\n",
       "143  NW_022145605.1  LOC100893229   lnc  22327453\n",
       "\n",
       "[144 rows x 4 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "int_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "check = int_check.merge(cdf, on=[\"chr\",\"pos\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>gene_x</th>\n",
       "      <th>hol</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene_y</th>\n",
       "      <th>hol_x</th>\n",
       "      <th>fst</th>\n",
       "      <th>hol_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, gene_x, hol, pos, gene_y, hol_x, fst, hol_y]\n",
       "Index: []"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Go terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['not_annot', 'exon', \"3'UTR\", 'intron', 'genes_overlap',\n",
       "       'alternative UTR', \"5'UTR\", 'pseudogene'], dtype=object)"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_df.hol.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "for_go =new_df[(new_df[\"hol\"]==\"exon\") | (new_df[\"hol\"]==\"3'UTR\") | (new_df[\"hol\"]==\"intron\") | (new_df[\"hol\"]==\"5'UTR\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "for_go.to_csv(\"loci_for_go.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes=for_go.gene.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['protein_external_id', 'preferred_name']\n"
     ]
    }
   ],
   "source": [
    "with open(\"SPU_LOC_conv.txt\") as f:\n",
    "    mapping = list(map(lambda x: x.strip().split(\"\\t\")[:2],f.readlines()))\n",
    "    print(mapping[0])\n",
    "    mapping = {key:val for val,key in mapping[1:]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "alt_gene=[]\n",
    "for gene in genes:\n",
    "    if gene in mapping.keys():\n",
    "        alt = mapping[gene]\n",
    "        x = alt.split(\".\")\n",
    "        y = x[1].split(\"-\")\n",
    "        alt_gene.append(y[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.savetxt(\"for_go_spus.csv\", alt_gene, delimiter=',', fmt=\"%s\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "980"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(np.unique(alt_gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
